<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />

<meta name="date" content="2017-02-08" />

<title>Batch effects of conversion efficiency</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Batch effects of conversion efficiency</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2017-02-08</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-03-06</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 4a0bfb9</p>
<div id="objective" class="section level3">
<h3>Objective</h3>
<p>Capture efficiency: how many of the added ERCC spike-in molecules were successfully sequenced</p>
<p>Compute conversion efficiency of each cell using the filerted set of cells. Compare conversion efficiency between batches: is there an individual effect or batch effect of coversion efficiency?</p>
<p>The definition of conversion efficiency: total molecule counts/ total read counts</p>
<p>We will use analysis of variance to test whether conversion rates are significantly different between individuals. Results below are described at an alpha level of .05. (Analysis of variance (ANOVA) is used to determine whether there are any significant differences between the means of three or more independent (unrelated) groups)</p>
</div>
<div id="input" class="section level3">
<h3>Input</h3>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;ggplot2&quot;)
library(&quot;cowplot&quot;)
theme_set(theme_bw(base_size = 16))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank(),
             legend.key = element_blank(),
             plot.title = element_text(size = rel(1)))
library(&quot;edgeR&quot;)
source(&quot;../../singleCellSeq/analysis/functions.R&quot;)</code></pre>
<p>Input annotation file.</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../../singleCellSeq/data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A03 NA19098.r1 NA19098.r1.A03
4    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
5    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
6    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06</code></pre>
<p>Input filtered read count data.</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../data/reads-filter.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input filtered molecule count data.</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input list of quality cells.</p>
<pre class="r"><code>quality_single_cells &lt;- scan(&quot;../data/quality-single-cells.txt&quot;,
                             what = &quot;character&quot;)</code></pre>
<p>Subset data so we only keep cells that passed QC.</p>
<pre class="r"><code>reads &lt;- reads[, colnames(reads) %in% quality_single_cells]
molecules &lt;- molecules[, colnames(molecules) %in% quality_single_cells]
anno &lt;- anno[anno$sample_id %in% quality_single_cells, ]
stopifnot(dim(reads) == dim(molecules),
          nrow(anno) == ncol(reads))</code></pre>
<p>Seperate ERCC and endogenous genes</p>
<pre class="r"><code>#seperate reads
reads_ENSG &lt;- reads[grep(&quot;ERCC&quot;, rownames(reads), invert = TRUE), ] #way to say not what you greped  
reads_ERCC &lt;- reads[grep(&quot;ERCC&quot;, rownames(reads), invert = FALSE), ]
#seperate moleclues  
molecules_ENSG &lt;- molecules[grep(&quot;ERCC&quot;, rownames(molecules), invert = TRUE), ]
molecules_ERCC &lt;- molecules[grep(&quot;ERCC&quot;, rownames(molecules), invert = FALSE), ]</code></pre>
</div>
<div id="reads-to-molecule-capture-efficiency" class="section level3">
<h3>Reads to molecule capture efficiency</h3>
<pre class="r"><code>#efficency for reads 
total_counts_ERCC &lt;- data.frame(total_reads = colSums(reads_ERCC),
                                total_molecules = colSums(molecules_ERCC))
# with is a function to modify  original data, this is like mutate because it adds a cloumn to the data
total_counts_ERCC$conversion &lt;- with(total_counts_ERCC,
                                     total_molecules/total_reads)
summary(total_counts_ERCC)</code></pre>
<pre><code>  total_reads    total_molecules   conversion     
 Min.   :10906   Min.   : 666    Min.   :0.01403  
 1st Qu.:31564   1st Qu.: 914    1st Qu.:0.02321  
 Median :38934   Median :1028    Median :0.02740  
 Mean   :39700   Mean   :1102    Mean   :0.02965  
 3rd Qu.:46506   3rd Qu.:1276    3rd Qu.:0.03427  
 Max.   :88786   Max.   :2035    Max.   :0.07813  </code></pre>
<pre class="r"><code>#efficiency for endogenous genes 
total_counts_ENSG &lt;- data.frame(total_reads = colSums(reads_ENSG),
                                total_molecules = colSums(molecules_ENSG))
total_counts_ENSG$conversion &lt;- with(total_counts_ENSG,
                                     total_molecules/total_reads)
summary(total_counts_ENSG)</code></pre>
<pre><code>  total_reads      total_molecules    conversion     
 Min.   : 625855   Min.   : 30408   Min.   :0.01712  
 1st Qu.:1959019   1st Qu.: 55561   1st Qu.:0.02471  
 Median :2396897   Median : 65636   Median :0.02882  
 Mean   :2334326   Mean   : 66634   Mean   :0.02944  
 3rd Qu.:2727310   3rd Qu.: 78173   3rd Qu.:0.03318  
 Max.   :4143812   Max.   :113832   Max.   :0.07225  </code></pre>
<pre class="r"><code>boxplot(total_counts_ERCC$conversion, total_counts_ENSG$conversion, names = c(&quot;ERCC&quot;, &quot;ENSG&quot;), main=&quot;Summary of conversion efficency&quot;)</code></pre>
<p><img src="figure/batch_effect_conversion_efficency.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /> This shows the capture efficiency is about the same between ERCC and ENSG but the variability is higher in the ERCCs.</p>
<pre class="r"><code>## create a color palette with one color per individual and different shades for repplicates
great_color &lt;- c(&quot;#CC3300&quot;, &quot;#FF9966&quot;, &quot;#006633&quot;, &quot;#009900&quot;, &quot;#99FF99&quot;, &quot;#3366FF&quot;, &quot;#6699FF&quot;, &quot;#66CCFF&quot;)

depth_plot_ensg &lt;- ggplot(data.frame(total_counts_ENSG,
                  batch = anno$batch), 
       aes(x = total_reads/10^6, y = total_molecules/10^3, col = factor(batch))) +
  geom_point() +
  scale_color_manual(values = great_color) +
  labs(x = &quot;Total number of reads (x10^6)&quot;,
       y = &quot;Total number of molecules (x10^3)&quot;,
       title = &quot;Effect of read depth on single cells (genes)&quot;)
#compare molecule to read counts 
depth_plot_ensg</code></pre>
<p><img src="figure/batch_effect_conversion_efficency.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /> This shows that as we get more reads he molecule number also goes up. Shows we are not at a saturdation.</p>
<pre class="r"><code>depth_plot_ercc &lt;- ggplot(data.frame(total_counts_ERCC,
                  batch = anno$batch), 
       aes(x = total_reads/10^6, y = total_molecules/10^3, col = factor(batch))) +
  geom_point() +
  scale_color_manual(values = great_color) +
  labs(x = &quot;Total number of reads (x10^6)&quot;,
       y = &quot;Total number of molecules (x10^3)&quot;,
       title = &quot;Effect of read depth on single cells (ERCC spike-in)&quot;)

depth_plot_ercc</code></pre>
<p><img src="figure/batch_effect_conversion_efficency.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /> This shows that for some ERCCs we did hit saturation because molecule counts are not increasing as the molecule count does.</p>
</div>
<div id="compare-variance-of-conversion-rates-between-batches" class="section level3">
<h3>Compare variance of conversion rates between batches</h3>
<div id="ensg" class="section level4">
<h4>ENSG</h4>
<pre class="r"><code>convertion_ensg &lt;- ggplot(data.frame(total_counts_ENSG,
                  batch = anno$batch),
       aes(x = factor(batch), y = conversion,
           fill = factor(batch)), height = 600, width = 2000) +
geom_violin(alpha = .5) + 
geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
  scale_fill_manual(values = great_color) +
labs(x = &quot;Batch&quot;, y = &quot;Read-to-molecule conversion efficiency&quot;, title = &quot;Conversion efficiency of reads to molecules \n of endogenous genes are affected \n by individual and technical C1 batch&quot;) +
theme(axis.text.x = element_text(hjust=1, angle = 45))

convertion_ensg</code></pre>
<p><img src="figure/batch_effect_conversion_efficency.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>Between individual</strong></p>
<p>For individual <span class="math inline">\(i\)</span>, sample <span class="math inline">\(j\)</span>, we have <span class="math inline">\(n_{ij}\)</span> cells. Here we compare the per cell conversion rate from molecule count to read count between the three individuals:</p>
<p><span class="math display">\[ Y_{ijk} = \mu_i + b_{ij} + \epsilon_{ijk} \]</span></p>
<p>where <span class="math inline">\(i = 1, 2, 3\)</span>, <span class="math inline">\(j = 1, 2, 3\)</span>, and <span class="math inline">\(k = 1, 2, \dots, n_{ij}\)</span>.</p>
<pre class="r"><code>df_ENSG &lt;- data.frame(total_counts_ENSG,
                      individual = factor(anno$individual),
                      replicate = factor(anno$replicate),
                      batch = factor(anno$batch))
#factor()- used to encode vector as a factor(category) - we added the annotation data to the ENSG count data frame in this step

library(lme4)</code></pre>
<pre><code>
Attaching package: &#39;lme4&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:nlme&#39;:

    lmList</code></pre>
<pre class="r"><code>#this package is used to fit and analyzed mixed models (linear/generalized linear/nonlinear)


lmer_ENSG_full &lt;- lmer(conversion ~ individual + (1| individual:replicate), 
                        data = df_ENSG)
lmer_ENSG_null &lt;- lmer(conversion ~ 1 + (1| individual:replicate), 
                        data = df_ENSG)
anova(lmer_ENSG_null, lmer_ENSG_full)</code></pre>
<pre><code>refitting model(s) with ML (instead of REML)</code></pre>
<pre><code>Data: df_ENSG
Models:
lmer_ENSG_null: conversion ~ 1 + (1 | individual:replicate)
lmer_ENSG_full: conversion ~ individual + (1 | individual:replicate)
               Df     AIC     BIC logLik deviance  Chisq Chi Df Pr(&gt;Chisq)
lmer_ENSG_null  3 -4506.8 -4493.8 2256.4  -4512.8                         
lmer_ENSG_full  5 -4523.8 -4502.2 2266.9  -4533.8 21.077      2  2.649e-05
                  
lmer_ENSG_null    
lmer_ENSG_full ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p><strong>for each individaul and between batches</strong></p>
<pre class="r"><code>for (i in 1:length(unique(anno$individual))) {
  print(unique(anno$individual)[i])
  select_individual &lt;- with(anno, individual == unique(individual)[i])
  print( summary(aov(conversion ~ batch, 
                     data = df_ENSG[select_individual, ]) ) )
}</code></pre>
<pre><code>[1] &quot;NA19098&quot;
             Df   Sum Sq   Mean Sq F value Pr(&gt;F)
batch         1 0.000035 3.474e-05    0.88   0.35
Residuals   140 0.005528 3.949e-05               
[1] &quot;NA19101&quot;
             Df   Sum Sq   Mean Sq F value  Pr(&gt;F)   
batch         2 0.000125 6.250e-05   5.722 0.00384 **
Residuals   198 0.002163 1.092e-05                   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
[1] &quot;NA19239&quot;
             Df   Sum Sq  Mean Sq F value   Pr(&gt;F)    
batch         2 0.000750 3.75e-04   32.11 5.99e-13 ***
Residuals   218 0.002546 1.17e-05                     
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>Shows a batch effect between replicates of individuals NA19101 and NA19239.</p>
</div>
<div id="ercc" class="section level4">
<h4>ERCC</h4>
<pre class="r"><code>convertion_ercc &lt;- ggplot(data.frame(total_counts_ERCC,
                  batch = anno$batch),
       aes(x = factor(batch), y = conversion,
           fill = factor(batch)), height = 600, width = 2000) +
geom_violin(alpha = .5) + 
geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
  scale_fill_manual(values = great_color) +
labs(x = &quot;Batch&quot;, y = &quot;Read-to-molecule conversion efficiency&quot;, title = &quot;Conversion efficiency of reads to molecules \n of ERCC controls are affected \n by individual and technical C1 batch&quot;) +
theme(axis.text.x = element_text(hjust=1, angle = 45))

convertion_ercc</code></pre>
<p><img src="figure/batch_effect_conversion_efficency.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>Between individual</strong></p>
<pre class="r"><code>df_ERCC &lt;- data.frame(total_counts_ERCC,
                      individual = factor(anno$individual),
                      replicate = factor(anno$replicate),
                      batch = factor(anno$batch))

library(lme4)
lmer_ERCC_full &lt;- lmer(conversion ~ individual + (1| individual:replicate), 
                        data = df_ERCC)
lmer_ERCC_null &lt;- lmer(conversion ~ 1 + (1| individual:replicate), 
                        data = df_ERCC)
anova(lmer_ERCC_null, lmer_ERCC_full)</code></pre>
<pre><code>refitting model(s) with ML (instead of REML)</code></pre>
<pre><code>Data: df_ERCC
Models:
lmer_ERCC_null: conversion ~ 1 + (1 | individual:replicate)
lmer_ERCC_full: conversion ~ individual + (1 | individual:replicate)
               Df     AIC     BIC logLik deviance  Chisq Chi Df Pr(&gt;Chisq)
lmer_ERCC_null  3 -3811.1 -3798.1 1908.5  -3817.1                         
lmer_ERCC_full  5 -3827.1 -3805.4 1918.5  -3837.1 19.994      2  4.553e-05
                  
lmer_ERCC_null    
lmer_ERCC_full ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p><strong>For each individual between batches</strong></p>
<pre class="r"><code>for (i in 1:length(unique(anno$individual))) {
  print(unique(anno$individual)[i])
  select_individual &lt;- with(anno, individual == unique(individual)[i])
  print( summary(aov(conversion ~ batch, 
                     data = df_ERCC[select_individual, ]) ) )
}</code></pre>
<pre><code>[1] &quot;NA19098&quot;
             Df   Sum Sq   Mean Sq F value Pr(&gt;F)
batch         1 0.000181 1.813e-04   1.823  0.179
Residuals   140 0.013922 9.944e-05               
[1] &quot;NA19101&quot;
             Df   Sum Sq   Mean Sq F value Pr(&gt;F)
batch         2 0.000008 4.190e-06   0.125  0.883
Residuals   198 0.006637 3.352e-05               
[1] &quot;NA19239&quot;
             Df   Sum Sq  Mean Sq F value   Pr(&gt;F)    
batch         2 0.001112 5.56e-04   8.019 0.000436 ***
Residuals   218 0.015114 6.93e-05                     
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>ERCC batch effect only shows up for NA19239.</p>
</div>
</div>
<div id="summary-plots" class="section level3">
<h3>Summary Plots</h3>
<pre class="r"><code>library(gridExtra)</code></pre>
<pre><code>
Attaching package: &#39;gridExtra&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:Biobase&#39;:

    combine</code></pre>
<pre><code>The following object is masked from &#39;package:BiocGenerics&#39;:

    combine</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    combine</code></pre>
<pre class="r"><code>grid.arrange(
  ggplot(data.frame(total_counts_ERCC,
                    batch = anno$batch),
         aes(x = factor(batch), y = conversion,
             fill = factor(batch)), height = 600, width = 2000) +
  geom_violin(alpha = .5) + 
  geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
    scale_fill_manual(values = great_color) +
  labs(x = &quot;Batch&quot;, y = &quot;Conversion rate&quot;, title = &quot;Batch effect of conversion rate (ERCC spike-in)&quot;) +
  theme(axis.text.x = element_text(hjust=1, angle = 45))
  ,
  ggplot(data.frame(total_counts_ENSG,
                    batch = anno$batch),
         aes(x = factor(batch), y = conversion,
             fill = factor(batch)), height = 600, width = 2000) +
  geom_violin(alpha = .5) + 
  geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
    scale_fill_manual(values = great_color) +
  labs(x = &quot;Batch&quot;, y = &quot;Conversion rate&quot;, title = &quot;Batch effect of conversion rate (genes)&quot;) +
  theme(axis.text.x = element_text(hjust=1, angle = 45)),
  ncol = 2
)</code></pre>
<p><img src="figure/batch_effect_conversion_efficency.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_grid(depth_plot_ensg + theme(legend.position = &quot;none&quot;), 
          convertion_ensg + theme(legend.position = &quot;none&quot;),
          depth_plot_ercc + theme(legend.position = &quot;none&quot;),
          convertion_ercc + theme(legend.position = &quot;none&quot;),
          labels = LETTERS[1:4])</code></pre>
<p><img src="figure/batch_effect_conversion_efficency.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_grid(convertion_ensg + theme(legend.position = &quot;none&quot;),
          convertion_ercc + theme(legend.position = &quot;none&quot;),
          labels = letters[3:4])</code></pre>
<p><img src="figure/batch_effect_conversion_efficency.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>need help with fitting the model and interpretting anova tables</strong></p>
<p>This analysis provides evidence for batch effects in individual and technical replicates for ERCC and for endogenous gene conversion rate.</p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] gridExtra_2.2.1        lme4_1.1-12            workflowr_0.4.0       
 [4] org.Hs.eg.db_3.4.0     GOstats_2.40.0         graph_1.52.0          
 [7] Category_2.40.0        Matrix_1.2-7.1         GO.db_3.4.0           
[10] AnnotationDbi_1.36.2   IRanges_2.8.1          S4Vectors_0.12.1      
[13] Biobase_2.34.0         BiocGenerics_0.20.0    Humanzee_0.1.0        
[16] sva_3.22.0             genefilter_1.56.0      mgcv_1.8-15           
[19] nlme_3.1-128           BiocInstaller_1.24.0   scales_0.4.1          
[22] VennDiagram_1.6.17     futile.logger_1.4.3    edgeR_3.16.5          
[25] limma_3.30.11          dplyr_0.5.0            testit_0.6            
[28] data.table_1.10.4      biomaRt_2.30.0         cowplot_0.7.0         
[31] stringr_1.2.0          reshape2_1.4.2         rmarkdown_1.3         
[34] Citrus_0.99            easyGgplot2_1.0.0.9000 ggplot2_2.2.1         

loaded via a namespace (and not attached):
 [1] httr_1.2.1             splines_3.3.2          assertthat_0.1        
 [4] RBGL_1.50.0            yaml_2.1.14            RSQLite_1.1-2         
 [7] backports_1.0.5        lattice_0.20-34        digest_0.6.12         
[10] minqa_1.2.4            colorspace_1.3-2       htmltools_0.3.5       
[13] plyr_1.8.4             GSEABase_1.36.0        XML_3.98-1.5          
[16] devtools_1.12.0        xtable_1.8-2           git2r_0.18.0          
[19] tibble_1.2             annotate_1.52.1        withr_1.0.2           
[22] lazyeval_0.2.0         survival_2.40-1        magrittr_1.5          
[25] memoise_1.0.0          evaluate_0.10          MASS_7.3-45           
[28] tools_3.3.2            munsell_0.4.3          locfit_1.5-9.1        
[31] lambda.r_1.1.9         nloptr_1.0.4           RCurl_1.95-4.8        
[34] rstudioapi_0.6         AnnotationForge_1.16.1 bitops_1.0-6          
[37] labeling_0.3           gtable_0.2.0           DBI_0.5-1             
[40] curl_2.3               R6_2.2.0               knitr_1.15.1          
[43] rprojroot_1.2          futile.options_1.0.0   stringi_1.1.2         
[46] Rcpp_0.12.9           </code></pre>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
