<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />

<meta name="date" content="2017-03-02" />

<title>Explore biological significance of single cell only genes</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Explore biological significance of single cell only genes</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2017-03-02</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-03-06</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 4a0bfb9</p>
<div id="pull-in-genes-lists" class="section level3">
<h3>Pull in genes lists</h3>
<pre class="r"><code>library(&quot;biomaRt&quot;)
library(&quot;data.table&quot;)
library(&quot;testit&quot;)
library(&quot;dplyr&quot;)
library(&quot;edgeR&quot;)
library(&quot;VennDiagram&quot;)
library(&quot;ggplot2&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;../../singleCellSeq/analysis/functions.R&quot;)
require(Humanzee)
library(&quot;GOstats&quot;)</code></pre>
<pre class="r"><code>anno &lt;- read.table(&quot;../../singleCellSeq/data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
#input reads (pre-filter)
reads &lt;- read.table(&quot;../../singleCellSeq/data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
#input molecule (pre-filter)
molecules &lt;- read.table(&quot;../../singleCellSeq/data/molecules.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
#input bulk reads (pre-filter )
reads_bulk &lt;- read.table(&quot;../../singleCellSeq/data/reads-bulk.txt&quot;, header = TRUE, stringsAsFactors = FALSE)</code></pre>
<p>Create vectors with the list of genes in each set.</p>
<pre class="r"><code>genes_observed_reads_bulk &lt;- rownames(reads_bulk)[rowSums(reads_bulk) &gt; 0]
genes_observed_reads &lt;- rownames(reads)[rowSums(reads) &gt; 0]
genes_observed_molecules &lt;- rownames(molecules)[rowSums(molecules) &gt; 0]
genes_observed_molecules_ENSG &lt;- genes_observed_molecules[grep(&quot;ERCC&quot;, genes_observed_molecules, invert = TRUE)]
write.table(genes_observed_molecules_ENSG, file = &quot;../output/gene-observed-molecules-ensg.txt&quot;, quote = FALSE, row.names = FALSE)
single_cell_observed &lt;- union(genes_observed_molecules, genes_observed_molecules)</code></pre>
<div id="genes-only-in-single-cell-molecules" class="section level4">
<h4>Genes only in single cell molecules</h4>
<pre class="r"><code>gene_onlysinglecell &lt;- setdiff(single_cell_observed,genes_observed_reads_bulk)
#subset molecules by these genes  
molecules_df= data.frame(molecules)
molecules_only_SC &lt;- molecules_df[rownames(molecules_df) %in% gene_onlysinglecell,]</code></pre>
<p>Count number of cells we see these genes in</p>
<pre class="r"><code>non_zero_cells &lt;- apply(molecules_only_SC, 1, function(c)sum(c!=0))

molecules_with_nonzero &lt;- cbind(molecules_only_SC, non_zero_cells)

summary(molecules_with_nonzero$non_zero_cells)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   1.00    1.00    2.00    3.11    3.50  112.00 </code></pre>
<pre class="r"><code>genes_most_expressed= row.names(molecules_with_nonzero[molecules_with_nonzero$non_zero_cells &gt; 20, 1:3])
genes_most_expressed</code></pre>
<pre><code>[1] &quot;ENSG00000121318&quot; &quot;ENSG00000011465&quot; &quot;ENSG00000214324&quot; &quot;ENSG00000204296&quot;
[5] &quot;ENSG00000230062&quot; &quot;ENSG00000176571&quot; &quot;ERCC-00048&quot;      &quot;ERCC-00083&quot;     
[9] &quot;ERCC-00164&quot;     </code></pre>
<pre class="r"><code>ensembl &lt;- useMart(host = &quot;grch37.ensembl.org&quot;,
                   biomart = &quot;ENSEMBL_MART_ENSEMBL&quot;,
                   dataset = &quot;hsapiens_gene_ensembl&quot;)

genes_most_expressed_info &lt;- getBM(attributes = c(&quot;ensembl_gene_id&quot;, &quot;chromosome_name&quot;,
                                                 &quot;external_gene_name&quot;, &quot;transcript_count&quot;,&quot;description&quot;), filters = &quot;ensembl_gene_id&quot;, values = genes_most_expressed[grep(&quot;ENSG&quot;, genes_most_expressed)],mart = ensembl)
genes_most_expressed_info</code></pre>
<pre><code>  ensembl_gene_id chromosome_name external_gene_name transcript_count
1 ENSG00000011465              12                DCN               22
2 ENSG00000121318              12            TAS2R10                1
3 ENSG00000176571               8              CNBD1                6
4 ENSG00000204296               6            C6orf10                8
5 ENSG00000214324               3            C3orf56                1
6 ENSG00000230062               6            ANKRD66                3
                                                                   description
1                                        decorin [Source:HGNC Symbol;Acc:2705]
2             taste receptor, type 2, member 10 [Source:HGNC Symbol;Acc:14918]
3 cyclic nucleotide binding domain containing 1 [Source:HGNC Symbol;Acc:26663]
4            chromosome 6 open reading frame 10 [Source:HGNC Symbol;Acc:13922]
5            chromosome 3 open reading frame 56 [Source:HGNC Symbol;Acc:32481]
6                      ankyrin repeat domain 66 [Source:HGNC Symbol;Acc:44669]</code></pre>
<pre class="r"><code>genes_not_singleton= row.names(molecules_with_nonzero[molecules_with_nonzero$non_zero_cells &gt; 1, 1:3])</code></pre>
<pre class="r"><code>not_singleton_genes_info &lt;- getBM(attributes = c(&quot;ensembl_gene_id&quot;, &quot;chromosome_name&quot;,
                                                 &quot;external_gene_name&quot;, &quot;transcript_count&quot;,&quot;description&quot;), filters = &quot;ensembl_gene_id&quot;, values = genes_not_singleton[grep(&quot;ENSG&quot;, genes_not_singleton)],mart = ensembl)
head(not_singleton_genes_info)</code></pre>
<pre><code>  ensembl_gene_id chromosome_name external_gene_name transcript_count
1 ENSG00000005981               7               ASB4                3
2 ENSG00000007216              17            SLC13A2                8
3 ENSG00000007908               1               SELE               11
4 ENSG00000011465              12                DCN               22
5 ENSG00000012504              12              NR1H4                9
6 ENSG00000019169               2              MARCO                4
                                                                                                     description
1                                        ankyrin repeat and SOCS box containing 4 [Source:HGNC Symbol;Acc:16009]
2 solute carrier family 13 (sodium-dependent dicarboxylate transporter), member 2 [Source:HGNC Symbol;Acc:10917]
3                                                                      selectin E [Source:HGNC Symbol;Acc:10718]
4                                                                          decorin [Source:HGNC Symbol;Acc:2705]
5                                  nuclear receptor subfamily 1, group H, member 4 [Source:HGNC Symbol;Acc:7967]
6                                   macrophage receptor with collagenous structure [Source:HGNC Symbol;Acc:6895]</code></pre>
</div>
</div>
<div id="go-analysis" class="section level3">
<h3>GO analysis</h3>
<p><a href="https://github.com/jdblischak/singleCellSeq/blob/master/analysis/cv-adjusted-profile-annotation.Rmd" class="uri">https://github.com/jdblischak/singleCellSeq/blob/master/analysis/cv-adjusted-profile-annotation.Rmd</a><br />
<a href="https://github.com/jhsiao999/Humanzee/blob/master/R/GOtest.R" class="uri">https://github.com/jhsiao999/Humanzee/blob/master/R/GOtest.R</a></p>
<pre class="r"><code>not_singleton_genes_GO&lt;- GOtest(my_ensembl_gene_universe = rownames(molecules),
                      my_ensembl_gene_test =row.names(molecules_with_nonzero[molecules_with_nonzero$non_zero_cells &gt; 1, ]),
                      pval_cutoff = 1, ontology=c(&quot;BP&quot;) )


gene_names &lt;- row.names(molecules_with_nonzero[molecules_with_nonzero$non_zero_cells &gt; 1, ])
gene_names_ensg &lt;- gene_names[grep(&quot;ERCC&quot;, gene_names, invert = TRUE)]
write.table(gene_names_ensg, file = &quot;../output/gene-names.txt&quot;, quote = FALSE)
save(not_singleton_genes_GO,file=&quot;../data/not_singleton_genes_GO.rda&quot;)
save(gene_names_ensg, file=&quot;../data/gene_names_ensg.txt&quot;)</code></pre>
<pre class="r"><code>load(&quot;../data/gene_names_ensg.txt&quot;)
load(&quot;../data/not_singleton_genes_GO.rda&quot;)
goterms_SC &lt;- summary(not_singleton_genes_GO$GO$BP, pvalue = .05)
goterms_SC</code></pre>
<pre><code>        Length          Class           Mode 
             1 GOHyperGResult             S4 </code></pre>
<pre class="r"><code>#size in how many in universe
#count is how many in set  </code></pre>
<p>Online GO analysis: <a href="http://consensuspathdb.org" class="uri">http://consensuspathdb.org</a></p>
</div>
<div id="look-at-gene-symbols" class="section level3">
<h3>Look at gene symbols</h3>
<p>535 non singleton single cell only ENSG genes.</p>
<pre class="r"><code>gene_symbols &lt;- read.table(file = &quot;../../singleCellSeq/data/gene-info.txt&quot;, sep = &quot;\t&quot;,header = TRUE, stringsAsFactors = FALSE, quote = &quot;&quot;)</code></pre>
<pre class="r"><code>library(&quot;biomaRt&quot;)
ensembl &lt;- useMart(host = &quot;grch37.ensembl.org&quot;,
                     biomart = &quot;ENSEMBL_MART_ENSEMBL&quot;,
                     dataset = &quot;hsapiens_gene_ensembl&quot;)

gene_info_non_sing_ENSG &lt;- getBM(attributes = c(&quot;ensembl_gene_id&quot;, 
                               &quot;chromosome_name&quot;,
                               &quot;external_gene_name&quot;,
                               &quot;description&quot;),
                      filters = &quot;ensembl_gene_id&quot;,
                      values =gene_names_ensg,
                      mart = ensembl)</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] gridExtra_2.2.1        lme4_1.1-12            workflowr_0.4.0       
 [4] org.Hs.eg.db_3.4.0     GOstats_2.40.0         graph_1.52.0          
 [7] Category_2.40.0        Matrix_1.2-7.1         GO.db_3.4.0           
[10] AnnotationDbi_1.36.2   IRanges_2.8.1          S4Vectors_0.12.1      
[13] Biobase_2.34.0         BiocGenerics_0.20.0    Humanzee_0.1.0        
[16] sva_3.22.0             genefilter_1.56.0      mgcv_1.8-15           
[19] nlme_3.1-128           BiocInstaller_1.24.0   scales_0.4.1          
[22] VennDiagram_1.6.17     futile.logger_1.4.3    edgeR_3.16.5          
[25] limma_3.30.11          dplyr_0.5.0            testit_0.6            
[28] data.table_1.10.4      biomaRt_2.30.0         cowplot_0.7.0         
[31] stringr_1.2.0          reshape2_1.4.2         rmarkdown_1.3         
[34] Citrus_0.99            easyGgplot2_1.0.0.9000 ggplot2_2.2.1         

loaded via a namespace (and not attached):
 [1] httr_1.2.1             splines_3.3.2          assertthat_0.1        
 [4] RBGL_1.50.0            yaml_2.1.14            RSQLite_1.1-2         
 [7] backports_1.0.5        lattice_0.20-34        digest_0.6.12         
[10] minqa_1.2.4            colorspace_1.3-2       htmltools_0.3.5       
[13] plyr_1.8.4             GSEABase_1.36.0        XML_3.98-1.5          
[16] devtools_1.12.0        xtable_1.8-2           git2r_0.18.0          
[19] tibble_1.2             annotate_1.52.1        withr_1.0.2           
[22] lazyeval_0.2.0         survival_2.40-1        magrittr_1.5          
[25] memoise_1.0.0          evaluate_0.10          MASS_7.3-45           
[28] tools_3.3.2            munsell_0.4.3          locfit_1.5-9.1        
[31] lambda.r_1.1.9         nloptr_1.0.4           RCurl_1.95-4.8        
[34] rstudioapi_0.6         AnnotationForge_1.16.1 bitops_1.0-6          
[37] labeling_0.3           gtable_0.2.0           DBI_0.5-1             
[40] curl_2.3               R6_2.2.0               knitr_1.15.1          
[43] rprojroot_1.2          futile.options_1.0.0   stringi_1.1.2         
[46] Rcpp_0.12.9           </code></pre>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
