<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittlemant" />


<title>Gene x sample count matrix Bri data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Gene x sample count matrix Bri data</h1>
<h4 class="author"><em>Briana Mittlemant</em></h4>
<h4 class="date"><em>YYYY-MM-DD</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-03-06</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 4a0bfb9</p>
<pre class="r"><code>library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)
library(&quot;VennDiagram&quot;)</code></pre>
<p>This analysis will take the matrix from the fastq to counts anayisis pipeline (sample-by-gene) to a standard gene-x-sample matrix. I will also create annotation files and romove the genes with no observed sequencing data in the experiment. I will use the data from git hub rather than the data I created because I want to look at the whole experiment rather than one plate at this point.</p>
<div id="read-counts-for-single-cell-samples" class="section level3">
<h3>Read Counts for single cell samples:</h3>
<pre class="r"><code>reads_raw_bri &lt;- fread(&quot;../../single-cell/data/counts-matrix/reads-raw-single-per-sample.txt&quot;)
setDF(reads_raw_bri)</code></pre>
<p>Explore this data:</p>
<pre class="r"><code>dim(reads_raw_bri)</code></pre>
<pre><code>[1]    72 20422</code></pre>
<p>Ths has rows for each individule and the columns give information about the sample then give counts for each gene.</p>
<p>Create annotation file for single cell samples:</p>
<pre class="r"><code>anno_bri &lt;- reads_raw_bri %&gt;%
  dplyr::select(individual:well) %&gt;%
  mutate(batch = paste(individual, replicate, sep = &quot;.&quot;),
         sample_id = paste(batch, well, sep = &quot;.&quot;))
head(anno_bri)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A03 NA19098.r1 NA19098.r1.A03
4    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
5    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
6    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06</code></pre>
<pre class="r"><code>dim(anno_bri)</code></pre>
<pre><code>[1] 72  5</code></pre>
<p>This annoation file has information about the sample, replicatte, batch.</p>
<p>Transpose the matrix so that it is gene-x-sample.</p>
<pre class="r"><code>reads_bri &lt;- reads_raw_bri %&gt;%
  dplyr::select(starts_with(&quot;ENSG&quot;), starts_with(&quot;ERCC&quot;)) %&gt;%
  t
colnames(reads_bri) &lt;- anno_bri$sample_id
reads_bri[1:5, 1:5]</code></pre>
<pre><code>                NA19098.r1.A01 NA19098.r1.A02 NA19098.r1.A03
ENSG00000186092              0              0              0
ENSG00000237683              0              0              0
ENSG00000235249              0              0              0
ENSG00000185097              0              0              0
ENSG00000269831              0              0              0
                NA19098.r1.A04 NA19098.r1.A05
ENSG00000186092              0              0
ENSG00000237683             48              0
ENSG00000235249              0              0
ENSG00000185097              0              0
ENSG00000269831              0              0</code></pre>
</div>
<div id="molecule-counts-for-single-cell-samples" class="section level3">
<h3>Molecule Counts for single cell samples</h3>
<p>Import eaw molecules for single cell samples.</p>
<pre class="r"><code>molecules_raw_bri_lane &lt;- fread(&quot;../../single-cell/data/counts-matrix/molecules-raw-single-per-lane.txt&quot;)
molecules_raw_bri &lt;- fread(&quot;../../single-cell/data/counts-matrix/molecules-raw-single-per-sample.txt&quot;)
                       
setDF(molecules_raw_bri)
setDF(molecules_raw_bri_lane)
dim(molecules_raw_bri)</code></pre>
<pre><code>[1]    72 20422</code></pre>
<pre class="r"><code>dim(molecules_raw_bri_lane)</code></pre>
<pre><code>[1]   216 20425</code></pre>
<p>This data has rows by individuale and has gene columns like the reads data.</p>
<p>Conform single cell samples in reads and molecules files are in same order.</p>
<pre class="r"><code>stopifnot(anno_bri$individual == molecules_raw_bri$individual,
          anno_bri$replicate == molecules_raw_bri$replicate,
          anno_bri$well == molecules_raw_bri$well)</code></pre>
<p>Transpose the matrix so it is also gene-x-sample. Had to change code to dplyr::select because select is also a function in biomaRt.</p>
<pre class="r"><code>molecules_bri &lt;- molecules_raw_bri %&gt;%
  dplyr::select(starts_with(&quot;ENSG&quot;), starts_with(&quot;ERCC&quot;)) %&gt;%
  t
colnames(molecules_bri) &lt;- anno_bri$sample_id
molecules_bri[1:5, 1:5]</code></pre>
<pre><code>                NA19098.r1.A01 NA19098.r1.A02 NA19098.r1.A03
ENSG00000186092              0              0              0
ENSG00000237683              0              0              0
ENSG00000235249              0              0              0
ENSG00000185097              0              0              0
ENSG00000269831              0              0              0
                NA19098.r1.A04 NA19098.r1.A05
ENSG00000186092              0              0
ENSG00000237683              2              0
ENSG00000235249              0              0
ENSG00000185097              0              0
ENSG00000269831              0              0</code></pre>
</div>
<div id="read-counts-for-bulk-samples" class="section level3">
<h3>Read counts for bulk samples</h3>
<pre class="r"><code>reads_bulk_raw_bri &lt;- fread(&quot;../../single-cell/data/counts-matrix/reads-raw-bulk-per-sample.txt&quot;)
setDF(reads_bulk_raw_bri)</code></pre>
<p>Create annotation file for the bulk samples as we did for the single cells.</p>
<pre class="r"><code>anno_bulk_bri &lt;- reads_bulk_raw_bri %&gt;%
  dplyr::select(individual:well) %&gt;%
  mutate(batch = paste(individual, replicate, sep = &quot;.&quot;),
         sample_id = paste(batch, well, sep = &quot;.&quot;))
head(anno_bulk_bri)</code></pre>
<pre><code>  individual replicate well      batch       sample_id
1    NA19098        r1 bulk NA19098.r1 NA19098.r1.bulk</code></pre>
<p>Transpose the bulk read information to match the gene-x-sample format.</p>
<pre class="r"><code>reads_bulk_bri &lt;- reads_bulk_raw_bri %&gt;%
  dplyr::select(starts_with(&quot;ENSG&quot;), starts_with(&quot;ERCC&quot;)) %&gt;%
  t
colnames(reads_bulk_bri) &lt;- anno_bulk_bri$sample_id

#pull in github data to compare
reads_bulk_raw &lt;- fread(&quot;../../singleCellSeq/data/reads-raw-bulk-per-sample.txt&quot;)
setDF(reads_bulk_raw) #converts the data.table to a data.frame
anno_bulk &lt;- reads_bulk_raw %&gt;%
  dplyr::select(individual:well) %&gt;%
  mutate(batch = paste(individual, replicate, sep = &quot;.&quot;),
         sample_id = paste(batch, well, sep = &quot;.&quot;))
reads_bulk &lt;- reads_bulk_raw %&gt;%
  dplyr::select(starts_with(&quot;ENSG&quot;), starts_with(&quot;ERCC&quot;)) %&gt;%
  t
colnames(reads_bulk) &lt;- anno_bulk$sample_id



reads_bulk_bri[1:15, 0:1]</code></pre>
<pre><code>ENSG00000186092 ENSG00000237683 ENSG00000235249 ENSG00000185097 
              0              65               0               0 
ENSG00000269831 ENSG00000269308 ENSG00000187634 ENSG00000268179 
              0               0               6               0 
ENSG00000188976 ENSG00000187961 ENSG00000187583 ENSG00000187642 
           4214             245              13               1 
ENSG00000188290 ENSG00000187608 ENSG00000188157 
              0            1656            1301 </code></pre>
<pre class="r"><code>reads_bulk[1:15, 0:1]</code></pre>
<pre><code>ENSG00000186092 ENSG00000237683 ENSG00000235249 ENSG00000185097 
              0              50               0               0 
ENSG00000269831 ENSG00000269308 ENSG00000187634 ENSG00000268179 
              0               0               2               0 
ENSG00000188976 ENSG00000187961 ENSG00000187583 ENSG00000187642 
           3575              54               8               2 
ENSG00000188290 ENSG00000187608 ENSG00000188157 
              0            1525             670 </code></pre>
<p>I show more reads for the bulk samples as well.</p>
</div>
<div id="observed-genes" class="section level3">
<h3>Observed genes:</h3>
<pre class="r"><code>stopifnot(rownames(reads_bulk_bri) == rownames(reads_bri),
          rownames(reads_bri) == rownames(molecules_bri))

genes_observed_reads_bulk_bri &lt;- rownames(reads_bulk_bri)[rowSums(reads_bulk_bri) &gt; 0]
genes_observed_reads_bri &lt;- rownames(reads_bri)[rowSums(reads_bri) &gt; 0]
genes_observed_molecules_bri &lt;- rownames(molecules_bri)[rowSums(molecules_bri) &gt; 0]
stopifnot(genes_observed_molecules_bri %in% genes_observed_reads_bri)</code></pre>
<pre class="r"><code>genes_venn_bri &lt;- venn.diagram(x = list(&quot;reads&quot; = genes_observed_reads_bri,
                                    &quot;molecules&quot; = genes_observed_molecules_bri,
                                    &quot;reads bulk&quot; = genes_observed_reads_bulk_bri),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(genes_venn_bri)</code></pre>
<p><img src="figure/bri_gene_x_sample_count_matrix.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /> This ven diagram shows the intesection of the genes found in each type of sequencing data. I will now remove all genes with no observed read in either bulk or single cell sample.</p>
<pre class="r"><code>genes_observed_bri &lt;- union(genes_observed_reads_bri, genes_observed_reads_bulk_bri)
reads_bri &lt;- reads_bri[rownames(reads_bri) %in% genes_observed_bri, ]
molecules_bri &lt;- molecules_bri[rownames(molecules_bri) %in% genes_observed_bri, ]
reads_bulk_bri &lt;- reads_bulk_bri[rownames(reads_bulk_bri) %in% genes_observed_bri, ]</code></pre>
<p>There are 17128 genes.</p>
</div>
<div id="write-files" class="section level3">
<h3>Write files:</h3>
<p>Write the files to my cell_cycle git hub.<br />
Annotations:</p>
<pre class="r"><code>write.table(anno_bri, &quot;../data/annotation_bri.txt&quot;, quote=FALSE, sep=&quot;\t&quot;, row.names = FALSE)
write.table(anno_bulk, &quot;../data/annotation-bulk_bri.txt&quot;, quote=FALSE, sep = &quot;\t&quot;, row.names=FALSE)</code></pre>
<p>Reads counts:</p>
<pre class="r"><code>write.table(reads_bri, &quot;../data/reads_bri.txt&quot;, quote = FALSE, sep = &quot;\t&quot;, col.names = NA)
write.table(reads_bulk, &quot;../data/reads-bulk_bri.txt&quot;, quote=FALSE, sep=&quot;\t&quot;, col.names = NA)</code></pre>
<p>Output molecule counts.</p>
<pre class="r"><code>write.table(molecules_bri, &quot;../data/molecules_bri.txt&quot;, quote= FALSE, sep= &quot;\t&quot;, col.names = NA)</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] gridExtra_2.2.1        lme4_1.1-12            workflowr_0.4.0       
 [4] org.Hs.eg.db_3.4.0     GOstats_2.40.0         graph_1.52.0          
 [7] Category_2.40.0        Matrix_1.2-7.1         GO.db_3.4.0           
[10] AnnotationDbi_1.36.2   IRanges_2.8.1          S4Vectors_0.12.1      
[13] Biobase_2.34.0         BiocGenerics_0.20.0    Humanzee_0.1.0        
[16] sva_3.22.0             genefilter_1.56.0      mgcv_1.8-15           
[19] nlme_3.1-128           BiocInstaller_1.24.0   scales_0.4.1          
[22] VennDiagram_1.6.17     futile.logger_1.4.3    edgeR_3.16.5          
[25] limma_3.30.11          dplyr_0.5.0            testit_0.6            
[28] data.table_1.10.4      biomaRt_2.30.0         cowplot_0.7.0         
[31] stringr_1.2.0          reshape2_1.4.2         rmarkdown_1.3         
[34] Citrus_0.99            easyGgplot2_1.0.0.9000 ggplot2_2.2.1         

loaded via a namespace (and not attached):
 [1] httr_1.2.1             splines_3.3.2          assertthat_0.1        
 [4] RBGL_1.50.0            yaml_2.1.14            RSQLite_1.1-2         
 [7] backports_1.0.5        lattice_0.20-34        digest_0.6.12         
[10] minqa_1.2.4            colorspace_1.3-2       htmltools_0.3.5       
[13] plyr_1.8.4             GSEABase_1.36.0        XML_3.98-1.5          
[16] devtools_1.12.0        xtable_1.8-2           git2r_0.18.0          
[19] tibble_1.2             annotate_1.52.1        withr_1.0.2           
[22] lazyeval_0.2.0         survival_2.40-1        magrittr_1.5          
[25] memoise_1.0.0          evaluate_0.10          MASS_7.3-45           
[28] tools_3.3.2            munsell_0.4.3          locfit_1.5-9.1        
[31] lambda.r_1.1.9         nloptr_1.0.4           RCurl_1.95-4.8        
[34] rstudioapi_0.6         AnnotationForge_1.16.1 bitops_1.0-6          
[37] labeling_0.3           gtable_0.2.0           DBI_0.5-1             
[40] curl_2.3               R6_2.2.0               knitr_1.15.1          
[43] rprojroot_1.2          futile.options_1.0.0   stringi_1.1.2         
[46] Rcpp_0.12.9           </code></pre>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
