<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>Cell to cell variation analaysis: paper figures</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Cell to cell variation analaysis: paper figures</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2017-2-17</em></h4>

</div>


<p><strong>Last updated:</strong> 2017-02-24</p>
<p><strong>Code version:</strong> 0506d1c</p>
<div id="objective" class="section level3">
<h3>Objective</h3>
<p>This page contains code for reproducing figures reported in the paper that are related to the analysis of cell-to-cell variaiton in transcriptional gene expression.</p>
<p>Coefficient of variation: standardized measure of dispersion of a probability distribution, ratio of <span class="math inline">\(\sigma\)</span> to <span class="math inline">\(\mu\)</span>.</p>
</div>
<div id="setup" class="section level3">
<h3>Setup</h3>
<pre class="r"><code>library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)
library(&quot;limma&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;grid&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;../../singleCellSeq/analysis/functions.R&quot;)
library(&quot;Humanzee&quot;)
library(&quot;cowplot&quot;)
library(&quot;MASS&quot;)
library(&quot;matrixStats&quot;)
source(&quot;../../singleCellSeq/code/plotting-functions.R&quot;)</code></pre>
<p>Import molecule counts before standardization and transformation and also log2-transformed counts after batch correction. Biologial variation analtsis of the individuals is performed on the batch-corrected and log2-transformed counts.</p>
<pre class="r"><code>#filtered annotation
anno_filter &lt;- read.table(&quot;../data/annotation-filter.txt&quot;, header = TRUE, stringsAsFactors = FALSE)
#filtered molecule counts  
molecules_filter &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header=TRUE, stringsAsFactors = FALSE)
stopifnot(NROW(anno_filter) == NCOL(molecules_filter))
#final processed molecule counts for endogenous genes   
molecules_final&lt;- read.table(&quot;../data/molecules-final.txt&quot;, header=TRUE, stringsAsFactors = FALSE)
stopifnot(NROW(anno_filter) == NCOL(molecules_final))
#import gene symbols
gene_symbols &lt;- read.table(file = &quot;../../singleCellSeq//data/gene-info.txt&quot;, sep = &quot;\t&quot;,
                           header = TRUE, stringsAsFactors = FALSE, quote = &quot;&quot;)
#import cell-cycle gene list  
cell_cycle_genes &lt;- read.table(&quot;../../singleCellSeq/data/cellcyclegenes.txt&quot;,
                               header = TRUE, sep = &quot;\t&quot;,
                               stringsAsFactors = FALSE)
#import pluripotency gene list (IPSC check)
pluripotency_genes &lt;- read.table(&quot;../../singleCellSeq/data/pluripotency-genes.txt&quot;,
                               header = TRUE, sep = &quot;\t&quot;,
                               stringsAsFactors = FALSE)$To</code></pre>
<p>Load gene CVs computed over all cells and CV computed over the expressed cells (should go back and create this files on my own)</p>
<pre class="r"><code>load(&quot;../../singleCellSeq/data/cv-all-cells.rda&quot;)
load(&quot;../../singleCellSeq/data/cv-expressed-cells.rda&quot;)
##permutation results inclusing the detected/expressed cells
load(&quot;../../singleCellSeq/data/permuted-pval-expressed-set1.rda&quot;)
##load observed statistics (MAD)
load(&quot;../../singleCellSeq/data/mad-expressed.rda&quot;)</code></pre>
<p>Compute a matrix of 0’s and 1’s labeling non-detected and detected cells, respctively.</p>
<pre class="r"><code>molecules_expressed&lt;- molecules_filter 
molecules_expressed[which(molecules_filter &gt;0, arr.ind=TRUE)] &lt;-1
molecules_expressed &lt;- as.matrix((molecules_expressed))</code></pre>
<p>Take the genes subset inclded in final data.</p>
<pre class="r"><code>genes_included &lt;- Reduce(intersect, 
                         list(rownames(molecules_final), 
                              rownames(expressed_cv$NA19098), 
                              names(perm_pval_set1)) )

molecules_filter_subset &lt;- molecules_filter[
  which(rownames(molecules_filter) %in% genes_included), ]

molecules_final_subset &lt;- molecules_final[
  which(rownames(molecules_final) %in% genes_included), ]

molecules_expressed_subset &lt;- molecules_expressed[
  which(rownames(molecules_expressed) %in% genes_included), ]

molecules_final_expressed_subset &lt;- molecules_final_subset*molecules_expressed_subset
molecules_final_expressed_subset[which(molecules_expressed_subset == 0, 
                                       arr.ind = TRUE)] &lt;- NA

permuted_pval_subset &lt;- perm_pval_set1[which(names(perm_pval_set1) %in% genes_included)]

expressed_cv_subset &lt;- lapply(expressed_cv, function(x)
  x[which(rownames(x) %in% genes_included), ] )
names(expressed_cv_subset) &lt;- names(expressed_cv)

expressed_dm_subset &lt;- expressed_dm[which(rownames(expressed_dm) %in% genes_included), , ] 

dim(molecules_final_subset)</code></pre>
<pre><code>[1] 13043   564</code></pre>
<pre class="r"><code>dim(molecules_expressed_subset)</code></pre>
<pre><code>[1] 13043   564</code></pre>
<pre class="r"><code>all.equal(rownames(expressed_cv_subset$NA19098), 
          rownames(molecules_final_expressed_subset) )</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>all.equal(names(permuted_pval_subset), 
          rownames(molecules_expressed_subset) )</code></pre>
<pre><code>[1] TRUE</code></pre>
<p>Compute drop-out rates</p>
<ul>
<li>dropout : lack of molecule representation for expressed gene (look at proportion of cells a given gene is undetected, the average gene expression, and estimates of gene expression noise)</li>
</ul>
<pre class="r"><code>drop_out &lt;- lapply(unique(anno_filter$individual), function(ind) {
  temp_df &lt;- molecules_filter_subset[,anno_filter$individual == ind]
  zero_count &lt;- rowMeans(temp_df == 0)
  return(zero_count)
})  
names(drop_out) &lt;- unique(anno_filter$individual)
drop_out$all &lt;- rowMeans(as.matrix(molecules_filter_subset) == 0)


summary(drop_out$NA19098)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.00000 0.02113 0.21130 0.32060 0.59860 0.99300 </code></pre>
<pre class="r"><code>summary(drop_out$NA19101)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.00000 0.04975 0.29350 0.36500 0.66170 0.99500 </code></pre>
<pre class="r"><code>summary(drop_out$NA19239)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.00000 0.02715 0.25340 0.35060 0.65610 0.99550 </code></pre>
<pre class="r"><code>summary(drop_out$all)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.00000 0.03723 0.26060 0.34820 0.64540 0.93090 </code></pre>
<p>Statistical test for individual difference in dropout rate: logistic regression</p>
<ul>
<li>logistic regression- dependent variable is a categorical variable</li>
</ul>
<pre class="r"><code>if(!file.exists(&quot;../../singleCellSeq//data/sig-dropout.rda&quot;)) {
library(lmtest)
  drop_out_significance &lt;- sapply(1:NROW(molecules_expressed_subset), function(i) {
  fit &lt;- glm(molecules_expressed_subset[i,] ~ as.factor(anno_filter$individual),
             family = &quot;binomial&quot;)
  lrtest(fit)$P[2]  
})
  save(drop_out_significance, file = &quot;../data/sig-dropout.rda&quot;)
} else {
  load(&quot;../../singleCellSeq/data/sig-dropout.rda&quot;)
}
quantile(drop_out_significance, prob = seq(0,1,.1))[2] &lt; 1e-5</code></pre>
<pre><code> 10% 
TRUE </code></pre>
<pre class="r"><code>sum(drop_out_significance &lt; quantile(drop_out_significance, prob = seq(0,1,.1))[2])</code></pre>
<pre><code>[1] 1214</code></pre>
</div>
<div id="main-figure-in-paper-5-cell-to-cell-variation-in-gene-expression" class="section level3">
<h3>Main figure in paper 5: Cell to cell variation in gene expression</h3>
<div id="expressed-cells" class="section level4">
<h4>Expressed Cells</h4>
<pre class="r"><code># mean gene expression
genes &lt;- rownames(expressed_cv_subset[[1]])
venn_mean_rank &lt;- gplots::venn( 
  list(NA19098 = genes[ which(rank(expressed_cv_subset[[1]]$expr_mean) &gt; length(genes) - 1000 ) ],
       NA19101 = genes[ which(rank(expressed_cv_subset[[2]]$expr_mean) &gt; length(genes) - 1000 ) ],
       NA19239 = genes[ which(rank(expressed_cv_subset[[3]]$expr_mean) &gt; length(genes) - 1000 ) ] ) )</code></pre>
<p><img src="figure/cell_to_cell_variation.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># adjusted CV
venn_cv_rank &lt;- gplots::venn( 
  list(NA19098 = genes[ which(rank(expressed_dm_subset[[1]]) &gt; length(genes) - 1000 ) ],
       NA19101 = genes[ which(rank(expressed_dm_subset[[2]]) &gt; length(genes) - 1000 ) ],
       NA19239 = genes[ which(rank(expressed_dm_subset[[3]]) &gt; length(genes) - 1000 ) ] ) )</code></pre>
<p><img src="figure/cell_to_cell_variation.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /> ####All cells</p>
<p>Venn diagram of gene-specific mean and adjusted CV for when all cells are included. For more results for when including all cells.</p>
<pre class="r"><code>load(&quot;../../singleCellSeq/data/cv-all-cells.rda&quot;)
genes &lt;- rownames(ENSG_cv[[1]])
library(gplots)
venn_cv_rank &lt;- gplots::venn(
  list(NA19098 = genes[ which( rank(ENSG_cv_adj$NA19098$log10cv2_adj) 
                               &gt; length(genes) - 1000 ) ],
       NA19101 = genes[ which( rank(ENSG_cv_adj$NA19101$log10cv2_adj) 
                               &gt; length(genes) - 1000 ) ],
       NA19239 = genes[ which( rank(ENSG_cv_adj$NA19239$log10cv2_adj) 
                               &gt; length(genes) - 1000 ) ] ))</code></pre>
<p><img src="figure/cell_to_cell_variation.Rmd/unnamed-chunk-38-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gplots::venn( 
  list(NA19098 = genes[ which(rank(ENSG_cv[[1]]$mean) &gt; length(genes) - 1000 ) ],
       NA19101 = genes[ which(rank(ENSG_cv[[2]]$mean) &gt; length(genes) - 1000 ) ],
       NA19239 = genes[ which(rank(ENSG_cv[[3]]$mean) &gt; length(genes) - 1000 ) ] ) )</code></pre>
<p><img src="figure/cell_to_cell_variation.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="mean-cv-plot" class="section level4">
<h4>Mean CV plot</h4>
<p>CV and mean of batch-corrected data including all cells across all three indivivduals.</p>
<pre class="r"><code>par(mfrow = c(1,2))
all_cv &lt;- matrixStats::rowSds(2^molecules_final_subset)/rowMeans(2^molecules_final_subset)
all_mn &lt;- rowMeans(2^molecules_final_subset)
plot(y = all_cv,
     x = log2(all_mn),
     cex = .9, col = alpha(&quot;grey40&quot;, .8), lwd = .6,
     xlab = &quot;log2 gene mean abundance&quot;,
     ylab = &quot;Coefficient of variation&quot;,
     main = &quot;All cells, all indviduals&quot;,
     xlim = c(5,17), ylim = c(0,4.5))
points(y = all_cv,
       x = log2(all_mn), 
       col = rev(RColorBrewer::brewer.pal(10, &quot;RdYlBu&quot;))[
                cut(drop_out$all, breaks = seq(0, 1, by = .1),
                     include.lowest = TRUE)], 
       cex = .6, pch = 16)

# cv and mean of batch-corrected data including only expressed cells
# across all three indivivduals
plot(y = expressed_cv_subset$all$expr_cv,
     x = log2(expressed_cv_subset$all$expr_mean),
     cex = .9, col = alpha(&quot;grey40&quot;, .8), lwd = .6,
     xlab = &quot;log2 gene mean abundance&quot;,
     ylab = &quot;Coefficient of variation&quot;,
     main = &quot;Expressed cells, all individuals&quot;,
     xlim = c(5,17), ylim = c(0,4.5))
points(y = expressed_cv_subset$all$expr_cv,
       x = log2(expressed_cv_subset$all$expr_mean), 
       col = rev(RColorBrewer::brewer.pal(10, &quot;RdYlBu&quot;))[
                cut(drop_out$all, breaks = seq(0, 1, by = .1),
                     include.lowest = TRUE)], 
       cex = .6, pch = 16)</code></pre>
<p><img src="figure/cell_to_cell_variation.Rmd/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># make labels
plot(x = rep(1,10), y = 1:10, 
     pch = 15, cex = 2, axes = F, xlab = &quot;&quot;, ylab = &quot;&quot;,
     col = (RColorBrewer::brewer.pal(10, &quot;RdYlBu&quot;)), xlim = c(0, 13))
text(labels = rev( levels(cut(drop_out$NA19098, breaks = seq(0, 1, by = .1),
                     include.lowest = TRUE))) ,
     y = 1:10, x = 3, cex = .7, adj = 0)
title(main = &quot;drop-out rate&quot;)</code></pre>
<p><img src="figure/cell_to_cell_variation.Rmd/unnamed-chunk-41-1.png" width="672" style="display: block; margin: auto;" /> ####Additional analysis</p>
<p>Investigate the consistency of genes in the top ranked highly expressed list between the analysis including only the expressed cells and the analysis including all of the cells.</p>
<p><strong>Mean gene expression</strong></p>
<pre class="r"><code>expressed_cells_genes &lt;- rownames(expressed_cv_subset[[1]])
high_mean_genes_expressed_cells &lt;- venn_mean_rank &lt;- gplots::venn( 
  list(NA19098 = expressed_cells_genes[ 
          which(rank(expressed_cv_subset[[1]]$expr_mean) &gt; length(expressed_cells_genes) - 1000 ) ],
       NA19101 = expressed_cells_genes[ 
          which(rank(expressed_cv_subset[[2]]$expr_mean) &gt; length(expressed_cells_genes) - 1000 ) ],
       NA19239 = expressed_cells_genes[ 
         which(rank(expressed_cv_subset[[3]]$expr_mean) &gt; length(expressed_cells_genes) - 1000 ) ]),
  show.plot = FALSE)

all_cells_genes &lt;- rownames(ENSG_cv[[1]])
high_mean_genes_all_cells &lt;- gplots::venn( 
  list(NA19098 = all_cells_genes[ which(rank(ENSG_cv[[1]]$mean) &gt; length(all_cells_genes) - 1000 ) ],
       NA19101 = all_cells_genes[ which(rank(ENSG_cv[[2]]$mean) &gt; length(all_cells_genes) - 1000 ) ],
       NA19239 = all_cells_genes[ which(rank(ENSG_cv[[3]]$mean) &gt; length(all_cells_genes) - 1000 ) ] ),
  show.plot = FALSE)</code></pre>
<p>Check if the genes in the intersection set are the same: Yes, exactly the same.</p>
<pre class="r"><code># setdiff(y, x) finds the elements in y/x
# that is, elements in y but not in x
setequal(attr(high_mean_genes_expressed_cells, &quot;intersection&quot;)[[&quot;NA19098:NA19101:NA19239&quot;]],
           attr(high_mean_genes_all_cells, &quot;intersection&quot;)[[&quot;NA19098:NA19101:NA19239&quot;]]) </code></pre>
<pre><code>[1] TRUE</code></pre>
<p>adjusted CV</p>
<pre class="r"><code>expressed_cells_genes &lt;- rownames(expressed_cv_subset[[1]])
high_adjcv_genes_expressed_cells &lt;- gplots::venn( 
  list(NA19098 = expressed_cells_genes[ 
          which(rank(expressed_dm_subset[[1]]) &gt; length(expressed_cells_genes) - 1000 ) ],
       NA19101 = expressed_cells_genes[ 
          which(rank(expressed_dm_subset[[2]]) &gt; length(expressed_cells_genes) - 1000 ) ],
       NA19239 = expressed_cells_genes[ 
         which(rank(expressed_dm_subset[[3]]) &gt; length(expressed_cells_genes) - 1000 ) ]),
  show.plot = FALSE)

all_cells_genes &lt;- rownames(ENSG_cv[[1]])
high_adjcv_genes_all_cells &lt;- gplots::venn(
  list(NA19098 = all_cells_genes[ which( rank(ENSG_cv_adj$NA19098$log10cv2_adj) 
                               &gt; length(all_cells_genes) - 1000 ) ],
       NA19101 = all_cells_genes[ which( rank(ENSG_cv_adj$NA19101$log10cv2_adj) 
                               &gt; length(all_cells_genes) - 1000 ) ],
       NA19239 = all_cells_genes[ which( rank(ENSG_cv_adj$NA19239$log10cv2_adj) 
                               &gt; length(all_cells_genes) - 1000 ) ] ),
  show.plot = FALSE)</code></pre>
<p>Check if the genes in the intersection set are the same: NO!</p>
<pre class="r"><code>high_adjcv_genes_expressed_all &lt;- 
  gplots::venn(list(expressed = attr(high_adjcv_genes_expressed_cells, &quot;intersection&quot;)[[&quot;NA19098:NA19101:NA19239&quot;]],
        all = attr(high_adjcv_genes_all_cells, &quot;intersection&quot;)[[&quot;NA19098:NA19101:NA19239&quot;]]))</code></pre>
<p><img src="figure/cell_to_cell_variation.Rmd/unnamed-chunk-45-1.png" width="672" style="display: block; margin: auto;" /> Of the genes with high CV in the expressed cell analysis but not in all cell analysis, the gene expression distribution including the expressed cells versus the expression distribution including all the cells.</p>
<pre class="r"><code>high_adjcv_genes_expressed_but_not_all &lt;- attr(high_adjcv_genes_expressed_all, &quot;intersection&quot;)$expressed
par(mfrow = c(4,4))
for (i in 1:length(high_adjcv_genes_expressed_but_not_all)) {
plot_density_overlay(
    molecules = molecules_final_expressed_subset,
    annotation = anno_filter,
    which_gene = high_adjcv_genes_expressed_but_not_all[i], 
    labels = &quot;&quot;,
    xlims = c(4,12.5), ylims = c(0,1),
    cex.lab = 1.2,
    cex.axis = 1.2,
    gene_symbols = gene_symbols)
    individuals &lt;- unique(anno_filter$individual)
  library(scales)
  library(broman)
  crayon &lt;- brocolors(&quot;crayon&quot;)
  cols &lt;- c(&quot;Mango Tango&quot;, &quot;Green&quot;, &quot;Violet Blue&quot;)
  cols &lt;- alpha(crayon[cols], .7)

plot_density_overlay(
    molecules = molecules_final_subset,
    annotation = anno_filter,
    which_gene = high_adjcv_genes_expressed_but_not_all[i], 
    labels = &quot;&quot;,
    xlims = c(0,15), ylims = c(0,1.5),
    cex.lab = 1.2,
    cex.axis = 1.2,
    gene_symbols = gene_symbols)
    individuals &lt;- unique(anno_filter$individual)
  library(scales)
  library(broman)
  crayon &lt;- brocolors(&quot;crayon&quot;)
  cols &lt;- c(&quot;Mango Tango&quot;, &quot;Green&quot;, &quot;Violet Blue&quot;)
  cols &lt;- alpha(crayon[cols], .7)
}  </code></pre>
<p><img src="figure/cell_to_cell_variation.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/cell_to_cell_variation.Rmd/unnamed-chunk-46-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/cell_to_cell_variation.Rmd/unnamed-chunk-46-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Of the genes detected as high CV in all cell analysis but not in the expressed cells, make the distribution plot including only the expressed cells and including all the cells. Only print selected examples. Basically there is no individual difference in adjusted coefficent of variation after excluding the non-detected cells.</p>
<pre class="r"><code>high_adjcv_genes_all_but_not_expressed &lt;- attr(high_adjcv_genes_expressed_all, &quot;intersection&quot;)$all
par(mfrow = c(4,4))
#for (i in 1:length(high_adjcv_genes_all_but_not_expressed)) {
for (i in 1:8) {
plot_density_overlay(
    molecules = molecules_final_expressed_subset,
    annotation = anno_filter,
    which_gene = high_adjcv_genes_all_but_not_expressed[i], 
    labels = &quot;&quot;,
    xlims = c(4,12.5), ylims = c(0,1),
    cex.lab = 1.2,
    cex.axis = 1.2,
    gene_symbols = gene_symbols)
    individuals &lt;- unique(anno_filter$individual)
  library(scales)
  library(broman)
  crayon &lt;- brocolors(&quot;crayon&quot;)
  cols &lt;- c(&quot;Mango Tango&quot;, &quot;Green&quot;, &quot;Violet Blue&quot;)
  cols &lt;- alpha(crayon[cols], .7)

plot_density_overlay(
    molecules = molecules_final_subset,
    annotation = anno_filter,
    which_gene = high_adjcv_genes_expressed_but_not_all[i], 
    labels = &quot;&quot;,
    xlims = c(0,15), ylims = c(0,1.5),
    cex.lab = 1.2,
    cex.axis = 1.2,
    gene_symbols = gene_symbols)
    individuals &lt;- unique(anno_filter$individual)
  library(scales)
  library(broman)
  crayon &lt;- brocolors(&quot;crayon&quot;)
  cols &lt;- c(&quot;Mango Tango&quot;, &quot;Green&quot;, &quot;Violet Blue&quot;)
  cols &lt;- alpha(crayon[cols], .7)
}  </code></pre>
<p><img src="figure/cell_to_cell_variation.Rmd/unnamed-chunk-47-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="supplemental-figure-6-inter-individual-differences-in-regulatory-noise" class="section level3">
<h3>Supplemental Figure 6: Inter-individual differences in regulatory noise</h3>
<pre class="r"><code>#NANOG: a key pluripotency gene.
#All the other genes had significant difference between adjusted CV of the expressed cells.
query_genes &lt;- c(&quot;NANOG&quot;, &quot;TFPI&quot;, &quot;SLC9A5&quot;, &quot;SLC25A16&quot;, &quot;SREK1IP1&quot;)
query_genes_ensg &lt;- gene_symbols[which(gene_symbols$external_gene_name %in% query_genes), ]

## print permuted p-values of the chosen genes
pvals_tab &lt;- rbind(names(perm_pval_set1)[which(names(perm_pval_set1) %in% query_genes_ensg$ensembl_gene_id)],
      gene_symbols$external_gene_name[gene_symbols$ensembl_gene_id %in% names(perm_pval_set1)[which(names(perm_pval_set1) %in% query_genes_ensg$ensembl_gene_id)]],
      perm_pval_set1[which(names(perm_pval_set1) %in% query_genes_ensg$ensembl_gene_id)] )
      
pdf(&quot;../../singleCellSeq/analysis/figure/cv-adjusted-summary-pois-final.Rmd/chosen-genes.pdf&quot;,
    height = 8, width = 12)
#### specify the order of the plot, making same gene appear in the same column
layout( matrix(c(2, 5, 8, 11, 14, 
                 1, 4, 7, 10, 13,
                 3, 6, 9, 12, 15
                 ), nrow = 3, byrow = TRUE))
for (i in 1:nrow(query_genes_ensg)) {

  df &lt;- do.call(cbind,
    lapply(unique(anno_filter$individual), function(ind) {
    table(molecules_filter_subset[which(rownames(molecules_final_expressed_subset) %in% query_genes_ensg$ensembl_gene[i]), anno_filter$individual == ind] == 0)
    }) )
  colnames(df) &lt;- unique(anno_filter$individual)
  df &lt;- t(t(df)/colSums(df))
  barplot(df, col = c(&quot;grey40&quot;, &quot;grey80&quot;),
          ylab = &quot;Proportion of cells&quot;,
          cex.names = 0.9,
          cex.axis = 1.2, 
          cex.lab = 1.2)

  plot_density_overlay(
    molecules = molecules_final_expressed_subset,
    annotation = anno_filter,
    which_gene = query_genes_ensg$ensembl_gene_id[i], 
    labels = &quot;&quot;,
    xlims = c(4,12.5), ylims = c(0,1),
    cex.lab = 1.2,
    cex.axis = 1.2,
    gene_symbols = gene_symbols)
    individuals &lt;- unique(anno_filter$individual)
  library(scales)
  library(broman)
  crayon &lt;- brocolors(&quot;crayon&quot;)
  cols &lt;- c(&quot;Mango Tango&quot;, &quot;Green&quot;, &quot;Violet Blue&quot;)
  cols &lt;- alpha(crayon[cols], .7)

  barplot(height = abs(unlist(expressed_dm_subset[which(rownames(expressed_dm_subset) %in% query_genes_ensg$ensembl_gene_id[i]),])), col = cols,
          ylim = c(0,.7), border = &quot;white&quot;,
          ylab = &quot;Absolute value of adjusted CV&quot;,
          cex.names = 0.9,
          cex.axis = 1.2, 
          cex.lab = 1.2)
  abline(h = 0)

  # print MAD values  
  mad_val &lt;- sprintf(&quot;%.2f&quot;,round(drop_out$all[which(rownames(molecules_final_subset) %in% query_genes_ensg$ensembl_gene_id[i])], digits = 2))
  if (as.numeric(mad_val) &lt; .01) {
  mtext(text = paste0(&quot;MAD&lt;.01&quot;),
        side=3, line = 2)
  } else {
  mtext(text = paste0(&quot;MAD=&quot;, mad_val),
        side=3, line = 2)
  }
  
  # print permuted p-value for the adjusted coefficient of variation
  if (i !=2) {
  mtext(text = expression(paste(italic(&quot;p&quot;),&quot;&lt;&quot;,10^&quot;-4&quot;)), side=3)
  }
  if (i == 2) {
  mtext(text = expression(paste(italic(&quot;p&quot;),&quot;=.40&quot;)), side = 3)
  }
abline(h = 0)

}
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
</div>
<div id="suppelmental-figure-7-cell-to-cell-variation-of-pluripotent-genes" class="section level3">
<h3>Suppelmental Figure 7: Cell-to-cell variation of pluripotent genes</h3>
<pre class="r"><code>## plot out pluripotent genes over all cells and over only the expressed cells

query_genes &lt;- pluripotency_genes
query_genes_ensg &lt;- query_genes[which(query_genes %in% rownames(molecules_final_subset))]

pdf(&quot;../../singleCellSeq/analysis/figure/cv-adjusted-summary-pois-final.Rmd/pluri-dens.pdf&quot;,
    height = 10, width = 8)
par(mfrow = c(5,4))
for (i in 1:length(query_genes_ensg)) {
  plot_density_overlay(
    molecules = molecules_final_expressed_subset,
    annotation = anno_filter,
    which_gene = query_genes_ensg[i], 
#    labels = round(drop_out$all[which(rownames(molecules_final_subset) %in% query_genes_ensg[i])], digits = 2),
    labels = &quot;&quot;,
    xlims = c(0,15), ylims = NULL,
    gene_symbols = gene_symbols)
    individuals &lt;- unique(anno_filter$individual)
  library(scales)
  library(broman)
  crayon &lt;- brocolors(&quot;crayon&quot;)
  cols &lt;- c(&quot;Mango Tango&quot;, &quot;Green&quot;, &quot;Violet Blue&quot;)
  cols &lt;- alpha(crayon[cols], .7)
}
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
</div>
<div id="suppmental-figure-10-cv-versus-mean-before-mean-correction" class="section level3">
<h3>Suppmental figure 10: CV versus mean before mean correction</h3>
<p>Legend: Coefficients of variation plotted against average molecule counts across cells of each individual cell line. Grey dots represent endogeneous genes, and blue dots indicate ERCC spike-in control genes. Red curve depicts the expected coefficients of variation assuming the endogeneous genes follow a poisson distribution. Likewise, blue curve depicts the expected CVs of the ERCC spike-in control genes. Yellow curve predicts the expected CVs assuming standard deviation is 3 times the ERCC spike-in genes.</p>
<pre class="r"><code>theme_set(theme_bw(base_size = 12))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank())
cowplot::plot_grid(
    plot_poisson_cv(molecules_filter[grep(&quot;ERCC&quot;, rownames(molecules_filter),
                                          invert = TRUE), 
                                      anno_filter$individual == &quot;NA19098&quot;], 
                    molecules_filter[grep(&quot;ERCC&quot;, rownames(molecules_filter)), 
                                     anno_filter$individual == &quot;NA19098&quot;], 
                    is_log2count = FALSE,
                    main = &quot;CV across single cells of NA19098&quot;) +
                    theme(legend.position = &quot;none&quot;),
    plot_poisson_cv(molecules_filter[grep(&quot;ERCC&quot;, rownames(molecules_filter),
                                      invert = TRUE), 
                                  anno_filter$individual == &quot;NA19101&quot;], 
                molecules_filter[grep(&quot;ERCC&quot;, rownames(molecules_filter)), 
                                 anno_filter$individual == &quot;NA19101&quot;], 
                is_log2count = FALSE,
                main = &quot;CV across single cells of NA19101&quot;) +
                theme(legend.position = &quot;none&quot;),
    plot_poisson_cv(molecules_filter[grep(&quot;ERCC&quot;, rownames(molecules_filter),
                                  invert = TRUE), 
                              anno_filter$individual == &quot;NA19239&quot;], 
            molecules_filter[grep(&quot;ERCC&quot;, rownames(molecules_filter)), 
                             anno_filter$individual == &quot;NA19239&quot;], 
            is_log2count = FALSE,
            main = &quot;CV across single cells of NA19239&quot;) +
            theme(legend.position = &quot;none&quot;),
  nrow = 1,
  labels = LETTERS[1:3])</code></pre>
<p><img src="figure/cell_to_cell_variation.Rmd/unnamed-chunk-50-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="supplemental-figure-10-adjusted-cv-versus-mean" class="section level3">
<h3>Supplemental figure 10: Adjusted CV versus mean</h3>
<p>Investigating distance-to-the-median (DM). (A) to (C) correspond to cell lines NA19098, NA19101, NA19239. DM values of each gene are plotted against log10 of the average molecule counts.</p>
<p>All cells</p>
<pre class="r"><code>plot_grid(
  ggplot(data.frame(dm = ENSG_cv_adj$NA19098$log10cv2_adj,
                    log2_mean = log2(ENSG_cv_adj$NA19098$mean)),
         aes(x = log2_mean, y = dm)) +
      geom_point(cex = .4) +
      xlab(&quot;log2 average molecule count&quot;) +
      ylab(&quot;Adjusted CV value&quot;) +
      ggtitle(&quot;Adjusted CV of NA19098&quot;) +
      theme(legend.position = &quot;none&quot;),
    ggplot(data.frame(dm = ENSG_cv_adj$NA19101$log10cv2_adj,
                    log2_mean = log10(ENSG_cv_adj$NA19101$mean)),
         aes(x = log2_mean, y = dm)) +
      geom_point(cex = .4) +
      xlab(&quot;log2 average molecule count&quot;) +
      ylab(&quot;Adjusted CV value&quot;) +
      ggtitle(&quot;Adjusted CV of NA19101&quot;) +
      theme(legend.position = &quot;none&quot;),
  ggplot(data.frame(dm = ENSG_cv_adj$NA19239$log10cv2_adj,
                    log2_mean = log10(ENSG_cv_adj$NA19239$mean)),
         aes(x = log2_mean, y = dm)) +
      geom_point(cex = .4) +
      xlab(&quot;log2 average molecule count&quot;) +
      ylab(&quot;Adjusted value&quot;) +
      ggtitle(&quot;Adjusted CV of NA19239&quot;) +
      theme(legend.position = &quot;none&quot;),
  nrow = 1,
  labels = LETTERS[4:6] )</code></pre>
<p><img src="figure/cell_to_cell_variation.Rmd/unnamed-chunk-51-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="suppmental-figure-5-permutation-test-results" class="section level3">
<h3>Suppmental figure 5: permutation test results</h3>
<p>Legend: Permutation test results for when including only expressed cells. (A) Histogram of empirical p-values based on 12,192 permutations. (B) -log10 empirical p-values are plotted against average gene expression levels. Blue line displays predicted -log10 p-values using locally weighted scatterplot smooth (LOESS). (C) Median of Absolute Deviation (MAD) of genes versus average gene expression levels. LOESS was also used to depict predicted MAD values.</p>
<pre class="r"><code># Bins average gene expression and make a boxplot of -log10 p-values
# for each bin
load(&quot;../../singleCellSeq/data/permuted-pval-expressed-set1.rda&quot;)
load(&quot;../../singleCellSeq/data/mad-expressed.rda&quot;)

plot_grid(
  ggplot( data.frame(pvals = perm_pval_set1),
         aes(x = pvals) ) +
    geom_histogram() + xlim(0, 1) +
    labs(x = &quot;Permutation-based P-values&quot;, 
         y = &quot;Count&quot;, 
         title = &quot;Empirical P-values based on permutation&quot;),
  ggplot( data.frame(pvals = perm_pval_set1,
                     gene_mean = rowMeans(as.matrix(molecules_final_expressed_subset),
                                          na.rm = TRUE),
                     bins = cut_number(rowMeans(as.matrix(molecules_final_expressed_subset),
                                                na.rm = TRUE),
                                       n = 10)),
         aes(x = rowMeans(as.matrix(molecules_final_expressed_subset),
                          na.rm = TRUE), y = -log10(pvals)) ) +
    geom_point(alpha = .5, cex = .2) +
    stat_smooth() +
    theme(legend.position = &quot;none&quot;) +
    ylab(&quot;-log10 P-value&quot;) +
    xlab(&quot;Average log2-gene expression level&quot;) +
    ggtitle(&quot;Gene expression level and empirical P-values&quot;),
  ggplot( data.frame(mad = mad_expressed,
                   gene_mean = rowMeans(as.matrix(molecules_final_expressed_subset),
                                        na.rm = TRUE),
                   bins = cut_number(rowMeans(as.matrix(molecules_final_expressed_subset),
                                        na.rm = TRUE), n = 10)),
         aes(x = gene_mean, y = mad) ) +
      geom_point( alpha = .5, cex = .2) +
      stat_smooth(col = 11) +
      theme(legend.position = &quot;none&quot;) +
      ylab(&quot;Median absolute deviation (MAD)&quot;) +
      xlab(&quot;Average log2-gene expression level&quot;) +
      ggtitle(&quot;Gene expression level and MAD&quot;),
  ncol = 2,
  labels = LETTERS[1:3]
  )</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<pre><code>Warning: Removed 16 rows containing non-finite values (stat_bin).</code></pre>
<pre><code>`geom_smooth()` using method = &#39;gam&#39;</code></pre>
<pre><code>Warning: Removed 16 rows containing non-finite values (stat_smooth).</code></pre>
<pre><code>Warning: Removed 16 rows containing missing values (geom_point).</code></pre>
<pre><code>`geom_smooth()` using method = &#39;gam&#39;</code></pre>
<pre><code>Warning: Removed 16 rows containing non-finite values (stat_smooth).

Warning: Removed 16 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/cell_to_cell_variation.Rmd/unnamed-chunk-52-1.png" width="672" style="display: block; margin: auto;" /> ###Supplemental figure 4: The gene specific dropout rate</p>
<p>Undetected cell per gene versus gene abundance, variance, and coefficient of variation</p>
<pre class="r"><code># Expressed cells
par(mfrow = c(1,3))
# abundance
plot(y = log10(expressed_cv_subset$all$expr_mean),
     x = drop_out$all,
     xlab = &quot;Gene-specific dropout rate&quot;,
     ylab = &quot;log10 mean expression&quot;,
     pch = 16, cex =.3, 
     cex.axis = 1.5, 
     cex.lab = 1.5, ylim = c(1,5))
lines(lowess( log10(expressed_cv_subset$all$expr_mean) ~ drop_out$all),
      col = &quot;red&quot;)

# variance
plot(y = log10(expressed_cv_subset$all$expr_var),
     x = drop_out$all,
     xlab = &quot;Gene-specific dropout rate&quot;,
     ylab = &quot;log10 variance of expression&quot;,
     pch = 16, cex =.3, 
     cex.axis = 1.5, 
     cex.lab = 1.5)
lines(lowess( log10(expressed_cv_subset$all$expr_var) ~ drop_out$all),
      col = &quot;red&quot;)

# mean individual adjusted-CV
plot(y = expressed_dm_subset$all*(.5),
     x = drop_out$all,
     xlab = &quot;Gene-specific dropout rate&quot;,
     ylab = &quot;adjusted log10-CV&quot;,
     pch = 16, cex =.3, 
     cex.axis = 1.5, 
     cex.lab = 1.5, ylim = c(-.5, 1))
lines(lowess( expressed_dm_subset$all*(.5) ~ drop_out$all),
      col = &quot;red&quot;)
title(main = &quot;Expressed cells, three individuals&quot;, outer = TRUE, line = -1)</code></pre>
<p><img src="figure/cell_to_cell_variation.Rmd/unnamed-chunk-53-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># all cells
ii_genes &lt;- which(rownames(molecules_final) %in% rownames(molecules_final_subset))
par(mfrow = c(1,3))
# abundance
plot(y = log10(ENSG_cv$all$mean[ii_genes]),
     x = drop_out$all,
     xlab = &quot;Gene-specific dropout rate&quot;,
     ylab = &quot;log10 mean expression&quot;,
     pch = 16, cex =.3, 
     cex.axis = 1.5, 
     cex.lab = 1.5, ylim = c(1,5))
lines(lowess( log10(ENSG_cv$all$mean[ii_genes]) ~ drop_out$all),
      col = &quot;red&quot;)

# variance
plot(y = log10(ENSG_cv$all$sd^2)[ii_genes],
     x = drop_out$all,
     xlab = &quot;Gene-specific dropout rate&quot;,
     ylab = &quot;log10 variance of expression&quot;,
     pch = 16, cex =.3, 
     cex.axis = 1.5, 
     cex.lab = 1.5)
lines(lowess( log10(ENSG_cv$all$sd^2)[ii_genes] ~ drop_out$all),
      col = &quot;red&quot;)

plot(y = ENSG_cv_adj$all$log10cv2_adj[ii_genes]*(1/2),
     x = drop_out$all,
     xlab = &quot;Gene-specific dropout rate&quot;,
     ylab = &quot;adjusted log10-CV&quot;,
     pch = 16, cex =.3, 
     cex.axis = 1.5, 
     cex.lab = 1.5, ylim= c(-.5,1))
lines(lowess( ENSG_cv_adj$all$log10cv2_adj[ii_genes]*(1/2) ~ drop_out$all),
      col = &quot;red&quot;)
title(main = &quot;All cells, three individuals&quot;, outer = TRUE, line = -1)</code></pre>
<p><img src="figure/cell_to_cell_variation.Rmd/unnamed-chunk-54-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level3">
<h3>Session Information</h3>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] stats4    parallel  grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] broman_0.65-1          scales_0.4.1           gplots_3.0.1          
 [4] lmtest_0.9-35          zoo_1.7-14             VennDiagram_1.6.17    
 [7] futile.logger_1.4.3    mygene_1.10.0          GenomicFeatures_1.26.2
[10] AnnotationDbi_1.36.1   Biobase_2.34.0         GenomicRanges_1.26.2  
[13] GenomeInfoDb_1.10.3    IRanges_2.8.1          S4Vectors_0.12.1      
[16] BiocGenerics_0.20.0    matrixStats_0.51.0     MASS_7.3-45           
[19] Humanzee_0.1.0         edgeR_3.16.5           limma_3.30.9          
[22] dplyr_0.5.0            data.table_1.10.0      workflowr_0.3.0       
[25] rmarkdown_1.3          testit_0.6             cowplot_0.7.0         
[28] ggplot2_2.2.1         

loaded via a namespace (and not attached):
 [1] nlme_3.1-128               bitops_1.0-6              
 [3] RColorBrewer_1.1-2         httr_1.2.1                
 [5] rprojroot_1.2              tools_3.3.2               
 [7] backports_1.0.5            R6_2.2.0                  
 [9] rpart_4.1-10               KernSmooth_2.23-15        
[11] Hmisc_4.0-2                DBI_0.5-1                 
[13] lazyeval_0.2.0             mgcv_1.8-15               
[15] colorspace_1.3-2           nnet_7.3-12               
[17] gridExtra_2.2.1            git2r_0.18.0              
[19] chron_2.3-49               htmlTable_1.9             
[21] rtracklayer_1.34.1         labeling_0.3              
[23] caTools_1.17.1             checkmate_1.8.2           
[25] stringr_1.1.0              digest_0.6.12             
[27] Rsamtools_1.26.1           foreign_0.8-67            
[29] XVector_0.14.0             base64enc_0.1-3           
[31] htmltools_0.3.5            htmlwidgets_0.8           
[33] RSQLite_1.1-2              jsonlite_1.2              
[35] BiocParallel_1.8.1         gtools_3.5.0              
[37] acepack_1.4.1              RCurl_1.95-4.8            
[39] magrittr_1.5               Formula_1.2-1             
[41] Matrix_1.2-7.1             Rcpp_0.12.9               
[43] munsell_0.4.3              proto_1.0.0               
[45] sqldf_0.4-10               yaml_2.1.14               
[47] stringi_1.1.2              SummarizedExperiment_1.4.0
[49] zlibbioc_1.20.0            plyr_1.8.4                
[51] gdata_2.17.0               lattice_0.20-34           
[53] Biostrings_2.42.1          splines_3.3.2             
[55] locfit_1.5-9.1             knitr_1.15.1              
[57] biomaRt_2.30.0             futile.options_1.0.0      
[59] XML_3.98-1.5               evaluate_0.10             
[61] latticeExtra_0.6-28        lambda.r_1.1.9            
[63] gtable_0.2.0               assertthat_0.1            
[65] gsubfn_0.6-6               survival_2.40-1           
[67] tibble_1.2                 GenomicAlignments_1.10.0  
[69] memoise_1.0.0              cluster_2.0.5             </code></pre>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
