<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Compare count data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Compare count data</h1>

</div>


<pre class="r"><code>library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)
library(&quot;VennDiagram&quot;)</code></pre>
<p>Import data from John’s Github:</p>
<pre class="r"><code>reads_raw &lt;- fread(&quot;../../singleCellSeq/data/reads-raw-single-per-sample.txt&quot;)
bri_reads_raw &lt;- fread(&quot;../../single-cell/data/counts-matrix/reads-raw-single-per-sample.txt&quot;
)
setDF(bri_reads_raw)
reads_raw[1:10, 10:20]</code></pre>
<pre><code>##     ENSG00000187634 ENSG00000268179 ENSG00000188976 ENSG00000187961
##  1:               0               0              57               0
##  2:               0               0             140               0
##  3:               0               0               1               0
##  4:               0               0               8               0
##  5:               0               0             167               0
##  6:               0               0              55               0
##  7:               0               0               3               0
##  8:               0               0             164               0
##  9:               0               0             105               0
## 10:               0               0              76               0
##     ENSG00000187583 ENSG00000187642 ENSG00000188290 ENSG00000187608
##  1:               0               0               0               1
##  2:               0               0               0             249
##  3:               0               0               0               0
##  4:               0               0               0               3
##  5:               0               0               0              42
##  6:               0               0               0               8
##  7:               0               0               0              48
##  8:               0               0               0             216
##  9:               0               0               0               3
## 10:               0               0               0              69
##     ENSG00000188157 ENSG00000237330 ENSG00000131591
##  1:              30               0               0
##  2:              24               0               0
##  3:               2               0               0
##  4:              30               0               0
##  5:              77               0               0
##  6:              44               0               1
##  7:               8               0               5
##  8:              24               0               0
##  9:              18               0               0
## 10:              95               0               0</code></pre>
<pre class="r"><code>bri_reads_raw[1:10, 10:20]</code></pre>
<pre><code>##    ENSG00000187634 ENSG00000268179 ENSG00000188976 ENSG00000187961
## 1                0               0              58               1
## 2                0               0             140               0
## 3                0               0               1               0
## 4                0               0              10               1
## 5                0               0             168              79
## 6                0               0              90               0
## 7                0               0               4               1
## 8                0               0             168               0
## 9                0               0             122             122
## 10               0               0              77              17
##    ENSG00000187583 ENSG00000187642 ENSG00000188290 ENSG00000187608
## 1                0               0               0               1
## 2                0               0               0             246
## 3                0               0               0               0
## 4                0               0               0               3
## 5                0               0               0              43
## 6                0               0               0               9
## 7                0               0               0              49
## 8                0               0               0             215
## 9                0               0               0               3
## 10               0               0               0              96
##    ENSG00000188157 ENSG00000237330 ENSG00000131591
## 1               34               0               0
## 2               48               0               0
## 3                2               0               0
## 4               69               0               0
## 5               91               0               0
## 6              110               0               2
## 7               39               0               4
## 8               27               0               0
## 9               19               0               0
## 10             139               0               0</code></pre>
<p>My data consitently shows higher gene counts for the genes and samples I am looking at. I am going to transpose my data similar to the gene_x_sample_count_matrix.Rmd anaylsis so I can see if I detected more genes.</p>
<pre class="r"><code>bri_reads_raw &lt;- fread(&quot;../../single-cell/data/counts-matrix/reads-raw-single-per-sample.txt&quot;
)
setDF(bri_reads_raw)</code></pre>
<p>Create annotation file for the data.</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../../singleCellSeq//data/annotation.txt&quot;, header = TRUE,stringsAsFactors = FALSE)
anno_bri &lt;- bri_reads_raw %&gt;%
  select(individual:well) %&gt;%
  mutate(batch = paste(individual, replicate, sep = &quot;.&quot;),
         sample_id = paste(batch, well, sep = &quot;.&quot;))
head(anno_bri)</code></pre>
<pre><code>##   individual replicate well      batch      sample_id
## 1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
## 2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
## 3    NA19098        r1  A03 NA19098.r1 NA19098.r1.A03
## 4    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
## 5    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
## 6    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06</code></pre>
<p>Transpose matrix.</p>
<pre class="r"><code>reads_bri &lt;-  bri_reads_raw %&gt;%
  select(starts_with(&quot;ENSG&quot;), starts_with(&quot;ERCC&quot;)) %&gt;%
  t
colnames(reads_bri) &lt;- anno_bri$sample_id
reads_bri[1:5, 1:5]</code></pre>
<pre><code>##                 NA19098.r1.A01 NA19098.r1.A02 NA19098.r1.A03
## ENSG00000186092              0              0              0
## ENSG00000237683              0              0              0
## ENSG00000235249              0              0              0
## ENSG00000185097              0              0              0
## ENSG00000269831              0              0              0
##                 NA19098.r1.A04 NA19098.r1.A05
## ENSG00000186092              0              0
## ENSG00000237683             48              0
## ENSG00000235249              0              0
## ENSG00000185097              0              0
## ENSG00000269831              0              0</code></pre>
<div id="molecule-counts-for-single-cell-samples" class="section level3">
<h3>Molecule counts for single cell samples</h3>
<pre class="r"><code>molecules_raw_bri &lt;- fread(&quot;../../single-cell/data/counts-matrix/molecules-raw-single-per-sample.txt&quot;)
setDF(molecules_raw_bri)
dim(molecules_raw_bri)</code></pre>
<pre><code>## [1]    72 20422</code></pre>
<p>Confirm order:</p>
<pre class="r"><code>stopifnot(anno_bri$individual == molecules_raw_bri$individual,
          anno_bri$replicate == molecules_raw_bri$replicate,
          anno_bri$well == molecules_raw_bri$well)</code></pre>
<p>Transpoe matrix:</p>
<pre class="r"><code>molecules_bri &lt;- molecules_raw_bri %&gt;%
  select(starts_with(&quot;ENSG&quot;), starts_with(&quot;ERCC&quot;)) %&gt;%
  t
colnames(molecules_bri) &lt;- anno_bri$sample_id
molecules_bri[1:5, 1:5]</code></pre>
<pre><code>##                 NA19098.r1.A01 NA19098.r1.A02 NA19098.r1.A03
## ENSG00000186092              0              0              0
## ENSG00000237683              0              0              0
## ENSG00000235249              0              0              0
## ENSG00000185097              0              0              0
## ENSG00000269831              0              0              0
##                 NA19098.r1.A04 NA19098.r1.A05
## ENSG00000186092              0              0
## ENSG00000237683              2              0
## ENSG00000235249              0              0
## ENSG00000185097              0              0
## ENSG00000269831              0              0</code></pre>
</div>
<div id="read-counts-for-bulk-samples" class="section level3">
<h3>Read counts for bulk samples</h3>
<pre class="r"><code>reads_bulk_raw_bri &lt;- fread(&quot;../../single-cell/data/counts-matrix/reads-raw-bulk-per-sample.txt&quot;)
setDF(reads_bulk_raw_bri)</code></pre>
<p>Create annotation file for bulk samples.</p>
<pre class="r"><code>anno_bulk_bri &lt;- reads_bulk_raw_bri %&gt;%
  select(individual:well) %&gt;%
  mutate(batch = paste(individual, replicate, sep = &quot;.&quot;),
         sample_id = paste(batch, well, sep = &quot;.&quot;))
head(anno_bulk_bri)</code></pre>
<pre><code>##   individual replicate well      batch       sample_id
## 1    NA19098        r1 bulk NA19098.r1 NA19098.r1.bulk</code></pre>
<p>Transpose bulk read information.</p>
<pre class="r"><code>reads_bulk_bri &lt;- reads_bulk_raw_bri %&gt;%
  select(starts_with(&quot;ENSG&quot;), starts_with(&quot;ERCC&quot;)) %&gt;%
  t
colnames(reads_bulk_bri) &lt;- anno_bulk_bri$sample_id</code></pre>
</div>
<div id="observed-genes" class="section level3">
<h3>Observed genes</h3>
<pre class="r"><code>stopifnot(rownames(reads_bulk_bri) == rownames(reads_bri),
          rownames(reads_bri) == rownames(molecules_bri))

genes_observed_reads_bulk_bri &lt;- rownames(reads_bulk_bri)[rowSums(reads_bulk_bri) &gt; 0]
genes_observed_reads_bri &lt;- rownames(reads_bri)[rowSums(reads_bri) &gt; 0]
genes_observed_molecules_bri &lt;- rownames(molecules_bri)[rowSums(molecules_bri) &gt; 0]
stopifnot(genes_observed_molecules_bri %in% genes_observed_reads_bri)</code></pre>
<pre class="r"><code>genes_venn_bri &lt;- venn.diagram(x = list(&quot;reads&quot; = genes_observed_reads_bri,
                                    &quot;molecules&quot; = genes_observed_molecules_bri,
                                    &quot;reads bulk&quot; = genes_observed_reads_bulk_bri),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(genes_venn_bri)</code></pre>
<p><img src="compare_count_data_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<p>Unlike the full data from git_hub. This data set has reads that are shared between reads and bulk but not in molecules. This could be due to the small samples size in this set.</p>
</div>
<div id="look-between-github-analysis-and-my-analysis" class="section level3">
<h3>Look between github analysis and my analysis:</h3>
<pre class="r"><code>reads_raw &lt;- fread(&quot;../../singleCellSeq/data/reads-raw-single-per-sample.txt&quot;)
reads &lt;- reads_raw %&gt;%
  select(starts_with(&quot;ENSG&quot;), starts_with(&quot;ERCC&quot;)) %&gt;%
  t
genes_observed_reads &lt;- rownames(reads)[rowSums(reads) &gt; 0]


bri_github_ven &lt;- venn.diagram(x = list(&quot;reads&quot; = genes_observed_reads,
                                    &quot;reads_bri&quot; = genes_observed_reads_bri),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(bri_github_ven)</code></pre>
<p><img src="compare_count_data_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<p>The excess on the reads side may be due to the individual differneces. My analysis shows 59 unique genes not found in the githup analysis.</p>
<p>Compare UMI-tools and unitools packages:</p>
<ul>
<li><p>UMI-tools:</p></li>
<li><p>Extract: UMIs are removed and appended to the read name. Any other barcode, for example a library barcode, is left on the read.</p></li>
<li><p>dedup: (The purpose of this command is to deduplicate BAM files based on the first mapping co-ordinate and the UMI attached to the read. It is assumed that the FASTQ files were processed with extract_umi.py before mapping and thus the UMI is the last word of the read name.) dedup can be run with multiple methods to identify group of reads with the same (or similar) UMI(s), from which a single read is returned. All methods start by identifying the reads with the same mapping position.</p></li>
</ul>
<p>The simpliest methods, unique and percentile, group reads with the exact same UMI. The network-based methods, cluster, adjacency and directional, build networks where nodes are UMIs and edges connect UMIs with an edit distance &lt;= threshold (usually 1). The groups of reads are then defined from the network in a method-specific manner. For all the network-based methods, a single read is returned with the most abundant UMI in each read group. For details about how the read is selected, see ‘Selecting the representative read’ below.</p>
<ul>
<li><p>Umitools:</p></li>
<li><p>Trim: Incorporate the UMI into the read name in order to later identify while processing mapped reads.</p></li>
<li><p>rmdup: For any given start site, save only one read per UMI. Writes bed3+ to stdout with before and after counts per start.</p></li>
</ul>
<p>Difference in dedup step: UMI-tools explicitly says it deals with similar UMI’s, umitools talks about specifying –mismatched for dealling with this.</p>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
