<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />

<meta name="date" content="2017-02-15" />

<title>Data transformation</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Data transformation</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2017-02-15</em></h4>

</div>


<p><strong>Last updated:</strong> 2017-02-15</p>
<p><strong>Code version:</strong> 7b71f1a</p>
<div id="data-transformation" class="section level3">
<h3>Data transformation</h3>
<pre class="r"><code>library(&quot;ggplot2&quot;)
library(&quot;cowplot&quot;)
theme_set(theme_bw(base_size = 16))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank(),
             legend.key = element_blank(),
             plot.title = element_text(size = rel(1)))
source(&quot;../../singleCellSeq/analysis/functions.R&quot;)</code></pre>
<p>This file performs principal component analysis (PCA) and displays the resultd from each stage of our data transformation (normalization) pipeline. Furthermore, it quantifies and tests the PCA results using pca-utils. The methodology described in Worley et al., 2013. Briefly, the differences between samples are calculated using the squared Mahalanobis distance. The distances are then scales so that the hypothesis test that the samples from two groups come from seperate distributionss can be computed with an F-test.</p>
</div>
<div id="input" class="section level3">
<h3>Input</h3>
<p>Input filtered annotation.</p>
<pre class="r"><code>anno_filter&lt;- read.table(&quot;../data/annotation-filter.txt&quot;, header=TRUE, stringsAsFactors = FALSE)
head(anno_filter)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
4    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
5    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06
6    NA19098        r1  A07 NA19098.r1 NA19098.r1.A07</code></pre>
<p>Input filtered molecule counts.</p>
<pre class="r"><code>molecules_filter &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header = TRUE, stringsAsFactors = FALSE)
stopifnot(ncol(molecules_filter) == nrow(anno_filter),
          colnames(molecules_filter) == anno_filter$sample_id)</code></pre>
<p>Input standardized molecule counts (CPM).</p>
<pre class="r"><code>#from stabdarize_CPM.Rmd
molecules_cpm &lt;- read.table(&quot;../data/molecules-cpm.txt&quot;, header = TRUE,
                            stringsAsFactors = FALSE)
stopifnot(ncol(molecules_cpm) == nrow(anno_filter),
          colnames(molecules_cpm) == anno_filter$sample_id)</code></pre>
<p>Input Poisson GLM transfomred molecule counts per million,</p>
<pre class="r"><code>#from technical_noise_control.Rmd
molecules_cpm_trans &lt;- read.table(&quot;../data/molecules-cpm-trans.txt&quot;, header = TRUE,
                               stringsAsFactors = FALSE)
stopifnot(ncol(molecules_cpm_trans) == nrow(anno_filter),
          colnames(molecules_cpm_trans) == anno_filter$sample_id)</code></pre>
<p>Input final batch-corrected molecule counts per million.</p>
<pre class="r"><code>#from mixed_effect_batch_correction.Rmd
molecules_final &lt;- read.table(&quot;../data/molecules-final.txt&quot;, header = TRUE,
                              stringsAsFactors = FALSE)
stopifnot(ncol(molecules_final) == nrow(anno_filter),
          colnames(molecules_final) == anno_filter$sample_id)</code></pre>
</div>
<div id="pca" class="section level3">
<h3>PCA</h3>
<div id="pca-of-filtered-data" class="section level4">
<h4>PCA of filtered data</h4>
<pre class="r"><code>pca_molecules_filter &lt;- run_pca(molecules_filter)
pca_molecules_filter_title &lt;- &quot;The effect of technical batch on single cell gene expression for raw counts data&quot;
pca_molecules_filter_plot &lt;- plot_pca(pca_molecules_filter$PCs,
         explained = pca_molecules_filter$explained,
         metadata = anno_filter, color = &quot;individual&quot;,
         shape = &quot;replicate&quot;, alpha = 0.5, size = 3) +
  # labs(title = &quot;The effect of technical batch \n on single cell gene expression data \n when raw counts are used&quot;)
  labs(title = paste(strwrap(pca_molecules_filter_title, width = 50), collapse = &quot;\n&quot;))</code></pre>
<pre class="r"><code>#create data frame with PCA output 
pca_molecules_filter_out &lt;- data.frame(Obs.id = 1:nrow(anno_filter),
                                       Obs.batch = anno_filter$batch,
                                       pca_molecules_filter$PCs[, 1:2])
write.table(pca_molecules_filter_out,
            file = &quot;../data/pca-molecules-filter.txt&quot;,
            quote = FALSE, sep = &quot;\t&quot;, row.names = FALSE)</code></pre>
<pre class="r"><code>#pca-distances -i ../data/pca-molecules-filter.txt
#pca-dendrogram -i ../data/pca-molecules-filter.txt</code></pre>
</div>
<div id="pca-of-standardized-data" class="section level4">
<h4>PCA of standardized data</h4>
<pre class="r"><code>pca_molecules_cpm &lt;- run_pca(molecules_cpm)
pca_molecules_cpm_title &lt;- &quot;The effect of technical batch on single cell gene expression for log counts per million data&quot;
pca_molecules_cpm_plot &lt;- plot_pca(pca_molecules_cpm$PCs,
         explained = pca_molecules_cpm$explained,
         metadata = anno_filter, color = &quot;individual&quot;,
         shape = &quot;replicate&quot;, alpha = 0.5, size = 3) +
  labs(title = paste(strwrap(pca_molecules_cpm_title, width = 50), collapse = &quot;\n&quot;))
  # labs(title = &quot;The effect of technical batch \n on single cell gene expression data \n when log transformed counts per million data are used&quot;)</code></pre>
<pre class="r"><code>pca_molecules_cpm_out &lt;- data.frame(Obs.id = 1:nrow(anno_filter),
                                       Obs.batch = anno_filter$batch,
                                       pca_molecules_cpm$PCs[, 1:2])
write.table(pca_molecules_cpm_out,
            file = &quot;../data/pca-molecules-cpm.txt&quot;,
            quote = FALSE, sep = &quot;\t&quot;, row.names = FALSE)</code></pre>
<pre class="r"><code>#pca-distances -i ../data/pca-molecules-cpm.txt 
#pca-dendrogram -i ../data/pca-molecules-cpm.txt </code></pre>
</div>
<div id="pca-of-poisson-glm-transformed-molecule-counts-per-million" class="section level4">
<h4>PCA of Poisson GLM transformed molecule counts per million</h4>
<pre class="r"><code>pca_molecules_cpm_trans &lt;- run_pca(molecules_cpm_trans)
pca_molecules_cpm_trans_title &lt;- &quot;The effect of technical batch on single cell gene expression for Poisson transformed data&quot;
pca_molecules_cpm_trans_plot &lt;- plot_pca(pca_molecules_cpm_trans$PCs,
         explained = pca_molecules_cpm_trans$explained,
         metadata = anno_filter, color = &quot;individual&quot;,
         shape = &quot;replicate&quot;,  alpha = 0.5, size = 3) +
  labs(title = paste(strwrap(pca_molecules_cpm_trans_title, width = 50), collapse = &quot;\n&quot;))
  # labs(title = &quot;The effect of technical batch \n on single cell gene expression data \n when Poisson transformed data are used&quot;)</code></pre>
<pre class="r"><code>pca_molecules_cpm_trans_out &lt;- data.frame(Obs.id = 1:nrow(anno_filter),
                                       Obs.batch = anno_filter$batch,
                                       pca_molecules_cpm_trans$PCs[, 1:2])
write.table(pca_molecules_cpm_trans_out,
            file = &quot;../data/pca-molecules-cpm-trans.txt&quot;,
            quote = FALSE, sep = &quot;\t&quot;, row.names = FALSE)</code></pre>
<pre class="r"><code>#pca-distances -i ../data/pca-molecules-cpm-trans.txt
#pca-dendrogram -i ../data/pca-molecules-cpm-trans.txt</code></pre>
</div>
<div id="pca-of-final-batch-corrected-data" class="section level4">
<h4>PCA of final batch-corrected data</h4>
<pre class="r"><code>pca_final &lt;- run_pca(molecules_final)
pca_final_title &lt;- &quot;PCA of standardized, normalized, and batch corrected single cell gene expression data&quot;
pca_final_plot &lt;- plot_pca(pca_final$PCs, explained = pca_final$explained,
         metadata = anno_filter, color = &quot;individual&quot;,
         shape = &quot;replicate&quot;, alpha = 0.5, size = 3) +
  labs(title = paste(strwrap(pca_final_title, width = 50), collapse = &quot;\n&quot;))
  # labs(title = &quot;PCA of standardized, \n normalized, and batch corrected \n single cell gene expression data&quot;)</code></pre>
<pre class="r"><code>pca_final_out &lt;- data.frame(Obs.id = 1:nrow(anno_filter),
                            Obs.batch = anno_filter$batch,
                            pca_final$PCs[, 1:2])
write.table(pca_final_out,
            file = &quot;../data/pca-final.txt&quot;,
            quote = FALSE, sep = &quot;\t&quot;, row.names = FALSE)</code></pre>
<pre class="r"><code>#pca-distances -i ../data/pca-final.txt
#pca-dendrogram -i ../data/pca-final.txt</code></pre>
</div>
</div>
<div id="data-transformation-figure" class="section level3">
<h3>Data transformation figure</h3>
<p>Figure 4 in paper</p>
<pre class="r"><code>plot_final &lt;- plot_grid(
  pca_molecules_filter_plot +
    scale_color_discrete(name = &quot;Individual&quot;) +
    guides(shape = FALSE) + theme(legend.position = &quot;bottom&quot;),
  pca_molecules_cpm_plot +
    scale_shape(name = &quot;Replicate&quot;) +
    guides(color = FALSE) + theme(legend.position = &quot;bottom&quot;),
  pca_molecules_cpm_trans_plot +
    scale_color_discrete(name = &quot;Individual&quot;) +
    guides(shape = FALSE) + theme(legend.position = &quot;bottom&quot;),
  pca_final_plot +
    scale_shape(name = &quot;Replicate&quot;) +
    guides(color = FALSE) + theme(legend.position = &quot;bottom&quot;),
  labels = letters[1:4])
plot_final</code></pre>
<p><img src="figure/data_transformation.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Skip writting this figure as a tiff file</p>
</div>
<div id="session-information" class="section level3">
<h3>Session Information</h3>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] Humanzee_0.1.0      devtools_1.12.0     doMC_1.3.4         
 [4] iterators_1.0.8     foreach_1.4.3       scales_0.4.1       
 [7] VennDiagram_1.6.17  futile.logger_1.4.3 testit_0.6         
[10] data.table_1.10.0   biomaRt_2.30.0      tidyr_0.6.1        
[13] gridExtra_2.2.1     lme4_1.1-12         Matrix_1.2-7.1     
[16] edgeR_3.16.5        limma_3.30.9        cowplot_0.7.0      
[19] ggplot2_2.2.1       dplyr_0.5.0         workflowr_0.3.0    
[22] rmarkdown_1.3      

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.9          locfit_1.5-9.1       lattice_0.20-34     
 [4] assertthat_0.1       rprojroot_1.2        digest_0.6.12       
 [7] R6_2.2.0             plyr_1.8.4           futile.options_1.0.0
[10] backports_1.0.5      stats4_3.3.2         RSQLite_1.1-2       
[13] evaluate_0.10        httr_1.2.1           lazyeval_0.2.0      
[16] curl_2.3             minqa_1.2.4          rstudioapi_0.6      
[19] nloptr_1.0.4         S4Vectors_0.12.1     labeling_0.3        
[22] splines_3.3.2        statmod_1.4.27       stringr_1.1.0       
[25] RCurl_1.95-4.8       munsell_0.4.3        compiler_3.3.2      
[28] BiocGenerics_0.20.0  htmltools_0.3.5      tibble_1.2          
[31] IRanges_2.8.1        codetools_0.2-15     XML_3.98-1.5        
[34] withr_1.0.2          MASS_7.3-45          bitops_1.0-6        
[37] nlme_3.1-128         gtable_0.2.0         DBI_0.5-1           
[40] git2r_0.18.0         magrittr_1.5         stringi_1.1.2       
[43] lambda.r_1.1.9       tools_3.3.2          Biobase_2.34.0      
[46] yaml_2.1.14          AnnotationDbi_1.36.1 colorspace_1.3-2    
[49] memoise_1.0.0        knitr_1.15.1        </code></pre>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
