<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittlemant" />

<meta name="date" content="2017-02-05" />

<title>FIlter genes and low-quality single cells</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">FIlter genes and low-quality single cells</h1>
<h4 class="author"><em>Briana Mittlemant</em></h4>
<h4 class="date"><em>2017-02-05</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-03-06</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> d184779</p>
<pre class="r"><code>library(&quot;biomaRt&quot;)
library(&quot;testit&quot;)
library(&quot;dplyr&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;../../singleCellSeq/analysis/functions.R&quot;)</code></pre>
<p>This file uses the list of quality single cells defined by the quality control analysis to filter the count matrices. It also removes lowly expressed genes and genes with invalid molecule counts (greater than the maximum allowlable 1024).</p>
<p>Creates the following files:</p>
<ul>
<li><p>annotation-filter.txt - Annotation file with only the high quality single cells</p></li>
<li><p>reads-filter.txt - Read counts in high quality single cells for filtered genes</p></li>
<li><p>reads-bulk-filter.txt - Read counts in bulk samples for filtered genes</p></li>
<li><p>molecules-filter.txt - Molecule counts in high quality single cells for filtered genes</p></li>
</ul>
<pre class="r"><code>anno &lt;- read.table(&quot;../../singleCellSeq/data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A03 NA19098.r1 NA19098.r1.A03
4    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
5    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
6    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06</code></pre>
<p>Imput read counts:</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../../singleCellSeq/data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
stopifnot(ncol(reads) == nrow(anno),
          colnames(reads) == anno$sample_id)</code></pre>
<p>Imput molecule counts:</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../../singleCellSeq/data/molecules.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
stopifnot(ncol(molecules) == nrow(anno),
          colnames(molecules) == anno$sample_id)</code></pre>
<p>Input read counts for bulk samples (in order to filter genes).</p>
<pre class="r"><code>reads_bulk &lt;- read.table(&quot;../../singleCellSeq/data/reads-bulk.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
stopifnot(ncol(reads_bulk) == 9)</code></pre>
<p>Input annotation for bulk samples (for PCA plot).</p>
<pre class="r"><code>anno_bulk &lt;- read.table(&quot;../../singleCellSeq/data/annotation-bulk.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno_bulk)</code></pre>
<pre><code>  individual replicate well      batch       sample_id
1    NA19098        r1 bulk NA19098.r1 NA19098.r1.bulk
2    NA19098        r2 bulk NA19098.r2 NA19098.r2.bulk
3    NA19098        r3 bulk NA19098.r3 NA19098.r3.bulk
4    NA19101        r1 bulk NA19101.r1 NA19101.r1.bulk
5    NA19101        r2 bulk NA19101.r2 NA19101.r2.bulk
6    NA19101        r3 bulk NA19101.r3 NA19101.r3.bulk</code></pre>
<div id="filter-low-quality-single-cells" class="section level3">
<h3>Filter low quality single cells</h3>
<p>This uses the list of quality reads we found using the QC_single_cell_library analysis.</p>
<p>Imput list of quality single cells. This list is in my data folder on my cell cycle github.</p>
<pre class="r"><code>quality_single_cells &lt;- scan(&quot;../../cell-cycle/data/quality-single-cells.txt&quot;,
                             what = &quot;character&quot;)</code></pre>
<p>We filter the annotation, reads, and molecules data to only include quality single cells.</p>
<pre class="r"><code>anno_filter &lt;- anno %&gt;% filter(sample_id %in% quality_single_cells)
reads_filter &lt;- reads[, colnames(reads) %in% quality_single_cells]
molecules_filter &lt;- molecules[, colnames(molecules) %in% quality_single_cells]
stopifnot(nrow(anno_filter) == ncol(reads_filter),
          nrow(anno_filter) == ncol(molecules_filter),
          anno_filter$sample_id == colnames(reads_filter),
          anno_filter$sample_id == colnames(molecules_filter))</code></pre>
<p>This shows us the number of cells by batch. The number of good quality cells is not consitent across batches.</p>
<pre class="r"><code>table(anno_filter$individual, anno_filter$replicate)</code></pre>
<pre><code>         
          r1 r2 r3
  NA19098 85  0 57
  NA19101 80 70 51
  NA19239 74 68 79</code></pre>
</div>
<div id="filter-genes" class="section level3">
<h3>Filter genes</h3>
<p>We filter the genes to exlude both those that are lowly expressed or over-expressed (&gt;= 1024 molecules in a given cell).</p>
<p>We identify the lower cutoff using the mean log2 molecule counts per million (cpm) in the 564 high quality single cells.</p>
<pre class="r"><code>molecules_cpm_mean &lt;- rowMeans(cpm(molecules_filter, log = TRUE))
hist(molecules_cpm_mean, xlab = &quot;Mean log2 molecule cpm in single cells&quot;,
     ylab = &quot;Number of genes&quot;, main = &quot;Identifying expression cutoff&quot;)
lower_exp_cutoff &lt;- 2
abline(v = lower_exp_cutoff, col = &quot;red&quot;)</code></pre>
<p><img src="figure/filter_genes_low_quality_genes.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>Why are we worried about overexpressed genes?</strong></p>
<pre class="r"><code>genes_pass_filter &lt;- rownames(molecules_filter)[molecules_cpm_mean &gt; lower_exp_cutoff]</code></pre>
<p>13112 genes have a mean log2 molecule cpm greater than 2, including 48 ERCC genes.</p>
<p>Next we identify any genes which have greater than 1024 molecules in any given single cell. These are above our theoretical maximum number of UMIs (it can happen when a highly expressed gene as multiple start sites), and thus we cannot correct them for the collision probability. <strong>(is there another way to deal with this, isn’t it a problem that we exclude genes just because we cant deal with them)</strong></p>
<pre class="r"><code>overexpressed_rows &lt;- apply(molecules_filter, 1, function(x) any(x &gt;= 1024))
overexpressed_genes &lt;- rownames(molecules_filter)[overexpressed_rows]
overexpressed_genes</code></pre>
<pre><code>[1] &quot;ENSG00000198888&quot; &quot;ENSG00000198804&quot; &quot;ENSG00000198712&quot; &quot;ENSG00000198899&quot;
[5] &quot;ENSG00000198938&quot; &quot;ENSG00000198886&quot;</code></pre>
<p>I will look into what these genes are:</p>
<ul>
<li><p>ENSG00000198888- mitochondrially encoded NADH</p></li>
<li><p>ENSG00000198804- mitochondrially encoded cytochrome c oxidase I</p></li>
<li><p>ENSG00000198712- mitochondrially encoded cytochrome c oxidase II</p></li>
<li><p>ENSG00000198899- mitochondrially encoded ATP synthase 6</p></li>
<li><p>ENSG00000198938- mitochondrially encoded cytochrome c oxidase III</p></li>
<li><p>ENSG00000198886- mitochondrially encoded NADH:ubiquinone oxidoreductase core subunit 4</p></li>
</ul>
<pre class="r"><code>ensembl &lt;- useMart(host = &quot;grch37.ensembl.org&quot;,
                   biomart = &quot;ENSEMBL_MART_ENSEMBL&quot;,
                   dataset = &quot;hsapiens_gene_ensembl&quot;)
overexpressed_genes_info &lt;- getBM(attributes = c(&quot;ensembl_gene_id&quot;, &quot;chromosome_name&quot;,
                                                 &quot;external_gene_name&quot;, &quot;transcript_count&quot;,
                                                 &quot;description&quot;),
                                  filters = &quot;ensembl_gene_id&quot;,
                                  values = overexpressed_genes[grep(&quot;ENSG&quot;, overexpressed_genes)],
                                  mart = ensembl)
overexpressed_genes_info</code></pre>
<pre><code>  ensembl_gene_id chromosome_name external_gene_name transcript_count
1 ENSG00000198712              MT             MT-CO2                1
2 ENSG00000198804              MT             MT-CO1                1
3 ENSG00000198886              MT             MT-ND4                1
4 ENSG00000198888              MT             MT-ND1                1
5 ENSG00000198899              MT            MT-ATP6                1
6 ENSG00000198938              MT             MT-CO3                1
                                                                     description
1  mitochondrially encoded cytochrome c oxidase II [Source:HGNC Symbol;Acc:7421]
2   mitochondrially encoded cytochrome c oxidase I [Source:HGNC Symbol;Acc:7419]
3     mitochondrially encoded NADH dehydrogenase 4 [Source:HGNC Symbol;Acc:7459]
4     mitochondrially encoded NADH dehydrogenase 1 [Source:HGNC Symbol;Acc:7455]
5           mitochondrially encoded ATP synthase 6 [Source:HGNC Symbol;Acc:7414]
6 mitochondrially encoded cytochrome c oxidase III [Source:HGNC Symbol;Acc:7422]</code></pre>
<p>6 genes have counts greater than 1024 in at least 1 cell. They are all mitochondrial genes. This includes 0 ERCC control genes.</p>
<p>Update list of genes passing filters.</p>
<pre class="r"><code>genes_pass_filter &lt;- setdiff(genes_pass_filter, overexpressed_genes)</code></pre>
<p>Filter the data to only include the subset of 13106 genes which pass the lower and upper expression cutoffs. This subset includes 48 ERCC genes.</p>
<pre class="r"><code>reads_filter &lt;- reads_filter[rownames(reads_filter) %in% genes_pass_filter, ]
molecules_filter &lt;- molecules_filter[rownames(molecules_filter) %in% genes_pass_filter, ]
reads_bulk_filter &lt;- reads_bulk[rownames(reads_bulk) %in% genes_pass_filter, ]
stopifnot(nrow(reads_filter) == length(genes_pass_filter),
          dim(reads_filter) == dim(molecules_filter),
          nrow(reads_bulk_filter) == nrow(molecules_filter))</code></pre>
<p>Output the filtered data to my data folder of the github cell cylce folder.</p>
<pre class="r"><code>write.table(anno_filter, &quot;../data/annotation-filter.txt&quot;, quote = FALSE,
            sep = &quot;\t&quot;, row.names = FALSE)
write.table(reads_filter, &quot;../data/reads-filter.txt&quot;, quote = FALSE,
            sep = &quot;\t&quot;, col.names = NA)
write.table(molecules_filter, &quot;../data/molecules-filter.txt&quot;, quote = FALSE,
            sep = &quot;\t&quot;, col.names = NA)
write.table(reads_bulk_filter, &quot;../data/reads-bulk-filter.txt&quot;, quote = FALSE,
            sep = &quot;\t&quot;, col.names = NA)</code></pre>
<p>Output new gene list that includes all filtered genes.</p>
<pre class="r"><code>write.table(genes_pass_filter, &quot;../data/genes-pass-filter.txt&quot;, quote = FALSE,
            sep = &quot;\t&quot;, row.names = FALSE, col.names = FALSE)</code></pre>
</div>
<div id="pca-of-filtered-data" class="section level3">
<h3>PCA of filtered data</h3>
<pre class="r"><code>pca_reads_filter &lt;- run_pca(reads_filter)
pca_reads_filter_plot &lt;- plot_pca(pca_reads_filter$PCs, explained = pca_reads_filter$explained,
         metadata = anno_filter, color = &quot;individual&quot;,
         shape = &quot;replicate&quot;) +
  labs(title = &quot;Filtered raw reads for single cells&quot;)
pca_reads_filter_plot</code></pre>
<p><img src="figure/filter_genes_low_quality_genes.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_molecules_filter &lt;- run_pca(molecules_filter)
pca_molecules_filter_plot &lt;- plot_pca(pca_molecules_filter$PCs, explained = pca_molecules_filter$explained,
         metadata = anno_filter, color = &quot;individual&quot;,
         shape = &quot;replicate&quot;) +
  labs(title = &quot;Filtered raw molecules for single cells&quot;)
pca_molecules_filter_plot</code></pre>
<p><img src="figure/filter_genes_low_quality_genes.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The individuals seperate on PC2 when we look at the data by molecule counts but not by read counts. The batch/replicate seperates slightly for NA1929 on PC2 in the molecule count PCA. wo</p>
<p>Because we start with the union of observed genes in the single cell or bulk samples and then the expression cutoff is based on the expresssion in the single cells, it is possible that a gene is not observed at all in the bulk samples. This causes an error with the PCA because that gene is invariant, so they are filtered before performing PCA. <strong>PLEASE EXPLAIN</strong></p>
<pre class="r"><code>reads_bulk_zeros &lt;- rowSums(reads_bulk_filter) == 0
rownames(reads_bulk_filter)[reads_bulk_zeros]</code></pre>
<pre><code>[1] &quot;ERCC-00048&quot;</code></pre>
<pre class="r"><code>pca_reads_bulk_filter &lt;- run_pca(reads_bulk_filter[!reads_bulk_zeros, ])
pca_reads_bulk_filter_plot &lt;- plot_pca(pca_reads_bulk_filter$PCs, explained = pca_reads_bulk_filter$explained,
         metadata = anno_bulk, color = &quot;individual&quot;,
         shape = &quot;replicate&quot;) +
  labs(title = &quot;Filtered raw reads for bulk samples&quot;)
pca_reads_bulk_filter_plot</code></pre>
<p><img src="figure/filter_genes_low_quality_genes.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] broman_0.65-1          gplots_3.0.1           matrixStats_0.51.0    
 [4] MASS_7.3-45            gridExtra_2.2.1        lme4_1.1-12           
 [7] workflowr_0.4.0        org.Hs.eg.db_3.4.0     GOstats_2.40.0        
[10] graph_1.52.0           Category_2.40.0        Matrix_1.2-7.1        
[13] GO.db_3.4.0            AnnotationDbi_1.36.2   IRanges_2.8.1         
[16] S4Vectors_0.12.1       Biobase_2.34.0         BiocGenerics_0.20.0   
[19] Humanzee_0.1.0         sva_3.22.0             genefilter_1.56.0     
[22] mgcv_1.8-15            nlme_3.1-128           BiocInstaller_1.24.0  
[25] scales_0.4.1           VennDiagram_1.6.17     futile.logger_1.4.3   
[28] edgeR_3.16.5           limma_3.30.11          dplyr_0.5.0           
[31] testit_0.6             data.table_1.10.4      biomaRt_2.30.0        
[34] cowplot_0.7.0          stringr_1.2.0          reshape2_1.4.2        
[37] rmarkdown_1.3          Citrus_0.99            easyGgplot2_1.0.0.9000
[40] ggplot2_2.2.1         

loaded via a namespace (and not attached):
 [1] httr_1.2.1             splines_3.3.2          gtools_3.5.0          
 [4] assertthat_0.1         RBGL_1.50.0            yaml_2.1.14           
 [7] RSQLite_1.1-2          backports_1.0.5        lattice_0.20-34       
[10] digest_0.6.12          RColorBrewer_1.1-2     minqa_1.2.4           
[13] colorspace_1.3-2       htmltools_0.3.5        plyr_1.8.4            
[16] GSEABase_1.36.0        XML_3.98-1.5           devtools_1.12.0       
[19] xtable_1.8-2           gdata_2.17.0           git2r_0.18.0          
[22] tibble_1.2             annotate_1.52.1        withr_1.0.2           
[25] lazyeval_0.2.0         survival_2.40-1        magrittr_1.5          
[28] memoise_1.0.0          evaluate_0.10          tools_3.3.2           
[31] munsell_0.4.3          locfit_1.5-9.1         lambda.r_1.1.9        
[34] caTools_1.17.1         nloptr_1.0.4           RCurl_1.95-4.8        
[37] rstudioapi_0.6         AnnotationForge_1.16.1 bitops_1.0-6          
[40] labeling_0.3           gtable_0.2.0           DBI_0.5-1             
[43] curl_2.3               R6_2.2.0               knitr_1.15.1          
[46] rprojroot_1.2          futile.options_1.0.0   KernSmooth_2.23-15    
[49] stringi_1.1.2          Rcpp_0.12.9           </code></pre>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
