<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Create gene-x-sample count matrices and annotation file</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Create gene-x-sample count matrices and annotation file</h1>

</div>


<p><strong>Last updated:</strong> 2017-01-30</p>
<p><strong>Code version:</strong> aedf051</p>
<p>This is recreating code from <a href="http://jdblischak.github.io/singleCellSeq/analysis/prepare-counts.html" class="uri">http://jdblischak.github.io/singleCellSeq/analysis/prepare-counts.html</a>.</p>
<pre class="r"><code>library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)
library(&quot;VennDiagram&quot;)</code></pre>
<p>This analysis will take the matrix from the fastq to counts anayisis pipeline (sample-by-gene) to a standard gene-x-sample matrix. I will also create annotation files and romove the genes with no observed sequencing data in the experiment. I will use the data from git hub rather than the data I created because I want to look at the whole experiment rather than one plate at this point.</p>
<div id="read-counts-for-single-cell-samples" class="section level3">
<h3>Read Counts for single cell samples:</h3>
<pre class="r"><code>reads_raw &lt;- fread(&quot;../../singleCellSeq/data/reads-raw-single-per-sample.txt&quot;)
setDF(reads_raw)</code></pre>
<p>Explore this data:</p>
<pre class="r"><code>dim(reads_raw)</code></pre>
<pre><code>[1]   864 20422</code></pre>
<p>Ths has rows for each individule and the columns give information about the sample then give counts for each gene.</p>
<p>Create annotation file for single cell samples:</p>
<pre class="r"><code>anno &lt;- reads_raw %&gt;%
  select(individual:well) %&gt;%
  mutate(batch = paste(individual, replicate, sep = &quot;.&quot;),
         sample_id = paste(batch, well, sep = &quot;.&quot;))
head(anno)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A03 NA19098.r1 NA19098.r1.A03
4    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
5    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
6    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06</code></pre>
<p>This annoation file has information about the sample, replicatte, batch.</p>
<p>Transpose the matrix so that it is gene-x-sample.</p>
<pre class="r"><code>reads &lt;- reads_raw %&gt;%
  select(starts_with(&quot;ENSG&quot;), starts_with(&quot;ERCC&quot;)) %&gt;%
  t
colnames(reads) &lt;- anno$sample_id
reads[1:5, 1:5]</code></pre>
<pre><code>                NA19098.r1.A01 NA19098.r1.A02 NA19098.r1.A03
ENSG00000186092              0              0              0
ENSG00000237683              0              0              0
ENSG00000235249              0              0              0
ENSG00000185097              0              0              0
ENSG00000269831              0              0              0
                NA19098.r1.A04 NA19098.r1.A05
ENSG00000186092              0              0
ENSG00000237683             42              0
ENSG00000235249              0              0
ENSG00000185097              0              0
ENSG00000269831              0              0</code></pre>
</div>
<div id="molecule-counts-for-single-cell-samples" class="section level3">
<h3>Molecule Counts for single cell samples</h3>
<p>Import eaw molecules for single cell samples.</p>
<pre class="r"><code>molecules_raw &lt;- fread(&quot;../../singleCellSeq/data/molecules-raw-single-per-sample.txt&quot;)
setDF(molecules_raw)
dim(molecules_raw)</code></pre>
<pre><code>[1]   864 20422</code></pre>
<p>This data has rows by individuale and has gene columns like the reads data.</p>
<p>Conform single cell samples in reads and molecules files are in same order.</p>
<pre class="r"><code>stopifnot(anno$individual == molecules_raw$individual,
          anno$replicate == molecules_raw$replicate,
          anno$well == molecules_raw$well)</code></pre>
<p>Transpose the matrix so it is also gene-x-sample.</p>
<pre class="r"><code>molecules &lt;- molecules_raw %&gt;%
  select(starts_with(&quot;ENSG&quot;), starts_with(&quot;ERCC&quot;)) %&gt;%
  t
colnames(molecules) &lt;- anno$sample_id
molecules[1:5, 1:5]</code></pre>
<pre><code>                NA19098.r1.A01 NA19098.r1.A02 NA19098.r1.A03
ENSG00000186092              0              0              0
ENSG00000237683              0              0              0
ENSG00000235249              0              0              0
ENSG00000185097              0              0              0
ENSG00000269831              0              0              0
                NA19098.r1.A04 NA19098.r1.A05
ENSG00000186092              0              0
ENSG00000237683              1              0
ENSG00000235249              0              0
ENSG00000185097              0              0
ENSG00000269831              0              0</code></pre>
<p>Now this file looks like the first one.</p>
</div>
<div id="read-counts-for-bulk-samples" class="section level3">
<h3>Read counts for bulk samples</h3>
<p>Import raw read counts for bulk samples.</p>
<pre class="r"><code>reads_bulk_raw &lt;- fread(&quot;../../singleCellSeq/data/reads-raw-bulk-per-sample.txt&quot;)
setDF(reads_bulk_raw) #converts the data.table to a data.frame</code></pre>
<p>Create annotation file for the bulk samples as we did with the single cells.</p>
<pre class="r"><code>anno_bulk &lt;- reads_bulk_raw %&gt;%
  select(individual:well) %&gt;%
  mutate(batch = paste(individual, replicate, sep = &quot;.&quot;),
         sample_id = paste(batch, well, sep = &quot;.&quot;))
head(anno_bulk)</code></pre>
<pre><code>  individual replicate well      batch       sample_id
1    NA19098        r1 bulk NA19098.r1 NA19098.r1.bulk
2    NA19098        r2 bulk NA19098.r2 NA19098.r2.bulk
3    NA19098        r3 bulk NA19098.r3 NA19098.r3.bulk
4    NA19101        r1 bulk NA19101.r1 NA19101.r1.bulk
5    NA19101        r2 bulk NA19101.r2 NA19101.r2.bulk
6    NA19101        r3 bulk NA19101.r3 NA19101.r3.bulk</code></pre>
<p>Now transpose the bulk read informaiton to match the gene-x-sample format.</p>
<pre class="r"><code>reads_bulk &lt;- reads_bulk_raw %&gt;%
  select(starts_with(&quot;ENSG&quot;), starts_with(&quot;ERCC&quot;)) %&gt;%
  t
colnames(reads_bulk) &lt;- anno_bulk$sample_id
reads_bulk[1:5, 1:5]</code></pre>
<pre><code>                NA19098.r1.bulk NA19098.r2.bulk NA19098.r3.bulk
ENSG00000186092               0               0               0
ENSG00000237683              50              61              39
ENSG00000235249               0               0               0
ENSG00000185097               0               0               0
ENSG00000269831               0               0               0
                NA19101.r1.bulk NA19101.r2.bulk
ENSG00000186092               0               0
ENSG00000237683              56              41
ENSG00000235249               0               0
ENSG00000185097               0               0
ENSG00000269831               0               0</code></pre>
</div>
<div id="observed-genes" class="section level3">
<h3>Observed genes</h3>
<p>I will look for genes out of the 20419 genes that were never observed in the experiment.</p>
<pre class="r"><code>stopifnot(rownames(reads_bulk) == rownames(reads),
          rownames(reads) == rownames(molecules))

genes_observed_reads_bulk &lt;- rownames(reads_bulk)[rowSums(reads_bulk) &gt; 0]
genes_observed_reads &lt;- rownames(reads)[rowSums(reads) &gt; 0]
genes_observed_molecules &lt;- rownames(molecules)[rowSums(molecules) &gt; 0]
stopifnot(genes_observed_molecules %in% genes_observed_reads)</code></pre>
<p>18726 genes has at least one observation in the single read data. 18726 genes had at least one observation in the single cell molecule data. 18056 genes has at least one observation in te bulk read data. As expected, all genes with at least one observed molecule in at least one single cell also had at least one observed read in at least one single cell.</p>
<pre class="r"><code>genes_venn &lt;- venn.diagram(x = list(&quot;reads&quot; = genes_observed_reads,
                                    &quot;molecules&quot; = genes_observed_molecules,
                                    &quot;reads bulk&quot; = genes_observed_reads_bulk),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(genes_venn)</code></pre>
<p><img src="figure/gene_x_sample_count_matrix.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This ven diagram shows the intesection of the genes found in each type of sequencing data. I will now remove all genes with no observed rean in either bulk or single cell sample.</p>
<pre class="r"><code>genes_observed &lt;- union(genes_observed_reads, genes_observed_reads_bulk)
reads &lt;- reads[rownames(reads) %in% genes_observed, ]
molecules &lt;- molecules[rownames(molecules) %in% genes_observed, ]
reads_bulk &lt;- reads_bulk[rownames(reads_bulk) %in% genes_observed, ]</code></pre>
<p>This leaves 19027 genes. (genes_observed)</p>
</div>
<div id="write-files" class="section level3">
<h3>Write files</h3>
<p>This code outputs the annotation files.</p>
<pre class="r"><code>write.table(anno, &quot;../../singleCellSeq/data/annotation.txt&quot;, quote=FALSE, sep=&quot;\t&quot;, row.names = FALSE)
write.table(anno_bulk, &quot;../../singleCellSeq/data/annotation-bulk.txt&quot;, quote=FALSE, sep = &quot;\t&quot;, row.names=FALSE)</code></pre>
<p>Output read counts.</p>
<pre class="r"><code>write.table(reads, &quot;../../singleCellSeq/data/reads.txt&quot;, quote = FALSE, sep = &quot;\t&quot;, col.names = NA)
write.table(reads_bulk, &quot;../../singleCellSeq/data/reads-bulk.txt&quot;, quote=FALSE, sep=&quot;\t&quot;, col.names = NA)</code></pre>
<p>Output molecule counts.</p>
<pre class="r"><code>write.table(molecules, &quot;../../singleCellSeq/data/molecules.txt&quot;, quote= FALSE, sep= &quot;\t&quot;, col.names = NA)</code></pre>
</div>
<div id="session-information" class="section level3">
<h3>Session Information</h3>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] VennDiagram_1.6.17  futile.logger_1.4.3 dplyr_0.5.0        
[4] data.table_1.10.0   devtools_1.12.0     workflowr_0.3.0    
[7] rmarkdown_1.3      

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.9          knitr_1.15.1         magrittr_1.5        
 [4] R6_2.2.0             stringr_1.1.0        tools_3.3.2         
 [7] DBI_0.5-1            lambda.r_1.1.9       git2r_0.18.0        
[10] withr_1.0.2          htmltools_0.3.5      yaml_2.1.14         
[13] lazyeval_0.2.0       rprojroot_1.2        digest_0.6.11       
[16] assertthat_0.1       tibble_1.2           futile.options_1.0.0
[19] memoise_1.0.0        evaluate_0.10        stringi_1.1.2       
[22] backports_1.0.5     </code></pre>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
