<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>Genome wide data comparison</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Genome wide data comparison</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2/6/2017</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-03-06</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> f3c5dc0</p>
<pre class="r"><code>library(&quot;biomaRt&quot;)
library(&quot;dplyr&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;../../singleCellSeq/analysis/functions.R&quot;)</code></pre>
<div id="input-read-data" class="section level3">
<h3>Input read data:</h3>
<pre class="r"><code>reads &lt;-read.table(&quot;../../singleCellSeq/data/reads.txt&quot;, header=TRUE, stringsAsFactors = FALSE)
#subset to look at same cells I did
reads_subset &lt;-reads[ ,grep(&quot;NA19098.r1.A|NA19098.r1.B|NA19098.r1.C|NA19098.r1.D|NA19098.r1.E|NA19098.r1.F&quot;, colnames(reads))]
reads_bri&lt;-read.table(&quot;../data/reads_bri.txt&quot;, header=TRUE, stringsAsFactors= FALSE)</code></pre>
<p>Create a single matrix with the common genes in both sets.</p>
<pre class="r"><code>#make my data different 
colnames(reads_bri)&lt;- paste0(colnames(reads_bri), &quot;-bri&quot;)
#list of common genes
common_genes &lt;- intersect(row.names(reads_subset), row.names(reads_bri))
reads_subset_common &lt;- reads_subset %&gt;% filter(row.names(reads_subset) %in% common_genes)
row.names(reads_subset_common)&lt;- common_genes
reads_bri_common &lt;- reads_bri %&gt;% filter(row.names(reads_bri) %in% common_genes)
row.names(reads_bri_common)&lt;- common_genes</code></pre>
<pre class="r"><code>#combine read matrics

reads_all_common &lt;-cbind(reads_subset_common, reads_bri_common)
#remove genes that are never seen
genes_in_reads_all_common&lt;- rownames(reads_all_common)[rowSums(reads_all_common) &gt; 0]
reads_all_common_no0 &lt;- reads_all_common %&gt;% filter(row.names(reads_all_common) %in% genes_in_reads_all_common)
row.names(reads_all_common_no0) &lt;- genes_in_reads_all_common</code></pre>
</div>
<div id="plots-to-explore-differences-in-data" class="section level2">
<h2>Plots to explore differences in data</h2>
<div id="create-pca-plots" class="section level3">
<h3>Create PCA plots</h3>
<pre class="r"><code>pca_commongenes &lt;- run_pca(reads_all_common_no0)

contains_bri&lt;- ifelse(grepl(&quot;bri&quot;, colnames(reads_all_common_no0)), &quot;Bri&quot;, &quot;git&quot;)
list_of_cells&lt;- colnames(reads_all_common_no0)

analysis_rounds&lt;- cbind(list_of_cells, contains_bri) 

pca_commongenes_plot &lt;- plot_pca(pca_commongenes$PCs, explained = pca_commongenes$explained, color=&quot;contains_bri&quot; ) + labs(title=&quot;Reads in github and Bri analysis&quot;) 




pca_commongenes_plot</code></pre>
<p><img src="figure/genome_wide_data_comparison.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="create-correlation-matrix" class="section level3">
<h3>Create correlation matrix</h3>
<pre class="r"><code>reads_subset_mean&lt;- apply(reads_subset_common, 1, mean)
reads_bri_mean&lt;-apply(reads_bri_common,1, mean)
distribution_reads &lt;-data.frame(reads_subset_mean, reads_bri_mean)
cor(distribution_reads)</code></pre>
<pre><code>                  reads_subset_mean reads_bri_mean
reads_subset_mean         1.0000000      0.8535153
reads_bri_mean            0.8535153      1.0000000</code></pre>
</div>
<div id="work-with-difference-between-matricies" class="section level3">
<h3>Work with difference between matricies:</h3>
<pre class="r"><code>reads_subset_common_matrix &lt;- data.matrix(reads_subset_common)
reads_bri_common_matrix &lt;- data.matrix(reads_bri_common)

#create matrix with difference of mine vs git data
diff_matrix= reads_bri_common_matrix - reads_subset_common_matrix
#create difference matrix with absolute difference between datasets
abs_diff_matrix= abs(reads_bri_common_matrix- reads_subset_common_matrix)

reads_dif_mean&lt;- apply(diff_matrix, 1, mean)

index_ERCC &lt;- grep(&quot;ERCC&quot;, names(reads_bri_mean))
ERCC_genes &lt;- names(reads_bri_mean)[index_ERCC]
which_ERCC &lt;- as.numeric(names(reads_bri_mean) %in% ERCC_genes) + 1
col_vec &lt;- c(&quot;coral&quot;, &quot;blue&quot;)[as.numeric(which_ERCC)]
plot(reads_bri_mean~reads_dif_mean, main=&quot;Mean difference in reads compared to Bri mean reads&quot;, xlab=&quot;Bri mean reads&quot;, ylab = &quot; Bri mean reads - git mean reads&quot; )</code></pre>
<p><img src="figure/genome_wide_data_comparison.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot abs value difference 
reads_abs_dif_mean &lt;- apply(abs_diff_matrix,1, mean)


library(&quot;scales&quot;)
plot(log10(reads_bri_mean) ~ log10(reads_abs_dif_mean), main= &quot;Absolute value of mean read counts&quot;, xlab=&quot;Bri mean reads&quot;, ylab = &quot;Absolute value of the difference between bri and git reads&quot;, pch = 16, cex = .6, 
     col = alpha(col_vec, .4))</code></pre>
<p><img src="figure/genome_wide_data_comparison.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#which gene is so different?
reads_abs_dif_mean[which.max(reads_abs_dif_mean)]</code></pre>
<pre><code>ERCC-00074 
  60974.06 </code></pre>
<p>This shows that the difference is in the direction of bri having larger read counts in most places.</p>
<pre class="r"><code>cut_off_difference&lt;- quantile(reads_abs_dif_mean, 0.95)
cut_off_difference</code></pre>
<pre><code>     95% 
117.1618 </code></pre>
<pre class="r"><code>#reads_abs_95 &lt;- reads_abs_dif_mean[cut_off_difference
reads_abs_dif_mean_df &lt;- data.frame(reads_abs_dif_mean)        

reads_abs_dif_95 &lt;- reads_abs_dif_mean_df[reads_abs_dif_mean_df$reads_abs_dif_mean &lt; cut_off_difference, , drop = FALSE]

reads_bri_mean_df&lt;- data.frame(reads_bri_mean)
common_genes_95 &lt;- intersect(row.names(reads_abs_dif_95),row.names(reads_bri_mean_df))
reads_bri_mean_95 &lt;- reads_bri_mean_df[row.names(reads_bri_mean_df) %in% common_genes_95, , drop=FALSE]

plot(reads_abs_dif_95$reads_abs_dif_mean ~ reads_bri_mean_95$reads_bri_mean, main=&quot;Difference plot for lower 95%&quot;, ylab=&quot;abs value of mean difference&quot;, xlab=&quot;Bri mean reads&quot;)</code></pre>
<p><img src="figure/genome_wide_data_comparison.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>percent_change_m&lt;- ((reads_bri_common_matrix- reads_subset_common_matrix)/2)
mean_percent_change&lt;- apply(percent_change_m,1, mean)
plot(mean_percent_change~ reads_bri_mean, main=&quot;Average change as a function of bri mean reads&quot;, ylab = &quot;(bri-git)/2&quot;)</code></pre>
<p><img src="figure/genome_wide_data_comparison.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>reads_subset_common_matrix &lt;- data.matrix(reads_subset_common)
reads_bri_common_matrix &lt;- data.matrix(reads_bri_common)

percent_diff&lt;- ((reads_subset_common_matrix - reads_bri_common_matrix)/(.5* (reads_subset_common_matrix + reads_bri_common_matrix)))

percent_diff_mean= apply(percent_diff,1, mean)
plot(percent_diff_mean~reads_bri_mean)</code></pre>
<p><img src="figure/genome_wide_data_comparison.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>percent_diff_mean[which.max(percent_diff_mean)]</code></pre>
<pre><code>ENSG00000183513 
      0.5207729 </code></pre>
</div>
<div id="look-into-the-outlier-ercc00074" class="section level3">
<h3>Look into the outlier: ERCC00074</h3>
<pre class="r"><code>reads_bri2&lt;-read.table(&quot;../data/reads_bri.txt&quot;, header=TRUE, stringsAsFactors= FALSE)
ERCC_74 &lt;- rbind(reads_subset[&quot;ERCC-00074&quot;,], reads_bri2[&quot;ERCC-00074&quot;,])
row.names(ERCC_74) &lt;- c(&quot;git&quot;, &quot;bri&quot;)
ERCC_74.T= ERCC_74 %&gt;% t
plot(ERCC_74.T[,&quot;git&quot;] ~ERCC_74.T[,&quot;bri&quot;], main=&quot;Compare ERCC74 in 2 data sets&quot;, ylab=&quot;git&quot;, xlab=&quot;bri&quot;)</code></pre>
<p><img src="figure/genome_wide_data_comparison.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ERCC_74.T_df=data.frame(ERCC_74.T)



ERCC74_diff= ERCC_74.T_df$bri- ERCC_74.T_df$git
ERCC_74_dif&lt;-cbind(ERCC_74.T_df,ERCC74_diff)

plot(ERCC_74_dif$ERCC74_diff ~ERCC_74_dif$bri, main=&quot;read count difference in ERCC74&quot;, xlab=&quot;bri read count&quot;, ylab=&quot;difference in read count&quot;)</code></pre>
<p><img src="figure/genome_wide_data_comparison.Rmd/unnamed-chunk-17-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>labels_boxplot= c(&quot;git&quot;, &quot;bri&quot;)
boxplot(ERCC74_diff, main=&quot;Difference in ERCC00074 read counts&quot;)</code></pre>
<p><img src="figure/genome_wide_data_comparison.Rmd/unnamed-chunk-17-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>boxplot(ERCC_74.T_df$git, ERCC_74.T_df$bri, las=2, names=c(&quot;git&quot;, &quot;bri&quot;), ylab=&quot;Read count&quot;)</code></pre>
<p><img src="figure/genome_wide_data_comparison.Rmd/unnamed-chunk-17-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ERCC_74_dif[which.max(ERCC_74_dif$ERCC74_diff),]</code></pre>
<pre><code>                 git    bri ERCC74_diff
NA19098.r1.E01 16512 141174      124662</code></pre>
<p>Check if this gene is in the quality single cells list.</p>
<pre class="r"><code>quality_single_cells &lt;- scan(&quot;../data/quality-single-cells.txt&quot;,
                             what = &quot;character&quot;)

&quot;NA19098.r1.E01&quot; %in% quality_single_cells</code></pre>
<pre><code>[1] TRUE</code></pre>
<p>This cell was included in the analysis.</p>
<p>For a baseline I will look at another ERCC gene</p>
<pre class="r"><code>&quot;ERCC-00045&quot; %in% common_genes_95</code></pre>
<pre><code>[1] FALSE</code></pre>
<pre class="r"><code>ERCC_67 &lt;- rbind(reads_subset[&quot;ERCC-00067&quot;,], reads_bri2[&quot;ERCC-00067&quot;,])
row.names(ERCC_67) &lt;- c(&quot;git&quot;, &quot;bri&quot;)
ERCC_67.T= ERCC_67 %&gt;% t
plot(ERCC_67.T[,&quot;git&quot;] ~ ERCC_67.T[,&quot;bri&quot;], main=&quot;Compare ERCC67 in 2 data sets&quot;, ylab=&quot;git&quot;, xlab=&quot;bri&quot;)</code></pre>
<p><img src="figure/genome_wide_data_comparison.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ERCC_67.T_df= data.frame(ERCC_67.T)

boxplot(ERCC_67.T_df$git, ERCC_67.T_df$bri, las=2, names=c(&quot;git&quot;, &quot;bri&quot;), ylab=&quot;Read count&quot;)</code></pre>
<p><img src="figure/genome_wide_data_comparison.Rmd/unnamed-chunk-19-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>boxplot(ERCC_67.T_df$git, ERCC_74.T_df$bri, las=2, names=c(&quot;git-67&quot;, &quot;git-74&quot;), ylab=&quot;Read count&quot;)</code></pre>
<p><img src="figure/genome_wide_data_comparison.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] broman_0.65-1          gplots_3.0.1           matrixStats_0.51.0    
 [4] MASS_7.3-45            gridExtra_2.2.1        lme4_1.1-12           
 [7] workflowr_0.4.0        org.Hs.eg.db_3.4.0     GOstats_2.40.0        
[10] graph_1.52.0           Category_2.40.0        Matrix_1.2-7.1        
[13] GO.db_3.4.0            AnnotationDbi_1.36.2   IRanges_2.8.1         
[16] S4Vectors_0.12.1       Biobase_2.34.0         BiocGenerics_0.20.0   
[19] Humanzee_0.1.0         sva_3.22.0             genefilter_1.56.0     
[22] mgcv_1.8-15            nlme_3.1-128           BiocInstaller_1.24.0  
[25] scales_0.4.1           VennDiagram_1.6.17     futile.logger_1.4.3   
[28] edgeR_3.16.5           limma_3.30.11          dplyr_0.5.0           
[31] testit_0.6             data.table_1.10.4      biomaRt_2.30.0        
[34] cowplot_0.7.0          stringr_1.2.0          reshape2_1.4.2        
[37] rmarkdown_1.3          Citrus_0.99            easyGgplot2_1.0.0.9000
[40] ggplot2_2.2.1         

loaded via a namespace (and not attached):
 [1] httr_1.2.1             splines_3.3.2          gtools_3.5.0          
 [4] assertthat_0.1         RBGL_1.50.0            yaml_2.1.14           
 [7] RSQLite_1.1-2          backports_1.0.5        lattice_0.20-34       
[10] digest_0.6.12          RColorBrewer_1.1-2     minqa_1.2.4           
[13] colorspace_1.3-2       htmltools_0.3.5        plyr_1.8.4            
[16] GSEABase_1.36.0        XML_3.98-1.5           devtools_1.12.0       
[19] xtable_1.8-2           gdata_2.17.0           git2r_0.18.0          
[22] tibble_1.2             annotate_1.52.1        withr_1.0.2           
[25] lazyeval_0.2.0         survival_2.40-1        magrittr_1.5          
[28] memoise_1.0.0          evaluate_0.10          tools_3.3.2           
[31] munsell_0.4.3          locfit_1.5-9.1         lambda.r_1.1.9        
[34] caTools_1.17.1         nloptr_1.0.4           RCurl_1.95-4.8        
[37] rstudioapi_0.6         AnnotationForge_1.16.1 bitops_1.0-6          
[40] labeling_0.3           gtable_0.2.0           DBI_0.5-1             
[43] curl_2.3               R6_2.2.0               knitr_1.15.1          
[46] rprojroot_1.2          futile.options_1.0.0   KernSmooth_2.23-15    
[49] stringi_1.1.2          Rcpp_0.12.9           </code></pre>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
