<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />

<meta name="date" content="2017-03-31" />

<title>Expression of non singleton expression</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Expression of non singleton expression</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2017-03-31</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-03-31</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 9edcd66</p>
<!-- Add your analysis here -->
<div id="pull-in-genes-lists" class="section level3">
<h3>Pull in genes lists</h3>
<pre class="r"><code>library(&quot;biomaRt&quot;)
library(&quot;data.table&quot;)
library(&quot;testit&quot;)
library(&quot;dplyr&quot;)
library(&quot;edgeR&quot;)
library(&quot;VennDiagram&quot;)
library(&quot;ggplot2&quot;)
library(&quot;vioplot&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;../../singleCellSeq/analysis/functions.R&quot;)</code></pre>
</div>
<div id="input-data" class="section level3">
<h3>Input data</h3>
<pre class="r"><code>anno &lt;- read.table(&quot;../../singleCellSeq/data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
#input reads (pre-filter)
reads &lt;- read.table(&quot;../../singleCellSeq/data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
#input molecule (pre-filter)
molecules &lt;- read.table(&quot;../../singleCellSeq/data/molecules.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
#input bulk reads (pre-filter )
reads_bulk &lt;- read.table(&quot;../../singleCellSeq/data/reads-bulk.txt&quot;, header = TRUE, stringsAsFactors = FALSE)</code></pre>
</div>
<div id="only-in-single-cell" class="section level3">
<h3>Only in Single cell</h3>
<p>This is the ENSG genes only in single cell.</p>
<pre class="r"><code>load(&quot;../data/molecules_only_SC_ENSG.Rda&quot;)</code></pre>
<pre class="r"><code>non_zero_cells_ENSG &lt;- apply(molecules_only_SC_ENSG, 1, function(c)sum(c!=0))

molecules_with_nonzero_ENSG &lt;- cbind(molecules_only_SC_ENSG, non_zero_cells_ENSG)</code></pre>
</div>
<div id="non-singletons" class="section level3">
<h3>Non singletons</h3>
<pre class="r"><code>genes_not_singleton_ENSG= row.names(molecules_with_nonzero_ENSG[molecules_with_nonzero_ENSG$non_zero_cells_ENSG &gt; 1, ])
length(genes_not_singleton_ENSG)</code></pre>
<pre><code>[1] 535</code></pre>
<pre class="r"><code>#535 ensg genes </code></pre>
<pre class="r"><code>load(&quot;../data/molecules_and_mean_order_new.Rda&quot;)

###help making these 1 plot 

which_onlySC_nonsing &lt;- as.numeric(rownames(molecules_and_mean_order_new) %in% genes_not_singleton_ENSG) + 1

                            
col_vec_onlySC_nonsing &lt;- c(&quot;gray70&quot;, &quot;darkviolet&quot;)[as.numeric(which_onlySC_nonsing)]
shap_vec_onlySC_nonsing&lt;- c(16,1)[as.numeric(which_onlySC_nonsing)]
library(&quot;scales&quot;)





mean_expression_nonsing &lt;- plot((molecules_and_mean_order_new$molecules_mean_new) , pch=shap_vec_onlySC_nonsing, cex=2, col=col_vec_onlySC_nonsing, main = &quot;Mean expression of molecules per gene&quot;, ylab=&quot; Molecule count&quot;, xlab = &quot;Genes ordered by expression&quot;)
legend(&quot;topright&quot;, c(&quot;Single cell only non singleton&quot;, &quot;All genes&quot;), pch=c(16,16), col=c(&quot;darkviolet&quot;, &quot;gray70&quot;) )</code></pre>
<p><img src="figure/non_singleton_expression.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="only-singleton" class="section level3">
<h3>Only singleton</h3>
<pre class="r"><code>genes_singleton_ENSG= row.names(molecules_with_nonzero_ENSG[molecules_with_nonzero_ENSG$non_zero_cells_ENSG == 1, ])
length(genes_singleton_ENSG)</code></pre>
<pre><code>[1] 421</code></pre>
<pre class="r"><code>which_onlySC_sing &lt;- as.numeric(rownames(molecules_and_mean_order_new) %in% genes_singleton_ENSG) + 1

                            
col_vec_onlySC_sing &lt;- c(&quot;gray70&quot;, &quot;cyan2&quot;)[as.numeric(which_onlySC_sing)]
shap_vec_onlySC_sing&lt;- c(16,1)[as.numeric(which_onlySC_sing)]
library(&quot;scales&quot;)





mean_expression_sing &lt;- plot((molecules_and_mean_order_new$molecules_mean_new) , pch=shap_vec_onlySC_sing, cex=2, col=col_vec_onlySC_sing, main = &quot;Mean expression of molecules per gene&quot;, ylab=&quot; Molecule count&quot;, xlab = &quot;Genes ordered by expression&quot;)
legend(&quot;topright&quot;, c(&quot;Single cell singletons&quot;, &quot;All genes&quot;), pch=c(16,16), col=c(&quot;cyan2&quot;, &quot;gray70&quot;) )</code></pre>
<p><img src="figure/non_singleton_expression.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] stats4    parallel  grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] cowplot_0.7.0          stringr_1.2.0          reshape2_1.4.2        
 [4] Citrus_0.99            easyGgplot2_1.0.0.9000 pamr_1.55             
 [7] survival_2.40-1        cluster_2.0.5          bladderbatch_1.12.0   
[10] BiocInstaller_1.24.0   sva_3.22.0             genefilter_1.56.0     
[13] mgcv_1.8-15            nlme_3.1-128           beeswarm_0.2.3        
[16] org.Hs.eg.db_3.4.0     scales_0.4.1           vioplot_0.2           
[19] sm_2.2-5.4             GO.db_3.4.0            GOstats_2.40.0        
[22] graph_1.52.0           Category_2.40.0        Matrix_1.2-7.1        
[25] AnnotationDbi_1.36.2   IRanges_2.8.1          S4Vectors_0.12.1      
[28] Biobase_2.34.0         BiocGenerics_0.20.0    Humanzee_0.1.0        
[31] ggplot2_2.2.1          VennDiagram_1.6.17     futile.logger_1.4.3   
[34] edgeR_3.16.5           limma_3.30.13          dplyr_0.5.0           
[37] testit_0.6             data.table_1.10.4      biomaRt_2.30.0        
[40] workflowr_0.4.0        rmarkdown_1.3         

loaded via a namespace (and not attached):
 [1] splines_3.3.2          assertthat_0.1         RBGL_1.50.0           
 [4] yaml_2.1.14            RSQLite_1.1-2          backports_1.0.5       
 [7] lattice_0.20-34        digest_0.6.12          colorspace_1.3-2      
[10] htmltools_0.3.5        plyr_1.8.4             GSEABase_1.36.0       
[13] XML_3.98-1.5           xtable_1.8-2           git2r_0.18.0          
[16] tibble_1.2             annotate_1.52.1        lazyeval_0.2.0        
[19] magrittr_1.5           memoise_1.0.0          evaluate_0.10         
[22] tools_3.3.2            munsell_0.4.3          locfit_1.5-9.1        
[25] lambda.r_1.1.9         RCurl_1.95-4.8         rstudioapi_0.6        
[28] AnnotationForge_1.16.1 bitops_1.0-6           labeling_0.3          
[31] gtable_0.2.0           DBI_0.6                R6_2.2.0              
[34] knitr_1.15.1           rprojroot_1.2          futile.options_1.0.0  
[37] stringi_1.1.2          Rcpp_0.12.9           </code></pre>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
