<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittlemant" />

<meta name="date" content="2017-02-04" />

<title>QC of single cell libraries</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">QC of single cell libraries</h1>
<h4 class="author"><em>Briana Mittlemant</em></h4>
<h4 class="date"><em>2017-02-04</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-03-06</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> f3c5dc0</p>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;cowplot&quot;)
theme_set(theme_bw(base_size = 16))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank(),
             legend.key = element_blank(),
             plot.title = element_text(size = rel(1)))
source(&quot;../../singleCellSeq/analysis/functions.R&quot;)</code></pre>
<p>Summary counts from featureCounts program using gather-summary-counts.py. These data were colected from the summary files of the combined samples. The code below makes sure all samples are included.</p>
<pre class="r"><code>summary_per_sample &lt;- read.table(&quot;../../singleCellSeq/data/summary-counts.txt&quot;,header=TRUE, stringsAsFactors= FALSE)

stopifnot(summary_per_sample$well != &quot;bulk&quot;,
          sum(summary_per_sample$rmdup == &quot;reads&quot;)==864,
          sum(summary_per_sample$rmdup == &quot;molecules&quot;)==864)</code></pre>
<p>Remove feautureCounts classification with zero counts.</p>
<pre class="r"><code>stopifnot(colSums(summary_per_sample[, c(7, 10:15)]) == 0)
summary_per_sample &lt;- summary_per_sample[, c(-7, -10:-15)]
head(summary_per_sample)</code></pre>
<pre><code>  individual replicate well     rmdup Assigned Unassigned_Ambiguity
1    NA19098        r1  A01 molecules    63322                 1419
2    NA19098        r1  A01     reads  1932782                40278
3    NA19098        r1  A02 molecules    63976                 1454
4    NA19098        r1  A02     reads  2039613                44664
5    NA19098        r1  A03 molecules    43630                  976
6    NA19098        r1  A03     reads  1006487                18865
  Unassigned_NoFeatures Unassigned_Unmapped
1                 54297                   0
2                885075             1093943
3                 49595                   0
4                737166             1140902
5                 33597                   0
6                419226              742614</code></pre>
<p>Input annotation.</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../../singleCellSeq//data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
stopifnot(anno$well != &quot;bulk&quot;, nrow(anno) == 864,
          rep(anno$individual, each = 2) == summary_per_sample$individual,
          rep(anno$replicate, each = 2) == summary_per_sample$replicate,
          rep(anno$well, each = 2) == summary_per_sample$well)
head(anno)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A03 NA19098.r1 NA19098.r1.A03
4    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
5    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
6    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06</code></pre>
<p>Input read counts</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../../singleCellSeq/data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
stopifnot(ncol(reads) == nrow(anno),
          colnames(reads) == anno$sample_id)</code></pre>
<p>Input molecule counts</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../../singleCellSeq//data/molecules.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
stopifnot(ncol(molecules) == nrow(anno),
          colnames(reads) == anno$sample_id)</code></pre>
<p>Input single cell observation quality control data.</p>
<pre class="r"><code>qc &lt;- read.table(&quot;../../singleCellSeq/data/qc-ipsc.txt&quot;, header = TRUE,
                 stringsAsFactors = FALSE)
stopifnot(qc$individual == anno$individual,
          qc$replicate == anno$replicate,
          qc$well == anno$well)
head(qc)</code></pre>
<pre><code>  individual replicate well cell_number concentration tra1.60
1    NA19098        r1  A01           1      1.734785       1
2    NA19098        r1  A02           1      1.723038       1
3    NA19098        r1  A03           1      1.512786       1
4    NA19098        r1  A04           1      1.347492       1
5    NA19098        r1  A05           1      2.313047       1
6    NA19098        r1  A06           1      2.056803       1</code></pre>
<div id="total-ercc-and-removal-of-na19098.r2" class="section level3">
<h3>Total ERCC and removal of NA19098.r2</h3>
<pre class="r"><code>summary_per_sample_reads &lt;- summary_per_sample[summary_per_sample$rmdup == &quot;reads&quot;,]
summary_per_sample_reads$sample_id &lt;- anno$sample_id
summary_per_sample_reads$batch &lt;- anno$batch
stopifnot(colnames(reads) == summary_per_sample_reads$sample_id )

summary_per_sample_reads$ERCC_reads &lt;- apply(reads[grep(&quot;ERCC&quot;, rownames(reads)), ],2,sum)
summary_per_sample_reads$ERCC_molecules &lt;- apply(molecules[grep(&quot;ERCC&quot;, rownames(molecules)), ],2,sum)</code></pre>
<pre class="r"><code>## create a color palette with one color per individual and different shades for repplicates
great_color &lt;- c(&quot;#CC3300&quot;, &quot;#FF9966&quot;, &quot;#FFCC99&quot;, &quot;#006633&quot;, &quot;#009900&quot;, &quot;#99FF99&quot;, &quot;#3366FF&quot;, &quot;#6699FF&quot;, &quot;#66CCFF&quot;)
great_color_8 &lt;- c(&quot;#CC3300&quot;, &quot;#FF9966&quot;, &quot;#006633&quot;, &quot;#009900&quot;, &quot;#99FF99&quot;, &quot;#3366FF&quot;, &quot;#6699FF&quot;, &quot;#66CCFF&quot;)

ercc_reads_plot &lt;- ggplot(summary_per_sample_reads,
                   aes(x = factor(batch), y = ERCC_reads,
                   fill = factor(batch)), height = 600, width = 2000) +
                   geom_violin(alpha = .5) + 
                   geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
                   scale_fill_manual(values = great_color) + 
                   labs(x = &quot;&quot;, y = &quot;Total ERCC read-counts per cell&quot;, 
                        title = &quot;Excess amount of ERCC in NA19098.r2&quot;) +
                   theme(axis.text.x = element_text(hjust=1, angle = 45))

ercc_molecule_plot &lt;- ggplot(summary_per_sample_reads,
                   aes(x = factor(batch), y = ERCC_molecules,
                   fill = factor(batch)), height = 600, width = 2000) +
                   geom_violin(alpha = .5) + 
                   geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
                   scale_fill_manual(values = great_color) + 
                   labs(x = &quot;&quot;, y = &quot;Total ERCC molecule-counts per cell&quot;, 
                        title = &quot;Excess amount of ERCC in NA19098.r2&quot;) +
                   theme(axis.text.x = element_text(hjust=1, angle = 45))

plot_grid(ercc_reads_plot + theme(legend.position=c(.8,.7)),
          ercc_molecule_plot + theme(legend.position = &quot;none&quot;),
          labels = LETTERS[1:2])</code></pre>
<p><img src="figure/QC_single_cell_library.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /> These plots show that in terms of the total read count and molecule count. NA19098.r2 had a large excess of ERCC per cell. We also see a lot more variability in the ERCC read and molecule counts in this sample/replicate.<br />
<strong>Why could this be? </strong><br />
Understand ERCC RNA spike-ins by looking at the thermofisher documentation.</p>
<ul>
<li><p>Purpose: RNA controlls for performance quality assessment (preformulated sets of 92 polyA transcripts from ERCC plasmid reference</p></li>
<li><p>Use: add Spike-in Mix to each sample (about 10^6-fold concentration)</p></li>
<li><p>Known mix1:mix2 ratios can be used to assess differencial gene expression</p></li>
</ul>
<p>Note from paper on this: “It appeared that too much ERCC spike-in mix was added to each cell”</p>
<p>Remove NA19098r2 for all the following analsis:</p>
<pre class="r"><code>remove_19098r2 &lt;- anno$batch != &quot;NA19098.r2&quot;
anno_rm &lt;- anno[remove_19098r2,]
summary_per_sample_reads_rm &lt;- summary_per_sample_reads[remove_19098r2,]
reads_rm &lt;- reads[, remove_19098r2]
molecules_rm &lt;- molecules[, remove_19098r2]

stopifnot(summary_per_sample_reads_rm$sample_id == colnames(reads_rm))</code></pre>
</div>
<div id="total-mapped-reads" class="section level3">
<h3>Total mapped reads</h3>
<pre class="r"><code>## add cell number per well by merging qc file
summary_per_sample_reads_qc &lt;- merge(summary_per_sample_reads_rm,qc,by=c(&quot;individual&quot;,&quot;replicate&quot;,&quot;well&quot;))
## calculate total mapped reads per sample
summary_per_sample_reads_qc$total_mapped &lt;- apply(summary_per_sample_reads_qc[,5:7],1,sum)

## cut off 
cut_off_reads &lt;- quantile(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 0,&quot;total_mapped&quot;], 0.95)

cut_off_reads</code></pre>
<pre><code>    95% 
1556255 </code></pre>
<p>The cut off is the 95% percentile when there are 0 cells in the well. This is used to look for wells that appeared to have a cell but do not produce reads at the level expected for 1 cell. We are loooking for cells that were bad quality and therefore didn’t produce ample reads. 95% cuttoff was used troughtout the study but it is an arbutrary number.</p>
<pre class="r"><code>summary_per_sample_reads_qc$cut_off_reads &lt;- summary_per_sample_reads_qc$total_mapped &gt; cut_off_reads

## numbers of cells 
sum(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1, &quot;total_mapped&quot;] &gt; cut_off_reads)</code></pre>
<pre><code>[1] 603</code></pre>
<p>Total mapped reads need to be higher than the 95% cuttoff found for 0 cell and the well needed to hold only one cell.</p>
<pre class="r"><code>sum(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1, &quot;total_mapped&quot;] &lt;= cut_off_reads)</code></pre>
<pre><code>[1] 96</code></pre>
<p>This is the number of wells that have 1 cell and have less than the 95% cut off for reads. In the data these cells act more like they did not have a cell in the well due to bad quality.</p>
<pre class="r"><code>## density plots
plot_reads &lt;- ggplot(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 0 |
                                           summary_per_sample_reads_qc$cell_number == 1 , ],
       aes(x = total_mapped, fill = as.factor(cell_number))) + 
       geom_density(alpha = 0.5) +
       geom_vline(xintercept = cut_off_reads, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
       labs(x = &quot;Total mapped reads per sample&quot;, title = &quot;Cutoff based on the number of total mapped reads&quot;, fill = &quot;Cell number&quot;)</code></pre>
</div>
<div id="unmapped-ratios" class="section level3">
<h3>Unmapped ratios</h3>
<pre class="r"><code>## calculate unmapped ratios
summary_per_sample_reads_qc$unmapped_ratios &lt;- summary_per_sample_reads_qc[,8]/apply(summary_per_sample_reads_qc[,5:8],1,sum)

## cut off 
cut_off_unmapped &lt;- quantile(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 0,&quot;unmapped_ratios&quot;], 0.05)

cut_off_unmapped</code></pre>
<pre><code>       5% 
0.3640165 </code></pre>
<p>This is similar to the step above. We are looking for the lower 5% of unmapped reads when no cell was present in the well. This is because when no cell is present we will have a lot of unmapped reads that come from the spikeins.</p>
<pre class="r"><code>summary_per_sample_reads_qc$cut_off_unmapped &lt;- summary_per_sample_reads_qc$unmapped_ratios &lt; cut_off_unmapped

## numbers of cells 
sum(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1, &quot;unmapped_ratios&quot;] &gt;= cut_off_unmapped)</code></pre>
<pre><code>[1] 101</code></pre>
<p>Now we look at the samples with 1 cell to see if they have more than the cuttoff of unmapped reads. This is to check mapping quality.</p>
<pre class="r"><code>sum(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1, &quot;unmapped_ratios&quot;] &lt; cut_off_unmapped)</code></pre>
<pre><code>[1] 598</code></pre>
<pre class="r"><code># density plots
plot_unmapped &lt;- ggplot(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 0 |
                                           summary_per_sample_reads_qc$cell_number == 1 , ],
       aes(x = unmapped_ratios *100, fill = as.factor(cell_number))) + 
       geom_density(alpha = 0.5) +
       geom_vline(xintercept = cut_off_unmapped *100, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
       labs(x = &quot;Unmapped reads / Total reads per sample (%)&quot;, title = &quot;Cutoff based on the percentage unmapped reads&quot;)</code></pre>
</div>
<div id="ercc-percentage" class="section level3">
<h3>ERCC Percentage</h3>
<pre class="r"><code>## calculate ercc reads percentage
summary_per_sample_reads_qc$ercc_percentage &lt;- apply(reads_rm[grep(&quot;ERCC&quot;, rownames(reads_rm)), ],2,sum)/apply(summary_per_sample_reads_qc[,5:7],1,sum) 

## cut off 
cut_off_ercc &lt;- quantile(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 0,&quot;ercc_percentage&quot;], 0.05)

cut_off_ercc</code></pre>
<pre><code>       5% 
0.0323008 </code></pre>
<p>This is looking at the percent of reads that map to the ERCC spike ins. We are looking at the lower 5% because higher than that assumes too much of the sample was composed of ERCC mapping reads rather than sample reads.</p>
<pre class="r"><code>summary_per_sample_reads_qc$cut_off_ercc &lt;- summary_per_sample_reads_qc$ercc_percentage &lt; cut_off_ercc

## numbers of cells 
sum(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1, &quot;ercc_percentage&quot;] &gt;= cut_off_ercc)</code></pre>
<pre><code>[1] 90</code></pre>
<pre class="r"><code>sum(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1, &quot;ercc_percentage&quot;] &lt; cut_off_ercc)</code></pre>
<pre><code>[1] 609</code></pre>
<pre class="r"><code>## density plots
plot_ercc &lt;- ggplot(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 0 |
                                           summary_per_sample_reads_qc$cell_number == 1 , ],
       aes(x = ercc_percentage *100, fill = as.factor(cell_number))) + 
       geom_density(alpha = 0.5) +
       geom_vline(xintercept = cut_off_ercc *100, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
       labs(x = &quot;ERCC reads / Total mapped reads per sample&quot;, title = &quot;Cutoff based on the percentage of ERCC reads&quot;)</code></pre>
<pre class="r"><code>## endogenous genes
reads_rm_gene &lt;- reads_rm[grep(&quot;ENSG&quot;, rownames(reads_rm)), ]

## number of genes detected 
summary_per_sample_reads_qc$gene_number &lt;- colSums(reads_rm_gene &gt;= 1)

## cut off 
cut_off_genes &lt;- quantile(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 0,&quot;gene_number&quot;], 0.95)

cut_off_genes</code></pre>
<pre><code>   95% 
6788.9 </code></pre>
<pre class="r"><code>summary_per_sample_reads_qc$cut_off_genes &lt;- summary_per_sample_reads_qc$gene_number &gt; cut_off_genes

## numbers of cells 
sum(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1, &quot;gene_number&quot;] &gt; cut_off_genes)</code></pre>
<pre><code>[1] 629</code></pre>
<pre class="r"><code>sum(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1, &quot;gene_number&quot;] &lt;= cut_off_genes)</code></pre>
<pre><code>[1] 70</code></pre>
<pre class="r"><code>## density plots
plot_gene &lt;- ggplot(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 0 |
                                           summary_per_sample_reads_qc$cell_number == 1 , ],
       aes(x = gene_number, fill = as.factor(cell_number))) + 
       geom_density(alpha = 0.5) +
       geom_vline(xintercept = cut_off_genes, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
       labs(x = &quot;Gene numbers per sample&quot;, title = &quot;Cutoff based on the number of detected genes&quot;)</code></pre>
<pre class="r"><code>plot_grid(plot_reads + theme(legend.position=c(.7,.7)),
          plot_unmapped + theme(legend.position = &quot;none&quot;),
          plot_ercc + theme(legend.position = &quot;none&quot;), 
          plot_gene + theme(legend.position = &quot;none&quot;),
          labels = LETTERS[3:6])</code></pre>
<p><img src="figure/QC_single_cell_library.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>We should look at the numbers that fail in each of these tests. If they are similar we should see if the same cells make up that count in each of the tests. We would expect this if each of these are good tests for cell quality</strong></p>
</div>
<div id="total-molecule-counts" class="section level3">
<h3>Total molecule counts</h3>
<pre class="r"><code>## calculate total gene molecule counts
summary_per_sample_reads_qc$total_gene_molecule &lt;- colSums(molecules_rm[grep(&quot;ENSG&quot;, rownames(molecules_rm)),])

## look for outiers
ggplot(summary_per_sample_reads_qc, aes(x = concentration, y = total_gene_molecule / 10^3,
  color = as.factor(cell_number))) +
  geom_text(aes(label = cell_number)) +
  labs(x = &quot;Concentration&quot;, y = &quot;Gene molecules (thousands)&quot;) +
  scale_color_brewer(palette = &quot;Dark2&quot;) +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/QC_single_cell_library.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>outliers &lt;- summary_per_sample_reads_qc %&gt;% filter(cell_number == 1, concentration &lt; 1.25, concentration &gt; .15,
                                   total_gene_molecule &gt; 100000)
#you could use code like this to look for the cells that failed the qc checks above.

outliers %&gt;% dplyr::select(sample_id)</code></pre>
<pre><code>        sample_id
1  NA19098.r3.B04
2  NA19098.r3.B11
3  NA19101.r1.B10
4  NA19101.r2.D07
5  NA19101.r3.C07
6  NA19101.r3.D08
7  NA19101.r3.F05
8  NA19101.r3.F10
9  NA19239.r2.A12
10 NA19239.r2.B07</code></pre>
</div>
<div id="linear-discrimination-analysis" class="section level3">
<h3>Linear Discrimination analysis</h3>
<p>Linear discrimination analysis:</p>
<ul>
<li><p>releated to ANOVA</p></li>
<li><p>Use pattern recognition and machine learning to find linear combination of features that characterizes or separates two or more classes of objects or events</p></li>
<li><p>Use resulting combination for dimensional reduction</p></li>
<li><p>one dependent variable as a linear combinaton of other features</p></li>
<li><p>also like PCA- linear combination to best explain data</p></li>
</ul>
<pre class="r"><code>library(MASS)</code></pre>
<pre class="r"><code>## create 3 groups according to cell number
group_3 &lt;- rep(&quot;multiple cells&quot;,dim(summary_per_sample_reads_qc)[1])
         group_3[grep(&quot;0&quot;, summary_per_sample_reads_qc$cell_number)] &lt;- &quot;no cells&quot;
         group_3[grep(&quot;1&quot;, summary_per_sample_reads_qc$cell_number)] &lt;- &quot;one cell&quot;
         
         
## create data frame
data_lda &lt;- data.frame(anno_rm,
                       cell_number = summary_per_sample_reads_qc$cell_number,
                       concentration = summary_per_sample_reads_qc$concentration,
                       total_gene_molecule = summary_per_sample_reads_qc$total_gene_molecule,
                       group = group_3)


## remove 19098.r1
data_con &lt;- data_lda %&gt;% filter(batch != &quot;NA19098.r1&quot;)
plot_before &lt;- ggplot(data_con, aes(x = concentration, y = total_gene_molecule / 10^3,
               color = as.factor(group))) +
               geom_text(aes(label = cell_number)) +
               labs(x = &quot;Sample concentration&quot;, 
                    y = &quot;Total gene molecule-count per sample (thousands)&quot;, 
                    title = &quot;Before linear discriminant analysis (LDA) \n A few samples with one cell (purple) \n clustered with samples with multiple cells (green)&quot;) +
               scale_color_brewer(palette = &quot;Dark2&quot;) +
               theme(legend.position = &quot;none&quot;)

## perform lda
data_con_lda &lt;- lda(group ~ concentration + total_gene_molecule,
                    data = data_con) 
#this analysis uses the group we gave and a function that looks at the number of cells we saw, the concentration of the well and the total molecules counts 
data_con_lda_p &lt;- predict(data_con_lda, 
                          newdata = data_con[,c(&quot;concentration&quot;, &quot;total_gene_molecule&quot;)])$class

#predict- predict results of a model fitting from varius model fitting functions 

## determine how well the model fix
table(data_con_lda_p, data_con[, &quot;group&quot;])</code></pre>
<pre><code>                
data_con_lda_p   multiple cells no cells one cell
  multiple cells             34        0       10
  no cells                    0        1        0
  one cell                   15       16      596</code></pre>
<pre class="r"><code>data_con$data_con_lda_p &lt;- data_con_lda_p

plot_after &lt;- ggplot(data_con, aes(x = concentration, y = total_gene_molecule / 10^3,
               color = as.factor(data_con_lda_p))) +
               geom_text(aes(label = cell_number)) +
               labs(x = &quot;Sample concentration&quot;, 
                    y = &quot;Total gene molecule-count per sample (thousands)&quot;, 
                    title = &quot;After linear discriminant analysis (LDA) \n Removal of samples classified as \n samples with multiple cells (green)&quot;) +
               scale_color_brewer(palette = &quot;Dark2&quot;) +
               theme(legend.position = &quot;none&quot;)

## identify the outlier 
outliers_lda &lt;- data_con %&gt;% filter(cell_number == 1, data_con_lda_p == &quot;multiple cells&quot;)
outliers_lda$sample_id</code></pre>
<pre><code> [1] &quot;NA19098.r3.B04&quot; &quot;NA19098.r3.B11&quot; &quot;NA19101.r1.B10&quot; &quot;NA19101.r2.D07&quot;
 [5] &quot;NA19101.r3.C07&quot; &quot;NA19101.r3.D08&quot; &quot;NA19101.r3.F05&quot; &quot;NA19101.r3.F10&quot;
 [9] &quot;NA19239.r2.A12&quot; &quot;NA19239.r2.B07&quot;</code></pre>
<pre class="r"><code>## The lds method identifies outliers
plot_grid(plot_before + theme(legend.position=c(.8,.85)), 
          plot_after + theme(legend.position = &quot;none&quot;),
          labels = LETTERS[1:2])</code></pre>
<p><img src="figure/QC_single_cell_library.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## create filter
summary_per_sample_reads_qc$molecule_outlier &lt;- summary_per_sample_reads_qc$sample_id %in% outliers_lda$sample_id</code></pre>
</div>
<div id="reads-to-molecule-conversion" class="section level3">
<h3>Reads to molecule conversion</h3>
<pre class="r"><code>## calculate convertion
summary_per_sample_reads_qc$ERCC_conversion &lt;- summary_per_sample_reads_qc$ERCC_molecules / summary_per_sample_reads_qc$ERCC_reads

summary_per_sample_reads_qc$conversion &lt;- summary_per_sample_reads_qc$total_gene_molecule /  colSums(reads_rm[grep(&quot;ENSG&quot;, rownames(reads_rm)),])

ggplot(summary_per_sample_reads_qc, aes(x = ERCC_conversion, y = conversion,
  color = as.factor(cell_number))) +
  geom_text(aes(label = cell_number)) +
  labs(x = &quot;Conversion of ERCC&quot;, y = &quot;Conversion of genes&quot;) +
  scale_color_brewer(palette = &quot;Dark2&quot;) +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/QC_single_cell_library.Rmd/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>out_ercc_con &lt;- summary_per_sample_reads_qc %&gt;% filter(cell_number == &quot;1&quot;, ERCC_conversion &gt; .094)

## try lda
data_lda$conversion &lt;- summary_per_sample_reads_qc$conversion
data_lda$ERCC_conversion &lt;- summary_per_sample_reads_qc$ERCC_conversion

data_ercc_lda &lt;- lda(group ~ ERCC_conversion + conversion,
                    data = data_lda)
data_ercc_lda_p &lt;- predict(data_ercc_lda, 
                          newdata = data_lda[,c(&quot;ERCC_conversion&quot;, &quot;conversion&quot;)])$class
table(data_con_lda_p, data_con[, &quot;group&quot;])</code></pre>
<pre><code>                
data_con_lda_p   multiple cells no cells one cell
  multiple cells             34        0       10
  no cells                    0        1        0
  one cell                   15       16      596</code></pre>
<pre class="r"><code>data_lda$data_ercc_lda_p &lt;- data_ercc_lda_p

## identify the outlier 
outliers_ercc &lt;- data_lda %&gt;% filter(cell_number == 1, data_ercc_lda_p == &quot;multiple cells&quot;)
outliers_ercc$sample_id</code></pre>
<pre><code> [1] &quot;NA19098.r1.F01&quot; &quot;NA19098.r3.B04&quot; &quot;NA19098.r3.B11&quot; &quot;NA19101.r2.D07&quot;
 [5] &quot;NA19101.r3.C07&quot; &quot;NA19101.r3.D08&quot; &quot;NA19101.r3.F05&quot; &quot;NA19101.r3.F10&quot;
 [9] &quot;NA19239.r2.A12&quot; &quot;NA19239.r2.B07&quot; &quot;NA19239.r3.G02&quot;</code></pre>
<p>This gives the samples that look like they had one cell but act like they had more than one in the LDA analysis of ercc conversion.</p>
<pre class="r"><code>## cutoff
out_ercc_con &lt;- summary_per_sample_reads_qc %&gt;% filter(cell_number == &quot;1&quot;, ERCC_conversion &gt; .08)

## create filter
summary_per_sample_reads_qc$conversion_outlier &lt;- summary_per_sample_reads_qc$sample_id %in% outliers_ercc$sample_id

plot_ercc_before &lt;- ggplot(data_lda, aes(x = ERCC_conversion, y = conversion,
               color = as.factor(group))) +
               geom_text(aes(label = cell_number)) +
               labs(x = &quot;Read-to-molecule conversion of ERCC&quot;, 
                    y = &quot;Read-to-molecule conversion of genes&quot;, 
                    title = &quot;Before linear discriminant analysis (LDA) \n A few samples with one cell (purple) \n clustered with samples with multiple cells (green)&quot;) +
               scale_color_brewer(palette = &quot;Dark2&quot;) +
               theme(legend.position = &quot;none&quot;)

plot_ercc_after &lt;- ggplot(data_lda, aes(x = ERCC_conversion, y = conversion,
               color = as.factor(data_ercc_lda_p))) +
               geom_text(aes(label = cell_number)) +
               labs(x = &quot;Read-to-molecule conversion of ERCC&quot;, 
                    y = &quot;Read-to-molecule conversion of genes&quot;, 
                    title = &quot;After linear discriminant analysis (LDA) \n Removal of samples classified as \n samples with multiple cells (green)&quot;) +
               scale_color_brewer(palette = &quot;Dark2&quot;) +
               theme(legend.position = &quot;none&quot;)

plot_grid(plot_ercc_before, 
          plot_ercc_after,
          labels = LETTERS[3:4])</code></pre>
<p><img src="figure/QC_single_cell_library.Rmd/unnamed-chunk-38-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="mitochondrial-genes" class="section level3">
<h3>Mitochondrial genes</h3>
<pre class="r"><code>## create a list of mitochondrial genes (13 protein-coding genes)
## MT-ATP6, MT-CYB, MT-ND1, MT-ND4, MT-ND4L, MT-ND5, MT-ND6, MT-CO2, MT-CO1, MT-ND2, MT-ATP8, MT-CO3, MT-ND3
mtgene &lt;- c(&quot;ENSG00000198899&quot;, &quot;ENSG00000198727&quot;, &quot;ENSG00000198888&quot;, &quot;ENSG00000198886&quot;, &quot;ENSG00000212907&quot;, &quot;ENSG00000198786&quot;, &quot;ENSG00000198695&quot;, &quot;ENSG00000198712&quot;, &quot;ENSG00000198804&quot;, &quot;ENSG00000198763&quot;,&quot;ENSG00000228253&quot;, &quot;ENSG00000198938&quot;, &quot;ENSG00000198840&quot;)

## reads of mt genes in single cells
mt_reads &lt;- reads_rm_gene[mtgene,]
dim(mt_reads)</code></pre>
<pre><code>[1]  13 768</code></pre>
<pre class="r"><code>stopifnot(colnames(reads_rm) == rownames(summary_per_sample_reads_qc$sample_id))

## mt ratio of single cell
summary_per_sample_reads_qc$mt_reads &lt;- apply(mt_reads, 2, sum)
summary_per_sample_reads_qc$mt_reads_ratio &lt;- summary_per_sample_reads_qc$mt_reads /summary_per_sample_reads_qc$total_mapped

## vs. number of genes detected
ggplot(summary_per_sample_reads_qc,
       aes(x = gene_number, y = mt_reads_ratio, 
       color = as.factor(cell_number))) +
       geom_text(aes(label = cell_number)) +
       labs(x = &quot;Number of genes&quot;, y = &quot;Mitochondrial ratio&quot;) +
       scale_color_brewer(palette = &quot;Dark2&quot;) +
       theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/QC_single_cell_library.Rmd/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="filter" class="section level3">
<h3>Filter</h3>
<div id="final-list" class="section level4">
<h4>Final list</h4>
<pre class="r"><code>## all filter
summary_per_sample_reads_qc$filter_all &lt;- summary_per_sample_reads_qc$cell_number == 1 &amp;
                                                 summary_per_sample_reads_qc$cut_off_reads &amp;
                                                 summary_per_sample_reads_qc$cut_off_unmapped &amp;
                                                 summary_per_sample_reads_qc$cut_off_ercc &amp;
                                                 summary_per_sample_reads_qc$cut_off_genes &amp;
                                                 summary_per_sample_reads_qc$molecule_outlier == &quot;FALSE&quot; &amp;
                                                 summary_per_sample_reads_qc$conversion_outlier == &quot;FALSE&quot;
                                                 
table(summary_per_sample_reads_qc[summary_per_sample_reads_qc$filter_all,
                           c(&quot;individual&quot;, &quot;replicate&quot;)])    </code></pre>
<pre><code>          replicate
individual r1 r2 r3
   NA19098 85  0 57
   NA19101 80 70 51
   NA19239 74 68 79</code></pre>
<p>This uses the information we found above to filter cells out of the analysis. The list created is the number of wells that passes all QC filters.</p>
<pre class="r"><code>stopifnot(nrow(summary_per_sample_reads_qc) == nrow(anno_rm))
quality_single_cells &lt;- anno_rm %&gt;%
  filter(summary_per_sample_reads_qc$filter_all) %&gt;%
  dplyr :: select(sample_id)

write.table(quality_single_cells,
            file = &quot;../../cell-cycle/data/quality-single-cells.txt&quot;, quote = FALSE,
            sep = &quot;\t&quot;, row.names = FALSE, col.names = FALSE)</code></pre>
<p>I wrote the quality single cells file to the data directory in my cell cycle git hub project.</p>
</div>
</div>
<div id="mito-ratios" class="section level3">
<h3>Mito Ratios</h3>
<pre class="r"><code>ggplot(summary_per_sample_reads_qc,
       aes(x = gene_number, y = mt_reads_ratio, 
       color = as.factor(filter_all))) +
       geom_text(aes(label = cell_number)) +
       labs(x = &quot;Number of genes&quot;, y = &quot;Mitochondrial ratio&quot;) +
       theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/QC_single_cell_library.Rmd/unnamed-chunk-43-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1,],
       aes(x = factor(filter_all), y = mt_reads_ratio,
           fill = factor(filter_all)), height = 600, width = 2000) +
geom_violin(alpha = .5) + 
geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
labs(x = &quot;Quality&quot;, y = &quot;Mitochonrial ratio&quot;, title = &quot;Mitochondrial ratio of libraries with 1 cell&quot;)</code></pre>
<p><img src="figure/QC_single_cell_library.Rmd/unnamed-chunk-44-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Filter describes weather the well made ti past the above QC filter step.</p>
<pre class="r"><code>## check the batch of those outliers
mito_outliers &lt;- summary_per_sample_reads_qc %&gt;% filter(filter_all == &quot;TRUE&quot;, mt_reads_ratio &gt; .15)
mito_outliers  %&gt;% dplyr::select(sample_id, mt_reads_ratio)</code></pre>
<pre><code>       sample_id mt_reads_ratio
1 NA19098.r1.D07      0.1573689
2 NA19098.r3.G03      0.1618823
3 NA19098.r3.G04      0.1508871
4 NA19101.r3.A09      0.1718202</code></pre>
<p>These are the top outliers in the true graph in terms of mitochondrial read ratios. 3 of the4 come from individual 19098so we will look further into this.</p>
<pre class="r"><code>## check if 19098 have high mt genes
ggplot(summary_per_sample_reads_qc[summary_per_sample_reads_qc$filter_all == &quot;TRUE&quot;,],
                   aes(x = factor(batch), y = mt_reads_ratio,
                   fill = factor(batch)), height = 600, width = 2000) +
                   geom_violin(alpha = .5) + 
                   geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
                   scale_fill_manual(values = great_color_8) +  
                   labs(x = &quot;batch&quot;, y = &quot;Mitochondrial ratio&quot;) +
                   theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<p><img src="figure/QC_single_cell_library.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The plots show that at an individual scale we dont see an increase in the mitochondrial gene ratio for 19098.</p>
<p><strong>Do we have an expected mitochondrial gene ratio for the data?</strong></p>
</div>
<div id="plots" class="section level3">
<h3>Plots</h3>
<p>This shows all of the QC plots that were created in this analysis file.</p>
<pre class="r"><code>genes_unmapped &lt;-  ggplot(summary_per_sample_reads_qc,
                   aes(x = gene_number, y = unmapped_ratios * 100,
                   col = as.factor(individual), height = 600, width = 2000)) +
                   geom_point(size = 3, alpha = 0.3) + 
                   geom_vline(xintercept = cut_off_genes, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
                   geom_hline(yintercept = cut_off_unmapped * 100, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
                   labs(x = &quot;Number of detected genes / sample&quot;, 
                        y = &quot;Percentage of unmapped reads in each sample (%)&quot;,
                        title = &quot;Number of genes detected as \n expressed as a function of \n the proportion of mapped reads&quot;) 

genes_ercc &lt;-  ggplot(summary_per_sample_reads_qc,
                   aes(x = gene_number, y = ercc_percentage * 100,
                   col = as.factor(individual), shape = as.factor(replicate), height = 600, width = 2000)) +
                   geom_point(size = 3, alpha = 0.3) +
                   geom_vline(xintercept = cut_off_genes, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
                   geom_hline(yintercept = cut_off_ercc * 100, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
                   labs(x = &quot;Number of detected genes / sample&quot;, 
                        y = &quot;Percentage of ERCC reads in each sample (%)&quot;,
                        title = &quot;Samples with zero cell \n (visual inspection) have \n low number of reads&quot;) 

reads_unmapped_num &lt;-  ggplot(summary_per_sample_reads_qc,
                   aes(x = total_mapped, y = unmapped_ratios * 100,
                   col = as.factor(individual), label = as.character(cell_number), height = 600, width = 2000)) +
                   geom_text(fontface = 3, alpha = 0.5) + 
                   geom_vline(xintercept = cut_off_reads, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
                   geom_hline(yintercept = cut_off_unmapped * 100, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
                   labs(x = &quot;Total mapped reads / sample&quot;, 
                        y = &quot;Percentage of unmapped reads in each sample (%)&quot;,
                        title = &quot;Samples with zero cells \n (visual inspection) have \n high proportion of unmapped reads&quot;) 

reads_ercc_num &lt;-  ggplot(summary_per_sample_reads_qc,
                   aes(x = total_mapped, y = ercc_percentage * 100,
                   col = as.factor(individual), label = as.character(cell_number), height = 600, width = 2000)) +
                   geom_text(fontface = 3, alpha = 0.5) + 
                   geom_vline(xintercept = cut_off_reads, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
                   geom_hline(yintercept = cut_off_ercc * 100, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
                   labs(x = &quot;Total mapped reads / sample&quot;,
                        y = &quot;Percentage of ERCC reads in each sample (%)&quot;,
                        title = &quot;Samples with zero cells \n (visual inspection) have \n high proportion of ERCC reads&quot;) 

plot_grid(genes_unmapped + theme(legend.position = &quot;none&quot;), 
          genes_ercc + theme(legend.position = &quot;none&quot;),
          reads_unmapped_num + theme(legend.position = &quot;none&quot;), 
          reads_ercc_num + theme(legend.position = &quot;none&quot;),
          labels = letters[1:4])</code></pre>
<p><img src="figure/QC_single_cell_library.Rmd/unnamed-chunk-47-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_grid(genes_unmapped + theme(legend.position = c(.75,.9)) + labs(col = &quot;Individual&quot;), 
          reads_unmapped_num + theme(legend.position = &quot;none&quot;), 
          reads_ercc_num + theme(legend.position = &quot;none&quot;),
          labels = letters[3:5],
          nrow = 1)</code></pre>
<p><img src="figure/QC_single_cell_library.Rmd/unnamed-chunk-48-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_grid(ercc_reads_plot + theme(legend.position = &quot;none&quot;),
          ercc_molecule_plot + theme(legend.position = &quot;none&quot;),
          plot_reads + theme(legend.position=c(.8,.85)),
          plot_unmapped + theme(legend.position = &quot;none&quot;),
          plot_ercc + theme(legend.position = &quot;none&quot;), 
          plot_gene + theme(legend.position = &quot;none&quot;),
          labels = letters[1:6],
          ncol = 2)</code></pre>
<p><img src="figure/QC_single_cell_library.Rmd/unnamed-chunk-49-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_grid(plot_before + theme(legend.position=c(.85,.85)) + labs(col = &quot;Cell number&quot;), 
          plot_after + theme(legend.position = &quot;none&quot;),
          plot_ercc_before, 
          plot_ercc_after,
          labels = letters[1:4])</code></pre>
<p><img src="figure/QC_single_cell_library.Rmd/unnamed-chunk-50-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="review" class="section level3">
<h3>Review:</h3>
<p>In this analysis we performed QC by checking the following criteria:</p>
<ul>
<li><p>Total reads in 0 cell wells</p></li>
<li><p>Proportion of mapped reads</p></li>
<li><p>ERCC read proportions</p></li>
<li><p>Number of genes</p></li>
<li><p>Total ERCC read counts</p></li>
<li><p>LDA analysis for multiple cells in one well</p></li>
<li><p>Conversion ratios for genes and ERCCs</p></li>
<li><p>Mitochondrial read ratios</p></li>
</ul>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] mygene_1.10.0          GenomicFeatures_1.26.3 GenomicRanges_1.26.3  
 [4] GenomeInfoDb_1.10.3    broman_0.65-1          gplots_3.0.1          
 [7] matrixStats_0.51.0     MASS_7.3-45            gridExtra_2.2.1       
[10] lme4_1.1-12            workflowr_0.4.0        org.Hs.eg.db_3.4.0    
[13] GOstats_2.40.0         graph_1.52.0           Category_2.40.0       
[16] Matrix_1.2-7.1         GO.db_3.4.0            AnnotationDbi_1.36.2  
[19] IRanges_2.8.1          S4Vectors_0.12.1       Biobase_2.34.0        
[22] BiocGenerics_0.20.0    Humanzee_0.1.0         sva_3.22.0            
[25] genefilter_1.56.0      mgcv_1.8-15            nlme_3.1-128          
[28] BiocInstaller_1.24.0   scales_0.4.1           VennDiagram_1.6.17    
[31] futile.logger_1.4.3    edgeR_3.16.5           limma_3.30.11         
[34] dplyr_0.5.0            testit_0.6             data.table_1.10.4     
[37] biomaRt_2.30.0         cowplot_0.7.0          stringr_1.2.0         
[40] reshape2_1.4.2         rmarkdown_1.3          Citrus_0.99           
[43] easyGgplot2_1.0.0.9000 ggplot2_2.2.1         

loaded via a namespace (and not attached):
 [1] minqa_1.2.4                colorspace_1.3-2          
 [3] rprojroot_1.2              htmlTable_1.9             
 [5] XVector_0.14.0             base64enc_0.1-3           
 [7] rstudioapi_0.6             sqldf_0.4-10              
 [9] splines_3.3.2              knitr_1.15.1              
[11] Formula_1.2-1              jsonlite_1.3              
[13] nloptr_1.0.4               Rsamtools_1.26.1          
[15] annotate_1.52.1            cluster_2.0.5             
[17] httr_1.2.1                 backports_1.0.5           
[19] assertthat_0.1             lazyeval_0.2.0            
[21] acepack_1.4.1              htmltools_0.3.5           
[23] tools_3.3.2                gtable_0.2.0              
[25] Rcpp_0.12.9                Biostrings_2.42.1         
[27] gdata_2.17.0               rtracklayer_1.34.2        
[29] proto_1.0.0                gtools_3.5.0              
[31] devtools_1.12.0            statmod_1.4.29            
[33] XML_3.98-1.5               zlibbioc_1.20.0           
[35] SummarizedExperiment_1.4.0 RBGL_1.50.0               
[37] lambda.r_1.1.9             RColorBrewer_1.1-2        
[39] yaml_2.1.14                curl_2.3                  
[41] memoise_1.0.0              rpart_4.1-10              
[43] latticeExtra_0.6-28        stringi_1.1.2             
[45] RSQLite_1.1-2              checkmate_1.8.2           
[47] caTools_1.17.1             BiocParallel_1.8.1        
[49] chron_2.3-50               bitops_1.0-6              
[51] evaluate_0.10              lattice_0.20-34           
[53] htmlwidgets_0.8            GenomicAlignments_1.10.0  
[55] labeling_0.3               GSEABase_1.36.0           
[57] AnnotationForge_1.16.1     plyr_1.8.4                
[59] magrittr_1.5               R6_2.2.0                  
[61] Hmisc_4.0-2                DBI_0.5-1                 
[63] gsubfn_0.6-6               foreign_0.8-67            
[65] withr_1.0.2                survival_2.40-1           
[67] RCurl_1.95-4.8             nnet_7.3-12               
[69] tibble_1.2                 futile.options_1.0.0      
[71] KernSmooth_2.23-15         locfit_1.5-9.1            
[73] git2r_0.18.0               digest_0.6.12             
[75] xtable_1.8-2               munsell_0.4.3             </code></pre>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
