<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>Quality control of UMIs and ERCC spike-ins</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Quality control of UMIs and ERCC spike-ins</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2/6/2017</em></h4>

</div>


<p><strong>Last updated:</strong> 2017-02-08</p>
<p><strong>Code version:</strong> 488eb05</p>
<div id="input" class="section level3">
<h3>Input</h3>
<pre class="r"><code>library(&quot;biomaRt&quot;)
library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)
library(&quot;limma&quot;)
library(&quot;ggplot2&quot;)
library(&quot;grid&quot;)
library(&quot;cowplot&quot;)
theme_set(theme_bw(base_size = 16))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank(),
             legend.key = element_blank(),
             plot.title = element_text(size = rel(1)))
source(&quot;../../singleCellSeq/analysis/functions.R&quot;)</code></pre>
<p>Input ERCC molecule counts calculated in capture efficiency.</p>
<pre class="r"><code>ercc &lt;- read.table(&quot;../../singleCellSeq/data/expected-ercc-molecules.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(ercc)</code></pre>
<pre><code>          id conc_mix1 ercc_molecules_well
1 ERCC-00130 30000.000          4877.93485
2 ERCC-00004  7500.000          1219.48371
3 ERCC-00136  1875.000           304.87093
4 ERCC-00108   937.500           152.43546
5 ERCC-00116   468.750            76.21773
6 ERCC-00092   234.375            38.10887</code></pre>
<p>Input annotation.</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../../singleCellSeq/data/annotation.txt&quot;, header=TRUE, stringsAsFactors = FALSE)
head(anno)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A03 NA19098.r1 NA19098.r1.A03
4    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
5    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
6    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06</code></pre>
<p>Input read counts and filter for quality cells.</p>
<pre class="r"><code>reads&lt;- read.table(&quot;../../singleCellSeq/data/reads.txt&quot;, header=TRUE, stringsAsFactors = FALSE)
quality_single_cells &lt;- scan(&quot;../data/quality-single-cells.txt&quot;, what = &quot;character&quot;)
reads&lt;- reads[, colnames(reads) %in% quality_single_cells]</code></pre>
<p>Input read counts in high quality cells for filtered genes.</p>
<pre class="r"><code>reads_filter &lt;- read.table(&quot;../data/reads-filter.txt&quot;, header=TRUE, stringsAsFactors = FALSE)</code></pre>
<p>Input molecule counts and filter for quality cells.</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../../singleCellSeq/data/molecules.txt&quot;, header=TRUE, stringsAsFactors = FALSE)
molecules &lt;- molecules[,colnames(molecules) %in% quality_single_cells]</code></pre>
<p>Input molecule counts in high quality cells for filtered genes.</p>
<pre class="r"><code>molecules_filter &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header = TRUE, stringsAsFactors = FALSE)</code></pre>
</div>
<div id="compare-reads-and-molecules" class="section level3">
<h3>Compare reads and molecules</h3>
<pre class="r"><code>## calculate mean accross all cells for each gene 
#in apply- 1=rows, 2=columns
reads_mean     &lt;- apply(reads, 1, mean)
molecules_mean &lt;- apply(molecules, 1, mean)
distribution &lt;- data.frame(reads_mean, molecules_mean)

reads_filter_mean &lt;- apply(reads_filter, 1, mean)
molecules_filter_mean &lt;- apply(molecules_filter, 1, mean)
distribution_filter &lt;- data.frame(reads_filter_mean, molecules_filter_mean)

## correlation between reads and molecules
cor(distribution)</code></pre>
<pre><code>               reads_mean molecules_mean
reads_mean      1.0000000      0.9277703
molecules_mean  0.9277703      1.0000000</code></pre>
<pre class="r"><code>cor(distribution_filter)</code></pre>
<pre><code>                      reads_filter_mean molecules_filter_mean
reads_filter_mean             1.0000000             0.9408234
molecules_filter_mean         0.9408234             1.0000000</code></pre>
<p>There is a higher correlation between molecules and read counts in the filtered data. I would expect a lower correlation if the reads come from sequencing each molecule multiple times. This could be driven by the large numbers of zeros in the data set.</p>
<pre class="r"><code>##select ERCC
#distribution has the mean read and molecule count for each gene
distribution$type &lt;- ifelse(grepl(&quot;ERCC&quot;, rownames(distribution)), &quot;ERCC&quot;, &quot;gene&quot;)
distribution_filter$type &lt;- ifelse(grepl(&quot;ERCC&quot;, rownames(distribution_filter)), &quot;ERCC&quot;, &quot;gene&quot;)

#color palete
cbPalette &lt;- c(&quot;#0000FF&quot;, &quot;#999999&quot;, &quot;#990033&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;, &quot;#009E73&quot;)

## plots
plot_mean_log &lt;- ggplot(distribution, aes(x = molecules_mean, y = reads_mean, col = type)) + 
  geom_point(size = 3, alpha = 0.5) +  
  scale_colour_manual(values=cbPalette) + 
  labs(x = &quot;Mean molecule-counts across samples&quot;, 
       y = &quot;Mean read-counts across samples&quot;, 
       title = &quot;Amplification bias is stronger \n in endogenous genes than in ERCC spike-ins&quot;) + 
  scale_x_log10() + 
  scale_y_log10()

plot_mean_log</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<p><img src="figure/quality_control_plotes.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /> Amplification bias is more reads per molecule. <strong>Why do we think the bias is stronger in endogenous gene?</strong></p>
<pre class="r"><code>plot_mean_filter_log &lt;- ggplot(distribution_filter, aes(x = molecules_filter_mean, y = reads_filter_mean, col = type)) + 
  geom_point(size = 3, alpha = 0.5) +  
  scale_colour_manual(values=cbPalette) + 
  labs(x = &quot;Mean molecule-counts across samples&quot;, 
       y = &quot;Mean read-counts&quot;, 
       title = &quot;Read-counts and molecule-counts of each gene (log scale)&quot;) + 
  scale_x_log10() + 
  scale_y_log10()

plot_mean_filter_log</code></pre>
<p><img src="figure/quality_control_plotes.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_mean &lt;- ggplot(distribution, aes(x = molecules_mean, y = reads_mean, col = type)) + 
  geom_point(size = 3, alpha = 0.5) +  
  scale_colour_manual(values=cbPalette) + 
  labs(x = &quot;Mean molecule-counts across samples&quot;, 
       y = &quot;Mean read-counts across samples&quot;, 
       title = &quot;Amplification bias is stronger \n in endogenous genes than in ERCC spike-ins&quot;) +
  geom_smooth(method = &quot;lm&quot;)

plot_mean</code></pre>
<p><img src="figure/quality_control_plotes.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>I can see the bias in this plot because the read-moleculre count is higher in endogenous genes.</p>
</div>
<div id="distribution-of-fold-change-to-mean" class="section level3">
<h3>Distribution of fold change to mean</h3>
<p>Look at the distribution of fold change to mean. As being reported by others, the lowly expressed genes show divergent read and molecule counts</p>
<pre class="r"><code>## calculate fold change to mean
distribution$fold_change_read     &lt;- log2(reads_mean/mean(reads_mean))
distribution$fold_change_molecule &lt;- log2(molecules_mean/mean(molecules_mean))

plot_distribution &lt;- ggplot(distribution, aes(x = fold_change_molecule, y = fold_change_read, col = type)) + 
  geom_point(size = 3, alpha = 0.5) + 
  scale_colour_manual(values=cbPalette) + 
  stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;) + 
  labs(x = &quot;log2 fold change to mean (molecule)&quot;, y =  &quot;log2 fold change to mean (reads)&quot;)

plot_distribution</code></pre>
<p><img src="figure/quality_control_plotes.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /> In all of these plots each dot represents one gene in the dataset.</p>
</div>
<div id="vizualising-capture-efficiency" class="section level3">
<h3>Vizualising capture efficiency</h3>
<p>Use only those 50 ERCC genes with at least 1 expected molecule per well.</p>
<pre class="r"><code>ercc_list &lt;- list()
for (spike in ercc$id[ercc$ercc_molecules_well &gt;= 1]) {
  if (spike %in% rownames(molecules)) {
    ercc_list$id &lt;- c(ercc_list$id, spike)
    ercc_list$observed_mean &lt;- c(ercc_list$observed_mean,
                                 mean(as.numeric(molecules[spike, ])))
    ercc_list$observed_sem &lt;- c(ercc_list$observed_sem,
                                 sd(as.numeric(molecules[spike, ])) /
                                 sqrt(ncol(molecules)))
    ercc_list$expected &lt;- c(ercc_list$expected,
                            ercc$ercc_molecules_well[ercc$id == spike])
  }
}
ercc_plot &lt;- as.data.frame(ercc_list, stringsAsFactors = FALSE)
str(ercc_plot)</code></pre>
<pre><code>&#39;data.frame&#39;:   50 obs. of  4 variables:
 $ id           : chr  &quot;ERCC-00130&quot; &quot;ERCC-00004&quot; &quot;ERCC-00136&quot; &quot;ERCC-00108&quot; ...
 $ observed_mean: num  298.1879 57.2784 20.1933 13.5833 0.0603 ...
 $ observed_sem : num  2.8448 0.6346 0.2841 0.2141 0.0106 ...
 $ expected     : num  4877.9 1219.5 304.9 152.4 76.2 ...</code></pre>
<pre class="r"><code>cor(ercc_plot$observed_mean, ercc_plot$expected)</code></pre>
<pre><code>[1] 0.9916157</code></pre>
<pre class="r"><code>ercc_list_filter &lt;- list()
for (spike in ercc$id[ercc$ercc_molecules_well &gt;= 0]) {
  if (spike %in% rownames(molecules_filter)) {
    ercc_list_filter$id &lt;- c(ercc_list_filter$id, spike)
    ercc_list_filter$observed_mean &lt;- c(ercc_list_filter$observed_mean,
                                 mean(as.numeric(molecules_filter[spike, ])))
    ercc_list_filter$observed_sem &lt;- c(ercc_list_filter$observed_sem,
                                 sd(as.numeric(molecules_filter[spike, ])) /
                                 sqrt(ncol(molecules_filter)))
    ercc_list_filter$expected &lt;- c(ercc_list_filter$expected,
                            ercc$ercc_molecules_well[ercc$id == spike])
  }
}
ercc_filter_plot &lt;- as.data.frame(ercc_list_filter, stringsAsFactors = FALSE)
str(ercc_filter_plot)</code></pre>
<pre><code>&#39;data.frame&#39;:   48 obs. of  4 variables:
 $ id           : chr  &quot;ERCC-00130&quot; &quot;ERCC-00004&quot; &quot;ERCC-00136&quot; &quot;ERCC-00108&quot; ...
 $ observed_mean: num  298.19 57.28 20.19 13.58 2.15 ...
 $ observed_sem : num  2.8448 0.6346 0.2841 0.2141 0.0674 ...
 $ expected     : num  4877.9 1219.5 304.9 152.4 38.1 ...</code></pre>
<pre class="r"><code>cor(ercc_filter_plot$observed_mean, ercc_filter_plot$expected)</code></pre>
<pre><code>[1] 0.9916498</code></pre>
<pre class="r"><code>p_efficiency &lt;- ggplot(ercc_plot, aes(x = expected, y = observed_mean, label = id)) +
  geom_point(col = &quot;red&quot;) +
  geom_errorbar(aes(ymin = observed_mean - observed_sem,
                    ymax = observed_mean + observed_sem), width = 0) +
  labs(x = &quot;Expected ERCC molecules&quot;,
       y = &quot;Observed ERCC molecules\n(mean +/- SEM)&quot;,
       title = &quot;ERCC capture efficiency&quot;)

p_efficiency_plot &lt;- p_efficiency + scale_x_log10() +
  scale_y_log10() +
  labs(x = &quot;Expected ERCC molecules&quot;,
       y = &quot;Observed ERCC molecules\n(mean +/- SEM)&quot;,
       title = &quot;ERCC capture efficiency \n (48 ERCC spike-in controls)&quot;)

p_efficiency_plot + geom_text(hjust = 0, nudge_x = 0.05, size = 4)</code></pre>
<p><img src="figure/quality_control_plotes.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p_efficiency_filter_plot &lt;- ggplot(ercc_filter_plot, aes(x = expected, y = observed_mean)) +
  geom_point(col = &quot;red&quot;) +
  geom_errorbar(aes(ymin = observed_mean - observed_sem,
                    ymax = observed_mean + observed_sem), width = 0) +
                    scale_x_log10() + scale_y_log10() +
                    labs(x = &quot;Expected ERCC molecules&quot;,
                    y = &quot;Observed ERCC molecules\n(mean +/- SEM)&quot;,
                    title = &quot;ERCC capture efficiency&quot;)

p_efficiency_filter_plot</code></pre>
<p><img src="figure/quality_control_plotes.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="calculate-capture-efficiency-per-cell" class="section level3">
<h3>Calculate Capture efficiency per cell</h3>
<pre class="r"><code>ercc_index &lt;- grep(&quot;ERCC&quot;, rownames(molecules_filter))
length(ercc_index)</code></pre>
<pre><code>[1] 48</code></pre>
<p>48 of the ERCC genes pass the filtering.</p>
<pre class="r"><code>efficiency &lt;- numeric(length = ncol(molecules_filter))
total_ercc_molecules &lt;- sum(ercc_filter_plot$expected)
for (i in 1:ncol(molecules_filter)) {
  efficiency[i] &lt;- sum(molecules_filter[ercc_index, i]) / total_ercc_molecules
}
summary(efficiency)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.03978 0.05459 0.06140 0.06580 0.07621 0.12150 </code></pre>
<p>Using ERCC expected values we calculate an efficienct and show a mean percentage as 6.5%. This seems pretty low!</p>
</div>
<div id="qc-plots-for-paper" class="section level3">
<h3>QC plots for paper:</h3>
<pre class="r"><code>plot_grid(plot_mean_log + theme(legend.position = c(.85,.25)) + labs (col = &quot;&quot;),
          p_efficiency_plot + theme(legend.position = &quot;none&quot;),
          labels = letters[6:7])</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<p><img src="figure/quality_control_plotes.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level3">
<h3>Session Information:</h3>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] gridExtra_2.2.1      lme4_1.1-12          Matrix_1.2-7.1      
 [4] scales_0.4.1         biomaRt_2.30.0       testit_0.6          
 [7] MASS_7.3-45          cowplot_0.7.0        ggplot2_2.2.1       
[10] edgeR_3.16.5         limma_3.30.9         BiocInstaller_1.24.0
[13] VennDiagram_1.6.17   futile.logger_1.4.3  dplyr_0.5.0         
[16] data.table_1.10.0    devtools_1.12.0      workflowr_0.3.0     
[19] rmarkdown_1.3       

loaded via a namespace (and not attached):
 [1] locfit_1.5-9.1       splines_3.3.2        lattice_0.20-34     
 [4] colorspace_1.3-2     htmltools_0.3.5      stats4_3.3.2        
 [7] yaml_2.1.14          XML_3.98-1.5         nloptr_1.0.4        
[10] withr_1.0.2          DBI_0.5-1            BiocGenerics_0.20.0 
[13] lambda.r_1.1.9       plyr_1.8.4           stringr_1.1.0       
[16] munsell_0.4.3        gtable_0.2.0         memoise_1.0.0       
[19] evaluate_0.10        labeling_0.3         Biobase_2.34.0      
[22] knitr_1.15.1         IRanges_2.8.1        parallel_3.3.2      
[25] AnnotationDbi_1.36.1 Rcpp_0.12.9          backports_1.0.5     
[28] S4Vectors_0.12.1     digest_0.6.12        stringi_1.1.2       
[31] rprojroot_1.2        tools_3.3.2          bitops_1.0-6        
[34] magrittr_1.5         lazyeval_0.2.0       RCurl_1.95-4.8      
[37] tibble_1.2           RSQLite_1.1-2        futile.options_1.0.0
[40] assertthat_0.1       minqa_1.2.4          rstudioapi_0.6      
[43] R6_2.2.0             nlme_3.1-128         git2r_0.18.0        </code></pre>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
