<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Brana Mittleman" />

<meta name="date" content="2017-01-11" />

<title>Recreate Gilad lab single cell</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Recreate Gilad lab single cell</h1>
<h4 class="author"><em>Brana Mittleman</em></h4>
<h4 class="date"><em>January 11, 2017</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-03-06</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> f3c5dc0</p>
<p>I will use this to document my recreating of the single cell data processing from <a href="http://jdblischak.github.io/singleCellSeq/analysis/process-samples.html" class="uri">http://jdblischak.github.io/singleCellSeq/analysis/process-samples.html</a>.</p>
<div id="general-information-for-committing-changes-to-the-git-hub-site" class="section level3">
<h3>General information for committing changes to the git hub site:</h3>
<pre><code>git add analysis/*.Rmd
git commit -m &quot;message&quot;</code></pre>
<p>Run <code>wflow_commit()</code> in R terminal.</p>
<pre><code>git push origin master</code></pre>
</div>
<div id="programs-for-analysis" class="section level3">
<h3>Programs for analysis</h3>
<p>First I need to document the programs I will need to add to midway to complete the processing recreation.</p>
<ul>
<li><p>umitools v2.1.1 -fastqc <strong>on midway- not told you need it in github commands!</strong></p></li>
<li>sickle version 1.33 <strong>used make function</strong></li>
<li>Subread version 1.5.0-p1 <strong>on midway</strong></li>
<li>samtools version 0.1.18-dev (r982:313) <strong>version 1.2 on midway</strong></li>
<li>UMI-tools (version e0ade5d)</li>
<li><p>featureCounts version 1.5.0-p1 <strong>i think this is a part of subread</strong></p></li>
</ul>
<p>To use modules on midway:</p>
<pre><code>module load [name]
module list #show loaded modules 
module unload [name]
</code></pre>
</div>
<div id="moving-the-files-from-pps-to-midway" class="section level3">
<h3>Moving the files from pps to midway</h3>
<p>First I copy the first individual, first plate from spudhead to gateway.</p>
<pre><code>scp 19098.1* pps-gateway:</code></pre>
<p>Move the files to midway:</p>
<pre><code>scp 19098.1* brimittleman@midway.rcc.uchicago.edu:</code></pre>
<p>Remove the files from gateway:</p>
<pre><code>rm 19098.1*</code></pre>
<p>Move to the midway cluster and move the files to a directory in scratch.</p>
<pre><code>mv 19098.1* scratch-midway/19098.1 </code></pre>
</div>
<div id="create-genome-for-mapping" class="section level3">
<h3>Create genome for mapping</h3>
<p>Create a code directory for scripts.</p>
<pre><code>mkdir code</code></pre>
<p>Copy contents of create-genome.sh from John’s github to a script in my own directory using vi.</p>
<pre><code>vi create-genome.sh</code></pre>
<p>Use escape then :wq to write out file. Make a file executable using the following:</p>
<pre><code>chmod 700 &lt;file&gt;</code></pre>
<p>Run each script seperatly:</p>
<pre><code>sbatch --mem=8g --wrap=&quot;create-genome.sh genome&quot;</code></pre>
<p>This submits the job.To view the job I use this command.</p>
<pre><code>squeue -u brimittleman</code></pre>
<p>Try this one command at a time in a genome directory. This is buiding the reference index genome for later transcriptome mapping.</p>
<pre><code>mkdir genome
cd genome
wget http://hgdownload.soe.ucsc.edu/goldenPath/hg19/chromosomes/chr{1..22}.fa.gz
wget http://hgdownload.soe.ucsc.edu/goldenPath/hg19/chromosomes/chr{X,Y,M}.fa.gz
gunzip chr*fa.gz
wget http://tools.invitrogen.com/downloads/ERCC92.fa
#load subread
subread-buildindex -o combined *.fa</code></pre>
<p>I need to move data and scripts to the gilad project folder. scratch is not permanant.</p>
<pre><code>mkdir /project/gilad/brimittleman</code></pre>
</div>
<div id="run-fastqc" class="section level3">
<h3>Run FastQC</h3>
<p>Create run-fastqc.sh file:</p>
<pre><code>vi run-fastqc.sh
chmod 700 run-fastqc.sh #make file executable</code></pre>
<p>Load the fastqc package and submit the job to the cluster:</p>
<pre><code>sbatch --mem=2g run-fastqc.sh ../19098.1/19098.1.A09.GGTATTCA.L003.R1.C6WURACXX.fastq.gz #Run this command for eachfastq. </code></pre>
<p><strong>Each new Session: load fastqc, subread, samtools. </strong></p>
<p>Problem at this step:</p>
<ul>
<li><p>Files -19098.1.C08.TTGGTACA.L002.R1.C6WYKACXX.fastq.gz to the end are empty. Cannot run the fast qc on these.</p></li>
<li><p>To check this I go back to the pps spudhead and look at the files using.</p></li>
</ul>
<pre><code> zcat file |wc -l</code></pre>
<ul>
<li>The files are correct on spudhead. I will recopy them over to midway.</li>
</ul>
<p>Return to workflow:</p>
<pre><code>ls fastqc/ | wc -l
#201
</code></pre>
</div>
<div id="run-fastqc-2" class="section level3">
<h3>Run Fastqc 2</h3>
<p>Create run-fastqc.sh file:</p>
<pre><code>vi run-fastqc.sh
chmod 700 run-fastqc.sh #make file executable</code></pre>
<p>I am redoing the run fastqc step using a code that can run all of the files at the same time. I created a file called wrap-sbatch.sh that will run the run-fastqc.sh script in a loop over all of the fastq files. To complete this task all of the .sh scripts have to be in the same directory as the data. I also added a module load command for the fastqc module within the run-fastqc.sh script for completness. I run the sbatch for all of the fastq files using the following command:</p>
<pre><code>./wrap_sbatch.sh *fastq.gz</code></pre>
<p>This submits a job per file and I can check which jobs are running using the squeue - u command.</p>
<pre><code>ls fastqc/ | wc -l</code></pre>
<p>I have 447 lines in the fastqc directory.</p>
</div>
<div id="trim-umi" class="section level3">
<h3>Trim UMI</h3>
<p>Create the trim.sh script using the code from github in the code folder. Run this on all the files by making a script like the wrap_sbatch.sh one above.</p>
<pre><code>./wrap_trim.sh *fastq.gz
ls trim/*fastq.gz | wc -l
#224 this is one file per fastq.gz file in the data.  
ls invalid/*fastq.gz | wc -l
#no invalid files were created </code></pre>
</div>
<div id="quality-trim-3-end-of-reads" class="section level3">
<h3>Quality trim 3’ end of reads</h3>
<p>This uses sickle.sh to perform quality trimming of the 2’ end of the reats. I am using sickle-1.33. I will create a new wrap function for this segmant of code as well that will wrap the sickle.sh code. This code is within the trim folder because those are the files that I will run the command on.</p>
<pre><code>./wrap_sickle.sh *fastq.gz
ls sickle/*fastq.gz | wc -l
#224</code></pre>
</div>
<div id="map-genome" class="section level3">
<h3>Map genome</h3>
<p>I will use the map-subujunc.sh script to map the fastq files to the genome. I will create a new wrap script for this job as well. In the wrap I will load the subread module. I am copying the genome directory to sickle directory so all of the paths are easy.</p>
<pre><code>./wrap_map.sh *fastq.gz
ls bam/*bam | wc -l
#224</code></pre>
<p>The bam files were empty because I exceeded the memory needed. I added –mem=12G to the wrap script. I also changed it to the mstephen partition with –partition=mstephens and changed account with –account=pi-mstephens.</p>
</div>
<div id="process-bam-files" class="section level3">
<h3>Process bam files</h3>
<p>I will use the process-bam.sh script to process the bam files. I will create a new wrap script for this and will have it load samtools module in the script. All of this takes place in the bam folder.</p>
</div>
<div id="re-organize-my-files" class="section level2">
<h2>Re-organize my files</h2>
<p>I am reorganizing my data and code according to John’s organization. I created a single cell directory in my home directory and in my project/gilad directory. I keep all of the scripts I have created in a code directory within the home single cell directory and edit my path variable to include this. The file containing my path directort in called .bashrc and is in the home directory. The fastq files were moved to a fastq directory within the single cell directory in the project folder. This way I am now keeping scripts in my home folder and have all large data in my project folder. I run jobs from the single cell directory in my project/gilad folder.</p>
<div id="umi-trim" class="section level3">
<h3>Umi trim</h3>
<p>-I need to update the trim.sh script for the umi_tools extract command.</p>
<p>The code below works but it puts the output in the directory I ran the script in instead of the trim folder. Going to wrap it for all the files anyway: <code>umi_tools extract --stdin=$FILE  --bc-pattern=[NNNNNGGG]  -L extract.log --stdout=$BASE.trim.fastq.gz \   -E $INVALID_DIR/$BASE.invalid.fastq \   2&gt; $INVALID_DIR/$BASE.trim.stats.txt | gzip -c &gt; $OUTDIR/$BASE.trim.fastq.gz</code></p>
<p>I added code to increase memory and run the code in the mstephens partition to the wrap_trim.sh script and ran it on all of the fastq files using the following code in the project single cell directory.</p>
<pre><code>wrap_trim.sh fastq/*fastq.gz</code></pre>
<p>Only works when I run this in the fastq directory. The output files have *.trim.fastq.gz. I will move these to the trim directory and more the trim directory out of the fastq directory.</p>
</div>
<div id="sickle" class="section level3">
<h3>Sickle</h3>
<p>Run the same wrap sick command within the trim directory. I will move this out when it is done.</p>
</div>
<div id="map-genome-1" class="section level3">
<h3>Map genome</h3>
<p>I am copying the genome directory to sickle directory so all of the paths are easy. Run the following command in the sick directory. I will more the files out later.</p>
<pre><code>wrap_map.sh *trim.sickle.fastq.gz</code></pre>
</div>
<div id="process-bam-files-1" class="section level3">
<h3>Process Bam files</h3>
<p>I copied the code from process-bam.sh and created the same file in my code directory. I will also make a wrap script for this code. I run these commands in the bam folder and will move them out after.</p>
<pre><code>wrap_processbam.sh *trim.sickle.bam</code></pre>
<p>Check for processense of intermediate files output during sorting.</p>
<pre><code>ls bam-processed/*sorted*0*bam</code></pre>
<p>No such files have been created.</p>
</div>
<div id="combine-single-cells" class="section level3">
<h3>Combine Single cells</h3>
<p>I will run this code</p>
<pre><code>#!/bin/bash
module load samtools
mkdir -p bam-combined
mkdir -p ~/log/combine
for IND in 19098 #19101,19239
do
  for REP in 1
  do
    for ROW in {A..F}
    do
      for COL in {1..12}
      do
        ID=`printf &quot;%s.%s.%s%02d\n&quot; $IND $REP $ROW $COL`
        echo $ID
        echo -e &quot;#!/bin/bash\nsamtools merge bam-combined/$ID.trim.sickle.sorted.combined.bam bam-processed/$ID*trim.sickle.sorted.bam&quot; &gt; tmp-$ID.sh
        sbatch tmp-$ID.sh
      done
    done
  done
done</code></pre>
<p>Files are now in bam-combined. There are 72 files. <strong>Missing G and H from the beginning. once I have pipeline working maybe I will go back and add these files</strong></p>
<p>Index each merged sample. Using the following code in a script called index.sh.</p>
<pre><code>
module load samtools
for IND in 19098 #19101,19239
do
  for REP in 1
  do
    for ROW in {A..F}
    do
      for COL in {1..12}
      do
        ID=`printf &quot;%s.%s.%s%02d\n&quot; $IND $REP $ROW $COL`
        echo $ID
        echo -e &quot;#!/bin/bash\nsamtools index bam-combined/$ID.trim.sickle.sorted.combined.bam&quot; &gt; tmp2-$ID.sh
        sbatch tmp2-$ID.sh
      done
    done
  done
done
</code></pre>
<pre><code>ls bam-combined/*bai | wc -l</code></pre>
<p>72</p>
<pre><code>cat tmp_index/slurm* | head</code></pre>
<p>No output.</p>
</div>
<div id="remove-duplicate-umis" class="section level3">
<h3>Remove Duplicate UMIs</h3>
<p>I will now use UMI_tools to remove duplicate UMIs.</p>
<pre><code>#Example code:
umi_tools dedup -I example.bam --output-stats=deduplicated -S deduplicated.bam
#options to think about
--edit-distance-threshold (int) #used 1 in original code </code></pre>
<p>I made the following changes to the original</p>
<pre><code>#prior code
dedup_umi.py --method=&quot;directional-adjacency&quot; --edit-distance-threshold=1 \
  -I $FILE -v 0 -S $OUTDIR/$BASE.rmdup.bam
#new code using UMI_tools 
umi_tools dedup  -I $FILE --output-stats=$OUTDIR/dedup_outputstat  -S $OUTDIR/$B
ASE.rmdup.bam</code></pre>
<p>Created a wrap script called wrap_dedup.sh. Run this on the processed and combined samples.</p>
<p>In the bam-combined folder.</p>
<pre><code> wrap_dedup.sh *bam</code></pre>
<p>To run this in the processed folder not including the bulk I use.</p>
<pre><code>wrap_dedup.sh `ls -1 *.bam | grep -v bulk`  </code></pre>
<p>The umi_tools module is only on midway 1. I need to specify the sandyb partition in my wrap script to be able to run this code over all of the bam files.</p>
<p>I created a new script to run the code on more than one directory at a time called 2dir_dedup.sh. I can use similar code to run the counting functions later on files from 2 directories.</p>
<pre><code>#!/bin/bash

module load samtools
module load python

for file in `ls bam-processed/*bam | grep -v bulk`  bam-combined/*bam
do
    sbatch --mem=4G rmdum-umi.sh $file
done</code></pre>
</div>
<div id="count-reads-per-gene" class="section level3">
<h3>Count reads per gene</h3>
<p>To run this I need to use the create exon script in R to make the necssesary file for the featurecounts tool to use. Do this on midway2 from the genome folder.</p>
<pre><code>Rscript create-exons.R ERCC92.gtf.1 &gt; exons.saf 
</code></pre>
<p>I created the count-reads-per-gene.sh and a code to run it on 2 directories. It is called 2dir_count.sh. I ran this script in the project single cell directory and the ouput is in counts.</p>
</div>
<div id="gather-total-counts-skip-this-section" class="section level3">
<h3>Gather total counts (Skip this section)</h3>
<p>I copied gather-total-counts.py code from the github. This section relies on text files I have not created in all of the directories. It is not necessary I do it.</p>
<pre><code>gather-total-counts.py &gt; total-counts.txt</code></pre>
</div>
<div id="gather-summmary-counts" class="section level3">
<h3>Gather summmary counts</h3>
<p>I copy the gather-summary-counts.py script and comment out the lines that check the number of files. Because I only did one plate these numbers would stop the code from running.</p>
<pre><code>gather-summary-counts.py &gt; summary-counts.txt</code></pre>
</div>
<div id="gather-gene-counts" class="section level3">
<h3>Gather gene counts</h3>
<p>Coppy the gather-gene-counts.py from github. Need to comment out lines that check the number or files becasue I am not working with all of the samples.</p>
<pre><code>mkdir -p counts-matrix
gather-gene-counts.py counts-matrix/ counts/*genecounts.txt</code></pre>
<p>In the end I have a directory called counts-matrix with 6 files in it that will be loaded ito R to continue the process. They are:</p>
<ul>
<li><p>molecules-raw-single-per-lane.txt</p></li>
<li><p>molecules-raw-single-per-sample.txt</p></li>
<li><p>reads-raw-bulk-per-lane.txt</p></li>
<li><p>reads-raw-bulk-per-sample.txt</p></li>
<li><p>reads-raw-single-per-lane.txt</p></li>
<li><p>reads-raw-single-per-sample.txt</p></li>
</ul>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] mygene_1.10.0          GenomicFeatures_1.26.3 GenomicRanges_1.26.3  
 [4] GenomeInfoDb_1.10.3    broman_0.65-1          gplots_3.0.1          
 [7] matrixStats_0.51.0     MASS_7.3-45            gridExtra_2.2.1       
[10] lme4_1.1-12            workflowr_0.4.0        org.Hs.eg.db_3.4.0    
[13] GOstats_2.40.0         graph_1.52.0           Category_2.40.0       
[16] Matrix_1.2-7.1         GO.db_3.4.0            AnnotationDbi_1.36.2  
[19] IRanges_2.8.1          S4Vectors_0.12.1       Biobase_2.34.0        
[22] BiocGenerics_0.20.0    Humanzee_0.1.0         sva_3.22.0            
[25] genefilter_1.56.0      mgcv_1.8-15            nlme_3.1-128          
[28] BiocInstaller_1.24.0   scales_0.4.1           VennDiagram_1.6.17    
[31] futile.logger_1.4.3    edgeR_3.16.5           limma_3.30.11         
[34] dplyr_0.5.0            testit_0.6             data.table_1.10.4     
[37] biomaRt_2.30.0         cowplot_0.7.0          stringr_1.2.0         
[40] reshape2_1.4.2         rmarkdown_1.3          Citrus_0.99           
[43] easyGgplot2_1.0.0.9000 ggplot2_2.2.1         

loaded via a namespace (and not attached):
 [1] minqa_1.2.4                colorspace_1.3-2          
 [3] rprojroot_1.2              htmlTable_1.9             
 [5] XVector_0.14.0             base64enc_0.1-3           
 [7] rstudioapi_0.6             sqldf_0.4-10              
 [9] splines_3.3.2              knitr_1.15.1              
[11] Formula_1.2-1              jsonlite_1.3              
[13] nloptr_1.0.4               Rsamtools_1.26.1          
[15] annotate_1.52.1            cluster_2.0.5             
[17] httr_1.2.1                 backports_1.0.5           
[19] assertthat_0.1             lazyeval_0.2.0            
[21] acepack_1.4.1              htmltools_0.3.5           
[23] tools_3.3.2                gtable_0.2.0              
[25] Rcpp_0.12.9                Biostrings_2.42.1         
[27] gdata_2.17.0               rtracklayer_1.34.2        
[29] proto_1.0.0                gtools_3.5.0              
[31] devtools_1.12.0            statmod_1.4.29            
[33] XML_3.98-1.5               zlibbioc_1.20.0           
[35] SummarizedExperiment_1.4.0 RBGL_1.50.0               
[37] lambda.r_1.1.9             RColorBrewer_1.1-2        
[39] yaml_2.1.14                curl_2.3                  
[41] memoise_1.0.0              rpart_4.1-10              
[43] latticeExtra_0.6-28        stringi_1.1.2             
[45] RSQLite_1.1-2              checkmate_1.8.2           
[47] caTools_1.17.1             BiocParallel_1.8.1        
[49] chron_2.3-50               bitops_1.0-6              
[51] evaluate_0.10              lattice_0.20-34           
[53] htmlwidgets_0.8            GenomicAlignments_1.10.0  
[55] labeling_0.3               GSEABase_1.36.0           
[57] AnnotationForge_1.16.1     plyr_1.8.4                
[59] magrittr_1.5               R6_2.2.0                  
[61] Hmisc_4.0-2                DBI_0.5-1                 
[63] gsubfn_0.6-6               foreign_0.8-67            
[65] withr_1.0.2                survival_2.40-1           
[67] RCurl_1.95-4.8             nnet_7.3-12               
[69] tibble_1.2                 futile.options_1.0.0      
[71] KernSmooth_2.23-15         locfit_1.5-9.1            
[73] git2r_0.18.0               digest_0.6.12             
[75] xtable_1.8-2               munsell_0.4.3             </code></pre>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
