<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />

<meta name="date" content="2017-01-06" />

<title>recreate_cell_cycle_analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">recreate_cell_cycle_analysis</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2017-01-06</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-03-06</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> f3c5dc0</p>
<p>General information for committing changes to the git hub site</p>
<pre><code>git add analysis/*.Rmd
git commit -m &quot;message&quot;</code></pre>
<p>Run <code>commit_site()</code> in R terminal.</p>
<pre><code> git push origin master</code></pre>
<div id="configure-program" class="section level4">
<h4>Configure program</h4>
<p>First I need to install the 2013-03-31 version of gmap in my directory on the midway cluster. I used the instructions in the INSTALL text file within the program folder downloaded from online.</p>
<pre><code>./configure --prefix=/home/brimittleman/software 
make
make install</code></pre>
</div>
<div id="build-reference" class="section level4">
<h4>Build reference</h4>
<p>Download the mouse genome for mapping my reads from: <a href="http://may2012.archive.ensembl.org/Mus_musculus/Info/Index" class="uri">http://may2012.archive.ensembl.org/Mus_musculus/Info/Index</a></p>
<p>Building the mouse reference database on gsnap. These are some of the options in the function I will need to run the reference build.</p>
<pre><code>    -D, --dir=STRING    destination (&quot;/home/brimittleman/software&quot;&quot;)
    
    -k, --kmer=INT      kmer value for gneome (default=15)
        
    -d, --db=STRING     Genome name
    
    -g                  if files are gziped </code></pre>
<p>Secifically I used this code to run the reference build.</p>
<pre><code>./gmap_build -g -d mouseref ../../mouse\ genome/*dna.chrom*fa.gz</code></pre>
<p>Submit this job to a computing node and requested 8g using command:</p>
<pre><code>
sinteractive --mem=8g
</code></pre>
<p>The files from the assembly are in /home/brimittleman/software/share/mouseref.</p>
<p>Download ERCC sequences to add to map.<br />
see : <a href="https://github.com/jdblischak/singleCellSeq/blob/2f6371b256829bfa799270909b6eea2814acf2a1/code/create-genome.sh" class="uri">https://github.com/jdblischak/singleCellSeq/blob/2f6371b256829bfa799270909b6eea2814acf2a1/code/create-genome.sh</a></p>
<pre><code>wget http://tools.thermofisher.com/downloads/ERCC92.fa
</code></pre>
<p>Combine the mouse reference and ERCC sequences and rebuild the genome. Need to unzip the mouse fastq files so I can run the build fucntion with the ERCC and the mouse files. Perform these commands in the mouse ref directory where the files are held.</p>
<pre><code>gunzip *dna.chrom*fa.gz</code></pre>
<p>In the software file I can now rerun the build with this command in the software/bin directory.</p>
<pre><code> ./gmap_build -d mouseref_ERCC  ../../mouse\ genome/*.fa </code></pre>
</div>
<div id="running-gsnap" class="section level4">
<h4>Running GSNAP:</h4>
<p>The general documentation looks like this. <code>gsnap --gunzip -d &lt;genome&gt; &lt;fastq_1.gz&gt; &lt;fastq_2.gz&gt;</code></p>
<p>To test this I will start with running the program on cell 47.</p>
<p>Files: <strong>ERR489047_2.fastq.gz</strong> <strong>ERR489047_1.fastq.gz</strong></p>
<p>Code I am working on for this allignment. Not correct yet.</p>
<pre><code>./gsnap --gunzip -d --db=mouseref_ERCC -D --dir=software/share/ -m 10 scratch-midway/ERR489047_1.fastq.gz ../../scratch-midway/ERR489047_2.fastq.gz</code></pre>
<p><strong>to have the correct default directory (seen by doing gmap â€“version), or use the -D flag to gmap to specify the correct genome directory.</strong> ```</p>
<p>try unzipping the files :<br />
gzip: ERR489047_1.fastq.gz: unexpected end of file</p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] mygene_1.10.0          GenomicFeatures_1.26.3 GenomicRanges_1.26.3  
 [4] GenomeInfoDb_1.10.3    broman_0.65-1          gplots_3.0.1          
 [7] matrixStats_0.51.0     MASS_7.3-45            gridExtra_2.2.1       
[10] lme4_1.1-12            workflowr_0.4.0        org.Hs.eg.db_3.4.0    
[13] GOstats_2.40.0         graph_1.52.0           Category_2.40.0       
[16] Matrix_1.2-7.1         GO.db_3.4.0            AnnotationDbi_1.36.2  
[19] IRanges_2.8.1          S4Vectors_0.12.1       Biobase_2.34.0        
[22] BiocGenerics_0.20.0    Humanzee_0.1.0         sva_3.22.0            
[25] genefilter_1.56.0      mgcv_1.8-15            nlme_3.1-128          
[28] BiocInstaller_1.24.0   scales_0.4.1           VennDiagram_1.6.17    
[31] futile.logger_1.4.3    edgeR_3.16.5           limma_3.30.11         
[34] dplyr_0.5.0            testit_0.6             data.table_1.10.4     
[37] biomaRt_2.30.0         cowplot_0.7.0          stringr_1.2.0         
[40] reshape2_1.4.2         rmarkdown_1.3          Citrus_0.99           
[43] easyGgplot2_1.0.0.9000 ggplot2_2.2.1         

loaded via a namespace (and not attached):
 [1] minqa_1.2.4                colorspace_1.3-2          
 [3] rprojroot_1.2              htmlTable_1.9             
 [5] XVector_0.14.0             base64enc_0.1-3           
 [7] rstudioapi_0.6             sqldf_0.4-10              
 [9] splines_3.3.2              knitr_1.15.1              
[11] Formula_1.2-1              jsonlite_1.3              
[13] nloptr_1.0.4               Rsamtools_1.26.1          
[15] annotate_1.52.1            cluster_2.0.5             
[17] httr_1.2.1                 backports_1.0.5           
[19] assertthat_0.1             lazyeval_0.2.0            
[21] acepack_1.4.1              htmltools_0.3.5           
[23] tools_3.3.2                gtable_0.2.0              
[25] Rcpp_0.12.9                Biostrings_2.42.1         
[27] gdata_2.17.0               rtracklayer_1.34.2        
[29] proto_1.0.0                gtools_3.5.0              
[31] devtools_1.12.0            statmod_1.4.29            
[33] XML_3.98-1.5               zlibbioc_1.20.0           
[35] SummarizedExperiment_1.4.0 RBGL_1.50.0               
[37] lambda.r_1.1.9             RColorBrewer_1.1-2        
[39] yaml_2.1.14                curl_2.3                  
[41] memoise_1.0.0              rpart_4.1-10              
[43] latticeExtra_0.6-28        stringi_1.1.2             
[45] RSQLite_1.1-2              checkmate_1.8.2           
[47] caTools_1.17.1             BiocParallel_1.8.1        
[49] chron_2.3-50               bitops_1.0-6              
[51] evaluate_0.10              lattice_0.20-34           
[53] htmlwidgets_0.8            GenomicAlignments_1.10.0  
[55] labeling_0.3               GSEABase_1.36.0           
[57] AnnotationForge_1.16.1     plyr_1.8.4                
[59] magrittr_1.5               R6_2.2.0                  
[61] Hmisc_4.0-2                DBI_0.5-1                 
[63] gsubfn_0.6-6               foreign_0.8-67            
[65] withr_1.0.2                survival_2.40-1           
[67] RCurl_1.95-4.8             nnet_7.3-12               
[69] tibble_1.2                 futile.options_1.0.0      
[71] KernSmooth_2.23-15         locfit_1.5-9.1            
[73] git2r_0.18.0               digest_0.6.12             
[75] xtable_1.8-2               munsell_0.4.3             </code></pre>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
