<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>Look at single cell only genes by individual</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Look at single cell only genes by individual</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2017/03/03</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-03-06</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> f3c5dc0</p>
<div id="pull-in-genes-lists" class="section level3">
<h3>Pull in genes lists</h3>
<pre class="r"><code>library(&quot;biomaRt&quot;)
library(&quot;data.table&quot;)
library(&quot;testit&quot;)
library(&quot;dplyr&quot;)
library(&quot;edgeR&quot;)
library(&quot;VennDiagram&quot;)
library(&quot;ggplot2&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;../../singleCellSeq/analysis/functions.R&quot;)
require(Humanzee)
library(&quot;GOstats&quot;)</code></pre>
<pre class="r"><code>anno &lt;- read.table(&quot;../../singleCellSeq/data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
#input reads (pre-filter)
reads &lt;- read.table(&quot;../../singleCellSeq/data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
#input molecule (pre-filter)
molecules &lt;- read.table(&quot;../../singleCellSeq/data/molecules.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
#input bulk reads (pre-filter )
reads_bulk &lt;- read.table(&quot;../../singleCellSeq/data/reads-bulk.txt&quot;, header = TRUE, stringsAsFactors = FALSE)</code></pre>
<p>Create vectors with the list of genes in each set.</p>
<pre class="r"><code>genes_observed_reads_bulk &lt;- rownames(reads_bulk)[rowSums(reads_bulk) &gt; 0]
genes_observed_reads_bulk_ENSG= genes_observed_reads_bulk[grep(&quot;ERCC&quot;, genes_observed_reads_bulk, invert = TRUE)]
genes_observed_reads &lt;- rownames(reads)[rowSums(reads) &gt; 0]
genes_observed_molecules &lt;- rownames(molecules)[rowSums(molecules) &gt; 0]
genes_observed_molecules_ENSG &lt;- genes_observed_molecules[grep(&quot;ERCC&quot;, genes_observed_molecules, invert = TRUE)]</code></pre>
<div id="genes-only-in-single-cell-molecules" class="section level4">
<h4>Genes only in single cell molecules</h4>
<pre class="r"><code>gene_onlysinglecell_ENSG &lt;- setdiff(genes_observed_molecules_ENSG,genes_observed_reads_bulk_ENSG)
#subset molecules by these genes  
molecules_df= data.frame(molecules)
molecules_only_SC_ENSG &lt;- molecules_df[rownames(molecules_df) %in% gene_onlysinglecell_ENSG,]
dim(molecules_only_SC_ENSG)</code></pre>
<pre><code>[1] 956 864</code></pre>
<p>Count number of cells we see these genes in</p>
<pre class="r"><code>non_zero_cells_ENSG &lt;- apply(molecules_only_SC_ENSG, 1, function(c)sum(c!=0))

molecules_with_nonzero_ENSG &lt;- cbind(molecules_only_SC_ENSG, non_zero_cells_ENSG)

summary(molecules_with_nonzero_ENSG$non_zero_cells_ENSG)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  1.000   1.000   2.000   2.899   3.000  45.000 </code></pre>
</div>
</div>
<div id="individual-level-analysis" class="section level2">
<h2>Individual level analysis</h2>
<div id="individual-19098" class="section level3">
<h3>Individual 19098</h3>
<pre class="r"><code>molecules_SC_ENSG_19098= molecules_with_nonzero_ENSG[,grep(&quot;19098&quot;, colnames(molecules_with_nonzero_ENSG))]
# this is a dataframe for the sc only genes in 19098
dim(molecules_SC_ENSG_19098)</code></pre>
<pre><code>[1] 956 288</code></pre>
<pre class="r"><code>non_zero_cells_ENSG_19098 &lt;- apply(molecules_SC_ENSG_19098, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19098 &lt;- cbind(molecules_SC_ENSG_19098, non_zero_cells_ENSG_19098)

summary(molecules_with_nonzero_ENSG_19098$non_zero_cells_ENSG_19098)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  0.0000  1.0000  0.8923  1.0000 13.0000 </code></pre>
<pre class="r"><code>plot(molecules_with_nonzero_ENSG_19098$non_zero_cells_ENSG_19098)</code></pre>
<p><img src="figure/sc_only_ind.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="individual-19101" class="section level3">
<h3>Individual 19101</h3>
<pre class="r"><code>molecules_SC_ENSG_19101= molecules_with_nonzero_ENSG[,grep(&quot;19101&quot;, colnames(molecules_with_nonzero_ENSG))]
# this is a dataframe for the sc only genes in 19101

dim(molecules_SC_ENSG_19101)</code></pre>
<pre><code>[1] 956 288</code></pre>
<pre class="r"><code>non_zero_cells_ENSG_19101 &lt;- apply(molecules_SC_ENSG_19101, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19101 &lt;- cbind(molecules_SC_ENSG_19101, non_zero_cells_ENSG_19101)

summary(molecules_with_nonzero_ENSG_19101$non_zero_cells_ENSG_19101)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  0.000   0.000   1.000   1.124   2.000  22.000 </code></pre>
<pre class="r"><code>plot(molecules_with_nonzero_ENSG_19101$non_zero_cells_ENSG_19101)</code></pre>
<p><img src="figure/sc_only_ind.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="individual-19239" class="section level3">
<h3>Individual 19239</h3>
<pre class="r"><code>molecules_SC_ENSG_19239= molecules_with_nonzero_ENSG[,grep(&quot;19239&quot;, colnames(molecules_with_nonzero_ENSG))]
# this is a dataframe for the sc only genes in 19239

dim(molecules_SC_ENSG_19239)</code></pre>
<pre><code>[1] 956 288</code></pre>
<pre class="r"><code>non_zero_cells_ENSG_19239 &lt;- apply(molecules_SC_ENSG_19239, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19239 &lt;- cbind(molecules_SC_ENSG_19239, non_zero_cells_ENSG_19239)

summary(molecules_with_nonzero_ENSG_19239$non_zero_cells_ENSG_19239)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  0.0000  0.0000  0.8818  1.0000 34.0000 </code></pre>
<pre class="r"><code>plot(molecules_with_nonzero_ENSG_19239$non_zero_cells_ENSG_19239)</code></pre>
<p><img src="figure/sc_only_ind.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="compare-individuals" class="section level3">
<h3>Compare individuals</h3>
<p>I want to look at the genes that are expressed in these individuals and see if there are genes that are unique to one individual</p>
<ul>
<li><p>find genes with non_zero &gt; 1 (exclude singleton)</p></li>
<li><p>make a list of these genes</p></li>
<li><p>create a ven diagram with these genes</p></li>
</ul>
<pre class="r"><code>#19098
genes_not_singleton_19098= row.names(molecules_with_nonzero_ENSG_19098[molecules_with_nonzero_ENSG_19098$non_zero_cells_ENSG_19098 &gt; 1, ])
length(genes_not_singleton_19098)</code></pre>
<pre><code>[1] 181</code></pre>
<pre class="r"><code>#19101
genes_not_singleton_19101= row.names(molecules_with_nonzero_ENSG_19101[molecules_with_nonzero_ENSG_19101$non_zero_cells_ENSG_19101 &gt; 1, ])
length(genes_not_singleton_19101)</code></pre>
<pre><code>[1] 241</code></pre>
<pre class="r"><code>#19239
genes_not_singleton_19239= row.names(molecules_with_nonzero_ENSG_19239[molecules_with_nonzero_ENSG_19239$non_zero_cells_ENSG_19239 &gt; 1, ])
length(genes_not_singleton_19239)</code></pre>
<pre><code>[1] 168</code></pre>
<pre class="r"><code>non_sing_venn &lt;- venn.diagram(x = list(&quot;19098&quot; = genes_not_singleton_19098,
                                    &quot;19101&quot; = genes_not_singleton_19101,
                                    &quot;19239&quot; = genes_not_singleton_19239),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(non_sing_venn)</code></pre>
<p><img src="figure/sc_only_ind.Rmd/ven%20diagram-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="replicate-level-analysis" class="section level2">
<h2>Replicate level analysis</h2>
<div id="section" class="section level3">
<h3>19098</h3>
<div id="r1" class="section level4">
<h4>R1</h4>
<pre class="r"><code>molecules_SC_ENSG_19098.R1= molecules_with_nonzero_ENSG_19098[,grep(&quot;r1&quot;, colnames(molecules_with_nonzero_ENSG_19098))]

non_zero_cells_ENSG_19098.R1 &lt;- apply(molecules_SC_ENSG_19098.R1, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19098.R1 &lt;- cbind(molecules_SC_ENSG_19098.R1, non_zero_cells_ENSG_19098.R1)

genes_not_singleton_19098.R1= row.names(molecules_with_nonzero_ENSG_19098.R1[molecules_with_nonzero_ENSG_19098.R1$non_zero_cells_ENSG_19098.R1  &gt; 1, ])
length(genes_not_singleton_19098.R1)</code></pre>
<pre><code>[1] 45</code></pre>
</div>
<div id="r2" class="section level4">
<h4>R2</h4>
<pre class="r"><code>molecules_SC_ENSG_19098.R2= molecules_with_nonzero_ENSG_19098[,grep(&quot;r2&quot;, colnames(molecules_with_nonzero_ENSG_19098))]

non_zero_cells_ENSG_19098.R2 &lt;- apply(molecules_SC_ENSG_19098.R2, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19098.R2 &lt;- cbind(molecules_SC_ENSG_19098.R2, non_zero_cells_ENSG_19098.R2)

genes_not_singleton_19098.R2= row.names(molecules_with_nonzero_ENSG_19098.R2[molecules_with_nonzero_ENSG_19098.R2$non_zero_cells_ENSG_19098.R2  &gt; 1, ])
length(genes_not_singleton_19098.R2)</code></pre>
<pre><code>[1] 28</code></pre>
</div>
<div id="r3" class="section level4">
<h4>R3</h4>
<pre class="r"><code>molecules_SC_ENSG_19098.R3= molecules_with_nonzero_ENSG_19098[,grep(&quot;r3&quot;, colnames(molecules_with_nonzero_ENSG_19098))]

non_zero_cells_ENSG_19098.R3 &lt;- apply(molecules_SC_ENSG_19098.R3, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19098.R3 &lt;- cbind(molecules_SC_ENSG_19098.R3, non_zero_cells_ENSG_19098.R3)

genes_not_singleton_19098.R3= row.names(molecules_with_nonzero_ENSG_19098.R3[molecules_with_nonzero_ENSG_19098.R3$non_zero_cells_ENSG_19098.R3  &gt; 1, ])
length(genes_not_singleton_19098.R3)</code></pre>
<pre><code>[1] 46</code></pre>
</div>
<div id="create-ven-diagram" class="section level4">
<h4>Create ven diagram</h4>
<pre class="r"><code>non_sing_venn_19098 &lt;- venn.diagram(x = list(&quot;R1&quot; = genes_not_singleton_19098.R1,
                                    &quot;R2&quot; = genes_not_singleton_19098.R2,
                                    &quot;R3&quot; = genes_not_singleton_19098.R3),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(non_sing_venn_19098)</code></pre>
<p><img src="figure/sc_only_ind.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="section-1" class="section level3">
<h3>19101</h3>
<div id="r1-1" class="section level4">
<h4>R1</h4>
<pre class="r"><code>molecules_SC_ENSG_19101.R1= molecules_with_nonzero_ENSG_19101[,grep(&quot;r1&quot;, colnames(molecules_with_nonzero_ENSG_19101))]

non_zero_cells_ENSG_19101.R1 &lt;- apply(molecules_SC_ENSG_19101.R1, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19101.R1 &lt;- cbind(molecules_SC_ENSG_19101.R1, non_zero_cells_ENSG_19101.R1)

genes_not_singleton_19101.R1= row.names(molecules_with_nonzero_ENSG_19101.R1[molecules_with_nonzero_ENSG_19101.R1$non_zero_cells_ENSG_19101.R1  &gt; 1, ])
length(genes_not_singleton_19101.R1)</code></pre>
<pre><code>[1] 87</code></pre>
</div>
<div id="r2-1" class="section level4">
<h4>R2</h4>
<pre class="r"><code>molecules_SC_ENSG_19101.R2= molecules_with_nonzero_ENSG_19101[,grep(&quot;r2&quot;, colnames(molecules_with_nonzero_ENSG_19101))]

non_zero_cells_ENSG_19101.R2 &lt;- apply(molecules_SC_ENSG_19101.R2, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19101.R2 &lt;- cbind(molecules_SC_ENSG_19101.R2, non_zero_cells_ENSG_19101.R2)

genes_not_singleton_19101.R2= row.names(molecules_with_nonzero_ENSG_19101.R2[molecules_with_nonzero_ENSG_19101.R2$non_zero_cells_ENSG_19101.R2  &gt; 1, ])
length(genes_not_singleton_19101.R2)</code></pre>
<pre><code>[1] 59</code></pre>
</div>
<div id="r3-1" class="section level4">
<h4>R3</h4>
<pre class="r"><code>molecules_SC_ENSG_19101.R3= molecules_with_nonzero_ENSG_19101[,grep(&quot;r3&quot;, colnames(molecules_with_nonzero_ENSG_19101))]

non_zero_cells_ENSG_19101.R3 &lt;- apply(molecules_SC_ENSG_19101.R3, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19101.R3 &lt;- cbind(molecules_SC_ENSG_19101.R3, non_zero_cells_ENSG_19101.R3)

genes_not_singleton_19101.R3= row.names(molecules_with_nonzero_ENSG_19101.R3[molecules_with_nonzero_ENSG_19101.R3$non_zero_cells_ENSG_19101.R3  &gt; 1, ])
length(genes_not_singleton_19101.R3)</code></pre>
<pre><code>[1] 42</code></pre>
</div>
<div id="create-ven-diagram-1" class="section level4">
<h4>Create ven diagram</h4>
<pre class="r"><code>non_sing_venn_19101 &lt;- venn.diagram(x = list(&quot;R1&quot; = genes_not_singleton_19101.R1,
                                    &quot;R2&quot; = genes_not_singleton_19101.R2,
                                    &quot;R3&quot; = genes_not_singleton_19101.R3),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(non_sing_venn_19101)</code></pre>
<p><img src="figure/sc_only_ind.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="section-2" class="section level3">
<h3>19239</h3>
<div id="r1-2" class="section level4">
<h4>R1</h4>
<pre class="r"><code>molecules_SC_ENSG_19239.R1= molecules_with_nonzero_ENSG_19239[,grep(&quot;r1&quot;, colnames(molecules_with_nonzero_ENSG_19239))]

non_zero_cells_ENSG_19239.R1 &lt;- apply(molecules_SC_ENSG_19239.R1, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19239.R1 &lt;- cbind(molecules_SC_ENSG_19239.R1, non_zero_cells_ENSG_19239.R1)

genes_not_singleton_19239.R1= row.names(molecules_with_nonzero_ENSG_19239.R1[molecules_with_nonzero_ENSG_19239.R1$non_zero_cells_ENSG_19239.R1  &gt; 1, ])
length(genes_not_singleton_19239.R1)</code></pre>
<pre><code>[1] 43</code></pre>
</div>
<div id="r2-2" class="section level4">
<h4>R2</h4>
<pre class="r"><code>molecules_SC_ENSG_19239.R2= molecules_with_nonzero_ENSG_19239[,grep(&quot;r2&quot;, colnames(molecules_with_nonzero_ENSG_19239))]

non_zero_cells_ENSG_19239.R2 &lt;- apply(molecules_SC_ENSG_19239.R2, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19239.R2 &lt;- cbind(molecules_SC_ENSG_19239.R2, non_zero_cells_ENSG_19239.R2)

genes_not_singleton_19239.R2= row.names(molecules_with_nonzero_ENSG_19239.R2[molecules_with_nonzero_ENSG_19239.R2$non_zero_cells_ENSG_19239.R2  &gt; 1, ])
length(genes_not_singleton_19239.R2)</code></pre>
<pre><code>[1] 43</code></pre>
</div>
<div id="r3-2" class="section level4">
<h4>R3</h4>
<pre class="r"><code>molecules_SC_ENSG_19239.R3= molecules_with_nonzero_ENSG_19239[,grep(&quot;r3&quot;, colnames(molecules_with_nonzero_ENSG_19239))]

non_zero_cells_ENSG_19239.R3 &lt;- apply(molecules_SC_ENSG_19239.R3, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19239.R3 &lt;- cbind(molecules_SC_ENSG_19239.R3, non_zero_cells_ENSG_19239.R3)

genes_not_singleton_19239.R3= row.names(molecules_with_nonzero_ENSG_19239.R3[molecules_with_nonzero_ENSG_19239.R3$non_zero_cells_ENSG_19239.R3  &gt; 1, ])
length(genes_not_singleton_19239.R3)</code></pre>
<pre><code>[1] 48</code></pre>
</div>
<div id="create-ven-diagram-2" class="section level4">
<h4>Create ven diagram</h4>
<pre class="r"><code>non_sing_venn_19239&lt;- venn.diagram(x = list(&quot;R1&quot; = genes_not_singleton_19239.R1,
                                    &quot;R2&quot; = genes_not_singleton_19239.R2,
                                    &quot;R3&quot; = genes_not_singleton_19239.R3),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(non_sing_venn_19239)</code></pre>
<p><img src="figure/sc_only_ind.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>&lt;&lt;&lt;&lt;&lt;&lt;&lt; HEAD</p>
</div>
</div>
</div>
<div id="pca-for-sc-only-genes" class="section level2">
<h2>PCA for SC only genes</h2>
<pre class="r"><code>pca_SConly &lt;- run_pca(molecules_only_SC_ENSG )
pca_SConly_plot &lt;- plot_pca(pca_SConly$PCs, explained = pca_SConly$explained,
         metadata = anno, color = &quot;individual&quot;,
         shape = &quot;replicate&quot;, alpha = 0.5, size = 3) +
        labs(title=&quot;PCA for single cell only genes&quot;) +
        geom_text(aes(label= ifelse(PC1 &lt; -60  |  PC2 &gt;40,colnames(molecules_only_SC_ENSG), &quot;&quot;) , vjust=&quot;inward&quot;, hjust=&quot;inward&quot;))
                      
                  

pca_SConly_plot</code></pre>
<p><img src="figure/sc_only_ind.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /> The green one is NA19101.r1.C11 and the red one is NA19098.r1.F01</p>
<p>Check if these cells passes the quality control steps in the original analysis.</p>
<pre class="r"><code>quality_single_cells &lt;- scan(&quot;../../singleCellSeq//data/quality-single-cells.txt&quot;,
                             what = &quot;character&quot;)



green_cell= c(&quot;NA19101.r1.C11&quot;)
red_cell=c(&quot;NA19098.r1.F01&quot;)

green_cell %in% quality_single_cells</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>red_cell %in% quality_single_cells</code></pre>
<pre><code>[1] FALSE</code></pre>
<p>Cell NA19101.r1.C11 is in the list of quality cells. Cell NA19098.r1.F01 was not included in the quality cells.</p>
<div id="exclude-singletons" class="section level3">
<h3>Exclude singletons</h3>
<pre class="r"><code>#ENSG genes that arnt singleton in single cell only
load(&quot;../data/gene_names_ensg.txt&quot;)
#subset molecules for this  

molecules_sc_notsing_ENSG= molecules[rownames(molecules) %in% gene_names_ensg, ]</code></pre>
<p>run and plot PCA</p>
<pre class="r"><code>pca_SConly_nonsing &lt;- run_pca(molecules_sc_notsing_ENSG)
pca_SConly_nonsing_plot &lt;- plot_pca(pca_SConly_nonsing$PCs, explained = pca_SConly_nonsing$explained,
         metadata = anno, color = &quot;individual&quot;,
         shape = &quot;replicate&quot;, alpha = 0.5, size = 3) +
        labs(title=&quot;PCA for single cell only genes seen in more than 1 cell&quot;) +
  geom_text(aes(label= ifelse(PC1 &lt; -15  |  PC2 &gt;10,colnames(molecules_only_SC_ENSG), &quot;&quot;) , vjust=&quot;inward&quot;, hjust=&quot;inward&quot;))
pca_SConly_nonsing_plot </code></pre>
<p><img src="figure/sc_only_ind.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>======= ## Session information</p>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] mygene_1.10.0          GenomicFeatures_1.26.3 GenomicRanges_1.26.3  
 [4] GenomeInfoDb_1.10.3    broman_0.65-1          gplots_3.0.1          
 [7] matrixStats_0.51.0     MASS_7.3-45            gridExtra_2.2.1       
[10] lme4_1.1-12            workflowr_0.4.0        org.Hs.eg.db_3.4.0    
[13] GOstats_2.40.0         graph_1.52.0           Category_2.40.0       
[16] Matrix_1.2-7.1         GO.db_3.4.0            AnnotationDbi_1.36.2  
[19] IRanges_2.8.1          S4Vectors_0.12.1       Biobase_2.34.0        
[22] BiocGenerics_0.20.0    Humanzee_0.1.0         sva_3.22.0            
[25] genefilter_1.56.0      mgcv_1.8-15            nlme_3.1-128          
[28] BiocInstaller_1.24.0   scales_0.4.1           VennDiagram_1.6.17    
[31] futile.logger_1.4.3    edgeR_3.16.5           limma_3.30.11         
[34] dplyr_0.5.0            testit_0.6             data.table_1.10.4     
[37] biomaRt_2.30.0         cowplot_0.7.0          stringr_1.2.0         
[40] reshape2_1.4.2         rmarkdown_1.3          Citrus_0.99           
[43] easyGgplot2_1.0.0.9000 ggplot2_2.2.1         

loaded via a namespace (and not attached):
 [1] minqa_1.2.4                colorspace_1.3-2          
 [3] rprojroot_1.2              htmlTable_1.9             
 [5] XVector_0.14.0             base64enc_0.1-3           
 [7] rstudioapi_0.6             sqldf_0.4-10              
 [9] splines_3.3.2              knitr_1.15.1              
[11] Formula_1.2-1              jsonlite_1.3              
[13] nloptr_1.0.4               Rsamtools_1.26.1          
[15] annotate_1.52.1            cluster_2.0.5             
[17] httr_1.2.1                 backports_1.0.5           
[19] assertthat_0.1             lazyeval_0.2.0            
[21] acepack_1.4.1              htmltools_0.3.5           
[23] tools_3.3.2                gtable_0.2.0              
[25] Rcpp_0.12.9                Biostrings_2.42.1         
[27] gdata_2.17.0               rtracklayer_1.34.2        
[29] proto_1.0.0                gtools_3.5.0              
[31] devtools_1.12.0            statmod_1.4.29            
[33] XML_3.98-1.5               zlibbioc_1.20.0           
[35] SummarizedExperiment_1.4.0 RBGL_1.50.0               
[37] lambda.r_1.1.9             RColorBrewer_1.1-2        
[39] yaml_2.1.14                curl_2.3                  
[41] memoise_1.0.0              rpart_4.1-10              
[43] latticeExtra_0.6-28        stringi_1.1.2             
[45] RSQLite_1.1-2              checkmate_1.8.2           
[47] caTools_1.17.1             BiocParallel_1.8.1        
[49] chron_2.3-50               bitops_1.0-6              
[51] evaluate_0.10              lattice_0.20-34           
[53] htmlwidgets_0.8            GenomicAlignments_1.10.0  
[55] labeling_0.3               GSEABase_1.36.0           
[57] AnnotationForge_1.16.1     plyr_1.8.4                
[59] magrittr_1.5               R6_2.2.0                  
[61] Hmisc_4.0-2                DBI_0.5-1                 
[63] gsubfn_0.6-6               foreign_0.8-67            
[65] withr_1.0.2                survival_2.40-1           
[67] RCurl_1.95-4.8             nnet_7.3-12               
[69] tibble_1.2                 futile.options_1.0.0      
[71] KernSmooth_2.23-15         locfit_1.5-9.1            
[73] git2r_0.18.0               digest_0.6.12             
[75] xtable_1.8-2               munsell_0.4.3             </code></pre>
<blockquote>
<blockquote>
<blockquote>
<blockquote>
<blockquote>
<blockquote>
<blockquote>
<p>86594a472ae2f2872b38459580b5cf8e13b8b533</p>
</blockquote>
</blockquote>
</blockquote>
</blockquote>
</blockquote>
</blockquote>
</blockquote>
</div>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
