<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />

<meta name="date" content="2016-02-08" />

<title>Final Subsample plots</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Final Subsample plots</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2016-02-08</em></h4>

</div>


<p><strong>Last updated:</strong> 2017-02-09</p>
<p><strong>Code version:</strong> 6ea08f2</p>
<p>A subsample of the full data, post mapping, was processed to create the count data used here.</p>
<p>Extra information about the subsample was created using the subsample-pipeline.py pipeline.<br />
###Input</p>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;tidyr&quot;)
library(&quot;ggplot2&quot;)
library(&quot;cowplot&quot;)
theme_set(theme_bw(base_size = 12))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank(),
             legend.key = element_blank(),
             plot.title = element_text(size = rel(1)))</code></pre>
<pre class="r"><code>d &lt;- read.table(&quot;../../singleCellSeq/data/subsampling-results.txt&quot;,
                header = TRUE, sep = &quot;\t&quot;, stringsAsFactors = FALSE)
str(d)</code></pre>
<pre><code>&#39;data.frame&#39;:   6240 obs. of  25 variables:
 $ type            : chr  &quot;reads&quot; &quot;molecules&quot; &quot;reads&quot; &quot;molecules&quot; ...
 $ depth           : int  50000 50000 50000 50000 50000 50000 250000 250000 250000 250000 ...
 $ gene_subset     : chr  &quot;all&quot; &quot;all&quot; &quot;lower&quot; &quot;lower&quot; ...
 $ seed            : int  1 1 1 1 1 1 1 1 1 1 ...
 $ subsampled_cells: int  5 5 5 5 5 5 5 5 5 5 ...
 $ individual      : chr  &quot;NA19098&quot; &quot;NA19098&quot; &quot;NA19098&quot; &quot;NA19098&quot; ...
 $ replicate       : logi  NA NA NA NA NA NA ...
 $ lower_q         : num  0 0 0 0 0.5 0.5 0 0 0 0 ...
 $ upper_q         : num  1 1 0.5 0.5 1 1 1 1 0.5 0.5 ...
 $ available_ensg  : int  12192 12192 12192 12192 12192 12192 12192 12192 12192 12192 ...
 $ used_ensg       : int  12192 12192 6096 6096 6096 6096 12192 12192 6096 6096 ...
 $ available_ercc  : int  43 43 43 43 43 43 43 43 43 43 ...
 $ used_ercc       : int  43 43 22 22 22 22 43 43 22 22 ...
 $ potential_cells : int  142 142 142 142 142 142 142 142 142 142 ...
 $ available_cells : int  142 142 142 142 142 142 142 142 142 142 ...
 $ pearson_ensg    : num  0.774 0.79 0.368 0.372 0.727 ...
 $ pearson_ercc    : num  0.893 0.907 0.456 0.406 0.914 ...
 $ spearman_ensg   : num  0.772 0.79 0.405 0.411 0.703 ...
 $ spearman_ercc   : num  0.854 0.853 0.456 0.383 0.955 ...
 $ detected_ensg   : int  7972 7972 2413 2413 5559 5559 9493 9493 3550 3550 ...
 $ detected_ercc   : int  29 29 10 10 20 20 35 35 14 14 ...
 $ mean_counts_ensg: num  22471 12126 1292 738 21180 ...
 $ mean_counts_ercc: num  568.8 269 8.2 3.2 561.4 ...
 $ var_pearson     : num  0.804 0.846 0.4 0.455 0.76 ...
 $ var_spearman    : num  0.762 0.792 0.451 0.467 0.693 ...</code></pre>
<pre class="r"><code>d_grouped &lt;- d %&gt;%
  group_by(type, depth, gene_subset, subsampled_cells,
           individual, potential_cells, available_cells,
           lower_q, upper_q, available_ensg, used_ensg,
           available_ercc, used_ercc) %&gt;%
  summarize(mean_detected = mean(detected_ensg),
            sem_detected = sd(detected_ensg) / sqrt(length(detected_ensg)),
            mean_bulk = mean(pearson_ensg),
            sem_bulk = sd(pearson_ensg) / sqrt(length(pearson_ensg)),
            mean_var = mean(var_pearson),
            sem_var = sd(var_pearson) / sqrt(length(var_pearson)))</code></pre>
<pre class="r"><code>d_filter &lt;- d_grouped %&gt;% filter(individual == &quot;NA19239&quot;,
                                type == &quot;molecules&quot;,
                                gene_subset %in% c(&quot;lower&quot;, &quot;upper&quot;))
d_filter$gene_subset &lt;- factor(d_filter$gene_subset,
                               levels = c(&quot;lower&quot;, &quot;upper&quot;),
                               labels = c(&quot;Lower 50% of expressed genes&quot;,
                                          &quot;Upper 50% of expressed genes&quot;))</code></pre>
<div id="figures" class="section level3">
<h3>Figures</h3>
<pre class="r"><code>plot_bulk_title &lt;- &quot;Correlation of gene expression levels \n between single cells and the corresponding bulk samples&quot;
plot_bulk_title &lt;- &quot;Correlation of gene expression levels between single cells and the corresponding bulk samples&quot;

plot_bulk &lt;- ggplot(d_filter,
                 aes(x = subsampled_cells, y = mean_bulk,
                     color = as.factor(depth))) +
  geom_line() +
  geom_errorbar(aes(ymin = mean_bulk - sem_bulk,
                    ymax = mean_bulk + sem_bulk),
                width = 1) +
  facet_wrap(~gene_subset) +
  scale_color_grey(start = 0.8, end = 0.2, name = &quot;Sequencing depth&quot;) +
  theme(legend.position = &quot;none&quot;) +
  labs(x = &quot;Number of subsampled cells&quot;,
       y = &quot;Pearson&#39;s r (+/- SEM)&quot;,
       title = paste(strwrap(plot_bulk_title, width = 80), collapse = &quot;\n&quot;))
plot_bulk</code></pre>
<p><img src="figure/subsample_plots.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /> This is comparing a subset of the single cells to the bulk samples. We are looking at the the correlation as a function of how many of the sampled cells the gene was found in. This uses molecule counts from individual NA19239.</p>
<pre class="r"><code>plot_detected &lt;- ggplot(d_filter,
                 aes(x = subsampled_cells, y = mean_detected,
                     color = as.factor(depth))) +
  geom_line() +
  geom_errorbar(aes(ymin = mean_detected - sem_detected,
                    ymax = mean_detected + sem_detected),
                width = 1) +
  facet_wrap(~gene_subset) +
  scale_color_grey(start = 0.8, end = 0.2, name = &quot;Sequencing depth&quot;,
                   labels = format(unique(d$depth), big.mark = &quot;,&quot;,
                                   scientifc = FALSE, trim = TRUE)) +
  theme(legend.position = c(0.75, 0.35),
        legend.key.size = grid::unit(0.2, &quot;in&quot;)) +
  labs(x = &quot;Number of subsampled cells&quot;,
       y = &quot;Number of genes detected&quot;,
       title = &quot;Number of genes detected as expressed in single cell samples \n at different sequence coverage&quot;)
plot_detected</code></pre>
<p><img src="figure/subsample_plots.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /> This plot looks at the number of genes detected at diffent sequencing depths as a function of the number of subsampled cells used in this analysis.</p>
<pre class="r"><code>plot_var &lt;- ggplot(d_filter,
                 aes(x = subsampled_cells, y = mean_var,
                     color = as.factor(depth))) +
  geom_line() +
  geom_errorbar(aes(ymin = mean_var - sem_var,
                    ymax = mean_var + sem_var),
                width = 1) +
  facet_wrap(~gene_subset) +
  scale_color_grey(start = 0.8, end = 0.2, name = &quot;Sequencing depth&quot;) +
  theme(legend.position = &quot;none&quot;) +
  labs(x = &quot;Number of subsampled cells&quot;,
       y = &quot;Pearson&#39;s r (+/- SEM)&quot;,
       title = &quot;Correlation of the variation in gene expression levels \n between subsets of single cells and the entire single cell dataset&quot;)
plot_var</code></pre>
<p><img src="figure/subsample_plots.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /> Correlation between subsample and all of the cell in terms of the variance gene expression levels. THe different colors are the sequencing depth.</p>
<p>Plot these next to eachother in a grid.</p>
<pre class="r"><code>plot_final &lt;- plot_grid(plot_bulk, plot_detected, plot_var,
                        ncol = 1, labels = letters[1:3], label_size = 12)
plot_final</code></pre>
<p><img src="figure/subsample_plots.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># png(&quot;../paper/figure/fig-subsample.png&quot;, width = 8, height = 12, units = &quot;in&quot;, res = 300)
tiff(&quot;../paper/figure/fig-subsample.tiff&quot;,
     width = 8, height = 12, units = &quot;in&quot;, res = 300, compression = &quot;zip&quot;)
plot_final
dev.off()</code></pre>
<pre><code>Warning in dev.off(): unable to open TIFF file &#39;../paper/figure/fig-
subsample.tiff&#39;</code></pre>
<pre><code>png 
  2 </code></pre>
</div>
<div id="number-of-molecules-per-sequencing-depth" class="section level3">
<h3>Number of molecules per sequencing depth</h3>
<pre class="r"><code>molecules_per_depth &lt;- d %&gt;%
  filter(gene_subset == &quot;all&quot;, type == &quot;molecules&quot;)
#subset for only molecule rows  </code></pre>
<pre class="r"><code>#molecule depth for endogenous genes 
plot_mol_depth_ensg &lt;- ggplot(molecules_per_depth,
                              aes(x = as.factor(depth),
                                  y = mean_counts_ensg)) +
  geom_boxplot() +
  facet_wrap(~individual) +
  scale_y_continuous(breaks = seq(0, 1e5, by = 1e4)) +
  theme(axis.text.x = element_text(angle = 90))
plot_mol_depth_ensg</code></pre>
<p><img src="figure/subsample_plots.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#molecule depth for ERCC
plot_mol_depth_ercc &lt;- plot_mol_depth_ensg %+% aes(y = mean_counts_ercc)
plot_mol_depth_ercc</code></pre>
<p><img src="figure/subsample_plots.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /> ###Session info</p>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] tidyr_0.6.1     gridExtra_2.2.1 lme4_1.1-12     Matrix_1.2-7.1 
 [5] edgeR_3.16.5    limma_3.30.9    cowplot_0.7.0   ggplot2_2.2.1  
 [9] dplyr_0.5.0     workflowr_0.3.0 rmarkdown_1.3  

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.9      knitr_1.15.1     magrittr_1.5     MASS_7.3-45     
 [5] splines_3.3.2    munsell_0.4.3    lattice_0.20-34  colorspace_1.3-2
 [9] R6_2.2.0         minqa_1.2.4      stringr_1.1.0    plyr_1.8.4      
[13] tools_3.3.2      grid_3.3.2       nlme_3.1-128     gtable_0.2.0    
[17] DBI_0.5-1        git2r_0.18.0     htmltools_0.3.5  yaml_2.1.14     
[21] lazyeval_0.2.0   assertthat_0.1   rprojroot_1.2    digest_0.6.12   
[25] tibble_1.2       nloptr_1.0.4     evaluate_0.10    labeling_0.3    
[29] stringi_1.1.2    scales_0.4.1     backports_1.0.5  locfit_1.5-9.1  </code></pre>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
