<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittl,ena" />

<meta name="date" content="2017-02-13" />

<title>Account for technical noise measured with ERCC controls using Poisson GLM transformation</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Account for technical noise measured with ERCC controls using Poisson GLM transformation</h1>
<h4 class="author"><em>Briana Mittl,ena</em></h4>
<h4 class="date"><em>2017-02-13</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-03-06</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> f3c5dc0</p>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;doMC&quot;)</code></pre>
<pre><code>Loading required package: foreach</code></pre>
<pre><code>foreach: simple, scalable parallel programming from Revolution Analytics
Use Revolution R for scalability, fault tolerance and more.
http://www.revolutionanalytics.com</code></pre>
<pre><code>Loading required package: iterators</code></pre>
<pre class="r"><code>registerDoMC(7)
library(&quot;ggplot2&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;../../singleCellSeq/analysis/functions.R&quot;)</code></pre>
<p>This file transforms the standardizes molecule counts by modeling the ERCC counts using a Poisson generalized linear model (glm).</p>
<p>Creates the following file and writes it to data directory in this github repo:</p>
<ul>
<li>molecules-cpm-trans.txt - Molecules in high quality single cells after linear transfmoration with ERCC</li>
</ul>
<div id="input" class="section level3">
<h3>Input</h3>
<p>Input filtered annotation.</p>
<pre class="r"><code>anno_filter &lt;- read.table(&quot;../data/annotation-filter.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno_filter)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
4    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
5    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06
6    NA19098        r1  A07 NA19098.r1 NA19098.r1.A07</code></pre>
<p>Input filtered molecule counts.</p>
<pre class="r"><code>molecules_filter &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header = TRUE,
                               stringsAsFactors = FALSE)
stopifnot(ncol(molecules_filter) == nrow(anno_filter),
          colnames(molecules_filter) == anno_filter$sample_id)</code></pre>
<p>Input standardized molecule counts.</p>
<pre class="r"><code>molecules_cpm &lt;- read.table(&quot;../data/molecules-cpm.txt&quot;, header = TRUE,
                            stringsAsFactors = FALSE)
stopifnot(ncol(molecules_cpm) == nrow(anno_filter),
          colnames(molecules_cpm) == anno_filter$sample_id)</code></pre>
<p>Input exprected ERCC molecule counts (found in capture efficiency analysis). Filter to only include those ERCC controls that passed the expression filter from the filter low quality genes and cells analysis.</p>
<pre class="r"><code>spike &lt;- read.table(&quot;../../singleCellSeq/data/expected-ercc-molecules.txt&quot;, header = TRUE,
                    sep = &quot;\t&quot;, stringsAsFactors = FALSE)
spike_input &lt;- spike$ercc_molecules_well[spike$id %in% rownames(molecules_filter)]
names(spike_input) &lt;- spike$id[spike$id %in% rownames(molecules_filter)]
spike_input &lt;- spike_input[order(names(spike_input))]
stopifnot(names(spike_input) == grep(&quot;ERCC&quot;, rownames(molecules_filter), value = TRUE))
#names() gets or sets names of an object </code></pre>
</div>
<div id="poisson-spike-in-only-model" class="section level3">
<h3>Poisson spike-in only model</h3>
<pre class="r"><code>#use lapply when you want a list back 
pois_glm_params &lt;- lapply(1:3,function(g) numeric(ncol(molecules_cpm)))
names(pois_glm_params) = c(&quot;intercept&quot;,&quot;nu&quot;,&quot;theta&quot;)
for (i in 1:ncol(molecules_cpm)) {
  #run glm model on a cell (564)
  fit &lt;- glm(molecules_filter[names(spike_input), i] ~ log(spike_input), family=&quot;poisson&quot;)
  pois_glm_params$intercept[i] &lt;- fit$coefficients[1]
  pois_glm_params$nu[i] &lt;- fit$coefficients[2]
}
#subtract intercepy divide by nu 
molecules_cpm_trans &lt;- sweep( sweep( molecules_cpm, 2, pois_glm_params$intercept, &quot;-&quot;), 2,pois_glm_params$nu, &quot;/&quot; )
#sweep- return an array obtained from input arrray by sweeping out a summary stat</code></pre>
<p>Write to file:</p>
<pre class="r"><code>write.table(round(molecules_cpm_trans, digits = 6), &quot;../data/molecules-cpm-trans.txt&quot;, quote = FALSE,
            sep = &quot;\t&quot;, col.names = NA)</code></pre>
</div>
<div id="pca" class="section level3">
<h3>PCA</h3>
<pre class="r"><code>pca_molecules_cpm_trans &lt;- run_pca(molecules_cpm_trans)
pca_molecules_cpm_trans_plot &lt;- plot_pca(pca_molecules_cpm_trans$PCs, explained = pca_molecules_cpm_trans$explained,
         metadata = anno_filter, color = &quot;individual&quot;,
         shape = &quot;replicate&quot;) +
  labs(title = &quot;Molecules (Poisson transformation with ERCC) for single cells&quot;)
pca_molecules_cpm_trans_plot</code></pre>
<p><img src="figure/technical_noise_control.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /> This did a linear transformation for all of the genes in one cell according to the nu and intercept from the glm model. 92 total ERCC genes.</p>
</div>
<div id="analysis-of-variance" class="section level3">
<h3>Analysis of variance</h3>
<p>Collect together all the normalized expression matrices for analysis</p>
<pre class="r"><code>cpm_mats &lt;- list(raw=molecules_cpm, molecules_cpm_trans=molecules_cpm_trans)</code></pre>
<p>We need a nested ANOVA since the replicates are not common across all individuals.</p>
<pre class="r"><code>my_nested_anova=function(temp) {
  global_mean=mean(temp$y)
  
  ind_means=temp %&gt;% group_by(individual) %&gt;% summarize(m=mean(y)) %&gt;% as.data.frame
  rownames(ind_means) = ind_means$individual
  temp$ind_means=ind_means[ temp$individual, &quot;m&quot; ]
  
  batch_means=temp %&gt;% group_by(batch) %&gt;% summarise(m=mean(y)) %&gt;% as.data.frame
  rownames(batch_means) = batch_means$batch 
  temp$batch_means=batch_means[ temp$batch, &quot;m&quot; ]
  
  c(ssa=sum( (temp$ind_means - global_mean)^2 ),
  ssb=sum( (temp$batch_means - temp$ind_means)^2 ),
  sse=sum( (temp$y - temp$batch_means)^2 ),
  sst=sum( (temp$y - global_mean)^2  ))
}</code></pre>
<p>Run ANOVAs per gene for each matrix and calculate variance components.</p>
<pre class="r"><code>anovas &lt;- lapply(cpm_mats, function(x) {
  foreach(i=1:nrow(x)) %dopar% my_nested_anova(data.frame(y=as.numeric(x[i,]), batch=anno_filter$batch, individual=anno_filter$individual))
})

variance_components &lt;- lapply( as.list(names(anovas)), function(name) {
  ss=do.call(rbind,anovas[[name]])[,1:3]
  colnames(ss)=c(&quot;individual&quot;,&quot;batch&quot;,&quot;residual&quot;)
  data.frame(sweep(ss,1,rowSums(ss),&quot;/&quot;), method=name)
} )
names(variance_components)=names(cpm_mats)

batch_over_explained &lt;- lapply( as.list(names(anovas)), function(name) {
  ss=do.call(rbind,anovas[[name]])[,1:2]
  colnames(ss)=c(&quot;individual&quot;,&quot;batch&quot;)
  data.frame( prop_batch=ss[,&quot;batch&quot;] / rowSums(ss), method=name)
} )
names(batch_over_explained) = names(cpm_mats)</code></pre>
<p>Plot proportions explained by batch vs.Â individual</p>
<pre class="r"><code>cbPalette &lt;- c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;, &quot;#009E73&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;)
ggplot( do.call(rbind,batch_over_explained), aes(prop_batch,col=method)) + geom_density(alpha=0.2, position=&quot;identity&quot;) + xlab(&quot;variance_batch / (variance_batch + variance_individual)&quot;) + theme(legend.position=c(.8,.8)) + scale_colour_manual(values=cbPalette)+ scale_fill_manual(values=cbPalette)</code></pre>
<p><img src="figure/technical_noise_control.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Shows that the transformation has normalized the variance. The plot is less squed toward batch vairance.</p>
<pre class="r"><code>ggplot( do.call(rbind,variance_components), aes(1-residual,col=method)) + geom_density() + xlab(&quot;proportion variance explained&quot;) + xlim(0,.5)+ scale_colour_manual(values=cbPalette) + theme(legend.position=c(.8,.8))</code></pre>
<pre><code>Warning: Removed 204 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/technical_noise_control.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /> <strong>What does this plot tell me?</strong></p>
<div id="notes-on-anova" class="section level4">
<h4>Notes on anova</h4>
<ul>
<li><p>difference among group means</p></li>
<li><p>partition observed variance into components of attributed to difference sources of variation</p></li>
<li><p>Calculate mean of means and SST ((each point - grand mean)^2)</p></li>
</ul>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] doMC_1.3.4             iterators_1.0.8        foreach_1.4.3         
 [4] tidyr_0.6.1            mygene_1.10.0          GenomicFeatures_1.26.3
 [7] GenomicRanges_1.26.3   GenomeInfoDb_1.10.3    broman_0.65-1         
[10] gplots_3.0.1           matrixStats_0.51.0     MASS_7.3-45           
[13] gridExtra_2.2.1        lme4_1.1-12            workflowr_0.4.0       
[16] org.Hs.eg.db_3.4.0     GOstats_2.40.0         graph_1.52.0          
[19] Category_2.40.0        Matrix_1.2-7.1         GO.db_3.4.0           
[22] AnnotationDbi_1.36.2   IRanges_2.8.1          S4Vectors_0.12.1      
[25] Biobase_2.34.0         BiocGenerics_0.20.0    Humanzee_0.1.0        
[28] sva_3.22.0             genefilter_1.56.0      mgcv_1.8-15           
[31] nlme_3.1-128           BiocInstaller_1.24.0   scales_0.4.1          
[34] VennDiagram_1.6.17     futile.logger_1.4.3    edgeR_3.16.5          
[37] limma_3.30.11          dplyr_0.5.0            testit_0.6            
[40] data.table_1.10.4      biomaRt_2.30.0         cowplot_0.7.0         
[43] stringr_1.2.0          reshape2_1.4.2         rmarkdown_1.3         
[46] Citrus_0.99            easyGgplot2_1.0.0.9000 ggplot2_2.2.1         

loaded via a namespace (and not attached):
 [1] minqa_1.2.4                colorspace_1.3-2          
 [3] rprojroot_1.2              htmlTable_1.9             
 [5] XVector_0.14.0             base64enc_0.1-3           
 [7] rstudioapi_0.6             sqldf_0.4-10              
 [9] codetools_0.2-15           splines_3.3.2             
[11] knitr_1.15.1               Formula_1.2-1             
[13] jsonlite_1.3               nloptr_1.0.4              
[15] Rsamtools_1.26.1           annotate_1.52.1           
[17] cluster_2.0.5              compiler_3.3.2            
[19] httr_1.2.1                 backports_1.0.5           
[21] assertthat_0.1             lazyeval_0.2.0            
[23] acepack_1.4.1              htmltools_0.3.5           
[25] tools_3.3.2                gtable_0.2.0              
[27] Rcpp_0.12.9                Biostrings_2.42.1         
[29] gdata_2.17.0               rtracklayer_1.34.2        
[31] proto_1.0.0                gtools_3.5.0              
[33] devtools_1.12.0            statmod_1.4.29            
[35] XML_3.98-1.5               zlibbioc_1.20.0           
[37] SummarizedExperiment_1.4.0 RBGL_1.50.0               
[39] lambda.r_1.1.9             RColorBrewer_1.1-2        
[41] yaml_2.1.14                curl_2.3                  
[43] memoise_1.0.0              rpart_4.1-10              
[45] latticeExtra_0.6-28        stringi_1.1.2             
[47] RSQLite_1.1-2              checkmate_1.8.2           
[49] caTools_1.17.1             BiocParallel_1.8.1        
[51] chron_2.3-50               bitops_1.0-6              
[53] evaluate_0.10              lattice_0.20-34           
[55] htmlwidgets_0.8            GenomicAlignments_1.10.0  
[57] labeling_0.3               GSEABase_1.36.0           
[59] AnnotationForge_1.16.1     plyr_1.8.4                
[61] magrittr_1.5               R6_2.2.0                  
[63] Hmisc_4.0-2                DBI_0.5-1                 
[65] gsubfn_0.6-6               foreign_0.8-67            
[67] withr_1.0.2                survival_2.40-1           
[69] RCurl_1.95-4.8             nnet_7.3-12               
[71] tibble_1.2                 futile.options_1.0.0      
[73] KernSmooth_2.23-15         locfit_1.5-9.1            
[75] git2r_0.18.0               digest_0.6.12             
[77] xtable_1.8-2               munsell_0.4.3             </code></pre>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
