<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="First Last" />


<title>Try Cellity for single cell quality</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cell cycle</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Try Cellity for single cell quality</h1>
<h4 class="author"><em>First Last</em></h4>
<h4 class="date"><em>YYYY-MM-DD</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-03-06</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> f3c5dc0</p>
<p>Package available at: <a href="https://github.com/Teichlab/cellity" class="uri">https://github.com/Teichlab/cellity</a>.</p>
<pre class="r"><code>library(cellity)</code></pre>
<pre><code></code></pre>
<pre><code>sROC 0.1-2 loaded</code></pre>
<pre class="r"><code>library(ggplot2)
theme_set(theme_bw(12))</code></pre>
<pre class="r"><code>data(&quot;sample_counts&quot;)
data(&quot;sample_stats&quot;)
head(sample_stats)</code></pre>
<pre><code>                       cell    total   mapped unmapped unique multi
688 ola_mES_lif_1_17.counts  8590414  6907697  1682717    100     0
167  ola_mES_2i_2_72.counts 19091056 15139851  3951205    100     0
710 ola_mES_lif_1_37.counts 16286506 13266685  3019821    100     0
60   ola_mES_2i_1_63.counts 21247692 17420807  3826885    100     0
64   ola_mES_2i_1_67.counts 13250828 11032819  2218009    100     0
676 ola_mES_a2i_3_94.counts  7189806  5635576  1554230    100     0
    intergenic intragenic exonic intronic ambigious exonicM alignments
688     100.00       0.00   0.00     0.00      0.00       0    6907697
167      10.10      80.21  56.05    24.15      9.69       0   15139851
710     100.00       0.00   0.00     0.00      0.00       0   13266685
60       11.67      81.22  33.98    47.25      7.10       0   17420807
64       10.05      81.34  51.28    30.06      8.61       0   11032819
676      28.24      63.87  53.28    10.59      7.88       0    5635576
    multi.intergenic multi.intragenic multi.exonic multi.intronic
688                0                0            0              0
167                0                0            0              0
710                0                0            0              0
60                 0                0            0              0
64                 0                0            0              0
676                0                0            0              0
    multi.ambigious  perfect partly_perfect mapped_no_correct      S_0
688               0  3940816        2966881                 0  4080656
167               0  9170872        5968979                 0  9463813
710               0  7475789        5790896                 0  7718481
60                0 11259672        6161135                 0 11627976
64                0  6324302        4708517                 0  6537438
676               0  2134023        3501553                 0  2201804
       S_1    S_2    S_3    S_4    S_5    S_6    S_7    S_8    S_9   S_10.
688 100572  96611  92092  51648  55652  55384  57126  52220  72013 2193723
167 242220 227580 225318 121555 142397 140321 144653 131113 207985 4092896
710 189778 181234 174493  94670  97539  97985 104384  97494 126801 4383826
60  253026 233955 217581 119767 119233 119844 126409 129666 201632 4271718
64  175626 158864 152961  86861  87396  87277  90788  89180 121017 3445411
676  84632  77381  76296  54056  53278  54325  56135  52752  60737 2864180
         I      D  INDEL
688  90137 103740 191614
167 193124 221082 409212
710 156290 182547 334600
60  205673 312753 511500
64  134538 167409 298044
676  48462  59668 107061</code></pre>
<pre class="r"><code>names(sample_stats)</code></pre>
<pre><code> [1] &quot;cell&quot;              &quot;total&quot;             &quot;mapped&quot;           
 [4] &quot;unmapped&quot;          &quot;unique&quot;            &quot;multi&quot;            
 [7] &quot;intergenic&quot;        &quot;intragenic&quot;        &quot;exonic&quot;           
[10] &quot;intronic&quot;          &quot;ambigious&quot;         &quot;exonicM&quot;          
[13] &quot;alignments&quot;        &quot;multi.intergenic&quot;  &quot;multi.intragenic&quot; 
[16] &quot;multi.exonic&quot;      &quot;multi.intronic&quot;    &quot;multi.ambigious&quot;  
[19] &quot;perfect&quot;           &quot;partly_perfect&quot;    &quot;mapped_no_correct&quot;
[22] &quot;S_0&quot;               &quot;S_1&quot;               &quot;S_2&quot;              
[25] &quot;S_3&quot;               &quot;S_4&quot;               &quot;S_5&quot;              
[28] &quot;S_6&quot;               &quot;S_7&quot;               &quot;S_8&quot;              
[31] &quot;S_9&quot;               &quot;S_10.&quot;             &quot;I&quot;                
[34] &quot;D&quot;                 &quot;INDEL&quot;            </code></pre>
<p>Mapping and gene count pipeline: <a href="https://github.com/Teichlab/celloline" class="uri">https://github.com/Teichlab/celloline</a></p>
<p>Talk to Joyce about this. Should we move all of the data over to project 2 and try to get this up and running?</p>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] cellity_1.2.0          doMC_1.3.4             iterators_1.0.8       
 [4] foreach_1.4.3          tidyr_0.6.1            mygene_1.10.0         
 [7] GenomicFeatures_1.26.3 GenomicRanges_1.26.3   GenomeInfoDb_1.10.3   
[10] broman_0.65-1          gplots_3.0.1           matrixStats_0.51.0    
[13] MASS_7.3-45            gridExtra_2.2.1        lme4_1.1-12           
[16] workflowr_0.4.0        org.Hs.eg.db_3.4.0     GOstats_2.40.0        
[19] graph_1.52.0           Category_2.40.0        Matrix_1.2-7.1        
[22] GO.db_3.4.0            AnnotationDbi_1.36.2   IRanges_2.8.1         
[25] S4Vectors_0.12.1       Biobase_2.34.0         BiocGenerics_0.20.0   
[28] Humanzee_0.1.0         sva_3.22.0             genefilter_1.56.0     
[31] mgcv_1.8-15            nlme_3.1-128           BiocInstaller_1.24.0  
[34] scales_0.4.1           VennDiagram_1.6.17     futile.logger_1.4.3   
[37] edgeR_3.16.5           limma_3.30.11          dplyr_0.5.0           
[40] testit_0.6             data.table_1.10.4      biomaRt_2.30.0        
[43] cowplot_0.7.0          stringr_1.2.0          reshape2_1.4.2        
[46] rmarkdown_1.3          Citrus_0.99            easyGgplot2_1.0.0.9000
[49] ggplot2_2.2.1         

loaded via a namespace (and not attached):
  [1] backports_1.0.5            Hmisc_4.0-2               
  [3] plyr_1.8.4                 lazyeval_0.2.0            
  [5] sp_1.2-4                   GSEABase_1.36.0           
  [7] splines_3.3.2              BiocParallel_1.8.1        
  [9] digest_0.6.12              htmltools_0.3.5           
 [11] gdata_2.17.0               magrittr_1.5              
 [13] checkmate_1.8.2            memoise_1.0.0             
 [15] cluster_2.0.5              Biostrings_2.42.1         
 [17] annotate_1.52.1            colorspace_1.3-2          
 [19] rrcov_1.4-3                RCurl_1.95-4.8            
 [21] jsonlite_1.3               org.Mm.eg.db_3.4.0        
 [23] zoo_1.7-14                 survival_2.40-1           
 [25] gtable_0.2.0               zlibbioc_1.20.0           
 [27] XVector_0.14.0             MatrixModels_0.4-1        
 [29] kernlab_0.9-25             car_2.1-4                 
 [31] prabclus_2.2-6             DEoptimR_1.0-8            
 [33] SparseM_1.74               VIM_4.6.0                 
 [35] sgeostat_1.0-27            futile.options_1.0.0      
 [37] mvtnorm_1.0-6              GGally_1.3.0              
 [39] DBI_0.5-1                  sROC_0.1-2                
 [41] Rcpp_0.12.9                laeken_0.4.6              
 [43] xtable_1.8-2               htmlTable_1.9             
 [45] foreign_0.8-67             mclust_5.2.2              
 [47] Formula_1.2-1              sqldf_0.4-10              
 [49] vcd_1.4-3                  AnnotationForge_1.16.1    
 [51] htmlwidgets_0.8            httr_1.2.1                
 [53] fpc_2.1-10                 RColorBrewer_1.1-2        
 [55] modeltools_0.2-21          acepack_1.4.1             
 [57] flexmix_2.3-13             reshape_0.8.6             
 [59] XML_3.98-1.5               nnet_7.3-12               
 [61] locfit_1.5-9.1             labeling_0.3              
 [63] munsell_0.4.3              tools_3.3.2               
 [65] gsubfn_0.6-6               RSQLite_1.1-2             
 [67] pls_2.6-0                  devtools_1.12.0           
 [69] cvTools_0.3.2              evaluate_0.10             
 [71] yaml_2.1.14                knitr_1.15.1              
 [73] robustbase_0.92-7          caTools_1.17.1            
 [75] RBGL_1.50.0                quantreg_5.29             
 [77] pbkrtest_0.4-6             compiler_3.3.2            
 [79] rstudioapi_0.6             curl_2.3                  
 [81] e1071_1.6-8                tibble_1.2                
 [83] statmod_1.4.29             robCompositions_2.0.3     
 [85] pcaPP_1.9-61               stringi_1.1.2             
 [87] trimcluster_0.1-2          lattice_0.20-34           
 [89] nloptr_1.0.4               lmtest_0.9-35             
 [91] bitops_1.0-6               rtracklayer_1.34.2        
 [93] R6_2.2.0                   latticeExtra_0.6-28       
 [95] topGO_2.26.0               KernSmooth_2.23-15        
 [97] codetools_0.2-15           lambda.r_1.1.9            
 [99] boot_1.3-18                gtools_3.5.0              
[101] assertthat_0.1             chron_2.3-50              
[103] SummarizedExperiment_1.4.0 proto_1.0.0               
[105] rprojroot_1.2              withr_1.0.2               
[107] GenomicAlignments_1.10.0   Rsamtools_1.26.1          
[109] diptest_0.75-7             rpart_4.1-10              
[111] class_7.3-14               minqa_1.2.4               
[113] mvoutlier_2.0.8            git2r_0.18.0              
[115] base64enc_0.1-3           </code></pre>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
