---
title: "Gene x sample count matrix Bri data"
author: "Briana Mittlemant"
date: YYYY-MM-DD
output: html_document
---

```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

```{r knitr-opts-chunk, include=FALSE}
```

**Last updated:** `r Sys.Date()`

**Code version:** `r workflowr::extract_commit(".", 1)$sha1`

```{r}
library("data.table")
library("dplyr")
library("VennDiagram")
```

This analysis will take the matrix from the fastq to counts anayisis pipeline (sample-by-gene) to a standard gene-x-sample matrix. I will also create annotation files and romove the genes with no observed sequencing data in the experiment. I will use the data from git hub rather than the data I created because I want to look at the whole experiment rather than one plate at this point.  

###Read Counts for single cell samples: 
```{r}
reads_raw_bri <- fread("../../single-cell/data/counts-matrix/reads-raw-single-per-sample.txt")
setDF(reads_raw_bri)
```

Explore this data:
```{r}
dim(reads_raw_bri)

```

Ths has rows for each individule and the columns give information about the sample then give counts for each gene.  
  
  
Create annotation file for single cell samples: 
```{r}
anno_bri <- reads_raw_bri %>%
  select(individual:well) %>%
  mutate(batch = paste(individual, replicate, sep = "."),
         sample_id = paste(batch, well, sep = "."))
head(anno_bri)
dim(anno_bri)
```

This annoation file has information about the sample, replicatte, batch.

Transpose the matrix so that it is gene-x-sample. 

```{r}
reads_bri <- reads_raw_bri %>%
  select(starts_with("ENSG"), starts_with("ERCC")) %>%
  t
colnames(reads_bri) <- anno_bri$sample_id
reads[1:5, 1:5]
```

###Molecule Counts for single cell samples  
Import eaw molecules for single cell samples.

```{r}
molecules_raw_bri <- fread("../../single-cell/data/counts-matrix/molecules-raw-single-per-lane.txt")
                       
setDF(molecules_raw_bri)
dim(molecules_raw_bri)
```

This data has rows by individuale and has gene columns like the reads data. 

Conform single cell samples in reads and molecules files are in same order. 

```{r}
#stopifnot(anno_bri$individual == molecules_raw_bri$individual,
#          anno_bri$replicate == molecules_raw_bri$replicate,
#          anno_bri$well == molecules_raw_bri$well)
```
