---
title: "Look for genes only in single cell for 1 individual"
author: "First Last"
date: YYYY-MM-DD
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

```{r}
library("data.table")
library("dplyr")
library("VennDiagram")
```

###Input
```{r}
#subset bulk for 19098
reads_bulk <- read.table("../../singleCellSeq/data/reads-bulk.txt", header = TRUE, stringsAsFactors = FALSE)
reads_bulk_19098 <- reads_bulk[,1:3]
head(reads_bulk_19098)

```

```{r}
#input molecule 
molecules <- read.table("../../singleCellSeq/data/molecules.txt", header = TRUE,
                    stringsAsFactors = FALSE)
molecules_19098 <- molecules[,1:288]
head(molecules_19098)
```
###Create gene lists for 19098  

```{r}
genes_observed_reads_bulk_19098 <- rownames(reads_bulk_19098)[rowSums(reads_bulk_19098) > 0]
genes_observed_molecules_19098 <- rownames(molecules_19098)[rowSums(molecules_19098) > 0]
```


###Ven diagram for single cell vs bulk  

```{r}
genes_venn_19098 <- venn.diagram(x = list("molecules" = genes_observed_molecules_19098,
                                    "reads bulk" = genes_observed_reads_bulk_19098),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(genes_venn_19098)
```

## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
