---
title: "Data transformation"
author: "Briana Mittleman"
date: 2017-02-15
output: html_document
---

```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

```{r knitr-opts-chunk, include=FALSE}
```

**Last updated:** `r Sys.Date()`

**Code version:** `r workflowr::extract_commit(".", 1)$sha1`

###Data transformation  
```{r}
library("ggplot2")
library("cowplot")
theme_set(theme_bw(base_size = 16))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank(),
             legend.key = element_blank(),
             plot.title = element_text(size = rel(1)))
source("../../singleCellSeq/analysis/functions.R")
```

This file performs principal component analysis (PCA) and displays the resultd from each stage of our data transformation (normalization) pipeline. Furthermore, it quantifies and tests the PCA results using pca-utils. The methodology described in Worley et al., 2013. Briefly, the differences between samples are calculated using the squared Mahalanobis distance. The distances are then scales so that the hypothesis test that the samples from two groups come from seperate distributionss can be computed with an F-test.   

###Input
Input filtered annotation.
```{r}
anno_filter<- read.table("../data/annotation-filter.txt", header=TRUE, stringsAsFactors = FALSE)
head(anno_filter)
```
Input filtered molecule counts.  
```{r}
molecules_filter <- read.table("../data/molecules-filter.txt", header = TRUE, stringsAsFactors = FALSE)
stopifnot(ncol(molecules_filter) == nrow(anno_filter),
          colnames(molecules_filter) == anno_filter$sample_id)
```

Input standardized molecule counts (CPM).  

```{r}
#from stabdarize_CPM.Rmd
molecules_cpm <- read.table("../data/molecules-cpm.txt", header = TRUE,
                            stringsAsFactors = FALSE)
stopifnot(ncol(molecules_cpm) == nrow(anno_filter),
          colnames(molecules_cpm) == anno_filter$sample_id)
```

Input Poisson GLM transfomred molecule counts per million,  

```{r}
#from technical_noise_control.Rmd
molecules_cpm_trans <- read.table("../data/molecules-cpm-trans.txt", header = TRUE,
                               stringsAsFactors = FALSE)
stopifnot(ncol(molecules_cpm_trans) == nrow(anno_filter),
          colnames(molecules_cpm_trans) == anno_filter$sample_id)
```

Input final batch-corrected molecule counts per million.  

```{r}
#from mixed_effect_batch_correction.Rmd
molecules_final <- read.table("../data/molecules-final.txt", header = TRUE,
                              stringsAsFactors = FALSE)
stopifnot(ncol(molecules_final) == nrow(anno_filter),
          colnames(molecules_final) == anno_filter$sample_id)

```


###PCA  
####PCA of filtered data  
```{r}
pca_molecules_filter <- run_pca(molecules_filter)
pca_molecules_filter_title <- "The effect of technical batch on single cell gene expression for raw counts data"
pca_molecules_filter_plot <- plot_pca(pca_molecules_filter$PCs,
         explained = pca_molecules_filter$explained,
         metadata = anno_filter, color = "individual",
         shape = "replicate", alpha = 0.5, size = 3) +
  # labs(title = "The effect of technical batch \n on single cell gene expression data \n when raw counts are used")
  labs(title = paste(strwrap(pca_molecules_filter_title, width = 50), collapse = "\n"))
```

```{r}
#create data frame with PCA output 
pca_molecules_filter_out <- data.frame(Obs.id = 1:nrow(anno_filter),
                                       Obs.batch = anno_filter$batch,
                                       pca_molecules_filter$PCs[, 1:2])
write.table(pca_molecules_filter_out,
            file = "../data/pca-molecules-filter.txt",
            quote = FALSE, sep = "\t", row.names = FALSE)
```


```{bash}


pca-distances -i ../data/pca-molecules-filter.txt
```

```{bash}
pca-dendrogram -i ../data/pca-molecules-filter.txt
```


####PCA of standardized data  
```{r}
pca_molecules_cpm <- run_pca(molecules_cpm)
pca_molecules_cpm_title <- "The effect of technical batch on single cell gene expression for log counts per million data"
pca_molecules_cpm_plot <- plot_pca(pca_molecules_cpm$PCs,
         explained = pca_molecules_cpm$explained,
         metadata = anno_filter, color = "individual",
         shape = "replicate", alpha = 0.5, size = 3) +
  labs(title = paste(strwrap(pca_molecules_cpm_title, width = 50), collapse = "\n"))
  # labs(title = "The effect of technical batch \n on single cell gene expression data \n when log transformed counts per million data are used")
```

```{r}
pca_molecules_cpm_out <- data.frame(Obs.id = 1:nrow(anno_filter),
                                       Obs.batch = anno_filter$batch,
                                       pca_molecules_cpm$PCs[, 1:2])
write.table(pca_molecules_cpm_out,
            file = "../data/pca-molecules-cpm.txt",
            quote = FALSE, sep = "\t", row.names = FALSE)
```

```{bash}

pca-distances -i ../data/pca-molecules-cpm.txt 
pca-dendrogram -i ../data/pca-molecules-cpm.txt 
```

####PCA of Poisson GLM transformed molecule counts per million  

```{r}
pca_molecules_cpm_trans <- run_pca(molecules_cpm_trans)
pca_molecules_cpm_trans_title <- "The effect of technical batch on single cell gene expression for Poisson transformed data"
pca_molecules_cpm_trans_plot <- plot_pca(pca_molecules_cpm_trans$PCs,
         explained = pca_molecules_cpm_trans$explained,
         metadata = anno_filter, color = "individual",
         shape = "replicate",  alpha = 0.5, size = 3) +
  labs(title = paste(strwrap(pca_molecules_cpm_trans_title, width = 50), collapse = "\n"))
  # labs(title = "The effect of technical batch \n on single cell gene expression data \n when Poisson transformed data are used")
```

```{r}
pca_molecules_cpm_trans_out <- data.frame(Obs.id = 1:nrow(anno_filter),
                                       Obs.batch = anno_filter$batch,
                                       pca_molecules_cpm_trans$PCs[, 1:2])
write.table(pca_molecules_cpm_trans_out,
            file = "../data/pca-molecules-cpm-trans.txt",
            quote = FALSE, sep = "\t", row.names = FALSE)
```

```{bash}
pca-distances -i ../data/pca-molecules-cpm-trans.txt
pca-dendrogram -i ../data/pca-molecules-cpm-trans.txt
```

####PCA of final batch-corrected data  

```{r}
pca_final <- run_pca(molecules_final)
pca_final_title <- "PCA of standardized, normalized, and batch corrected single cell gene expression data"
pca_final_plot <- plot_pca(pca_final$PCs, explained = pca_final$explained,
         metadata = anno_filter, color = "individual",
         shape = "replicate", alpha = 0.5, size = 3) +
  labs(title = paste(strwrap(pca_final_title, width = 50), collapse = "\n"))
  # labs(title = "PCA of standardized, \n normalized, and batch corrected \n single cell gene expression data")
```


```{r}
pca_final_out <- data.frame(Obs.id = 1:nrow(anno_filter),
                            Obs.batch = anno_filter$batch,
                            pca_final$PCs[, 1:2])
write.table(pca_final_out,
            file = "../data/pca-final.txt",
            quote = FALSE, sep = "\t", row.names = FALSE)
```

```{bash}
pca-distances -i ../data/pca-final.txt
pca-dendrogram -i ../data/pca-final.txt
```

###Data transformation figure  
Figure 4 in paper  

```{r}
plot_final <- plot_grid(
  pca_molecules_filter_plot +
    scale_color_discrete(name = "Individual") +
    guides(shape = FALSE) + theme(legend.position = "bottom"),
  pca_molecules_cpm_plot +
    scale_shape(name = "Replicate") +
    guides(color = FALSE) + theme(legend.position = "bottom"),
  pca_molecules_cpm_trans_plot +
    scale_color_discrete(name = "Individual") +
    guides(shape = FALSE) + theme(legend.position = "bottom"),
  pca_final_plot +
    scale_shape(name = "Replicate") +
    guides(color = FALSE) + theme(legend.position = "bottom"),
  labels = letters[1:4])
plot_final
```

Skip writting this figure as a tiff file  


###Session Information  

```{r}
sessionInfo()
```


