---
title: "Explore the ERCC genes that are only found in SC sequences"
author: "Briana Mittleman"
date: 2017-03-10
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```



Input packages and functions  
```{r packages}
library("biomaRt")
library("dplyr")
library("edgeR")
library("ggplot2")
theme_set(theme_bw(base_size = 12))
source("../../singleCellSeq/analysis/functions.R")
library(VennDiagram)
```


Input data:
```{r data}
anno <- read.table("../../singleCellSeq/data/annotation.txt", header = TRUE,
                   stringsAsFactors = FALSE)
#input reads (pre-filter)
reads <- read.table("../../singleCellSeq/data/reads.txt", header = TRUE,
                    stringsAsFactors = FALSE)
#input molecule (pre-filter)
molecules <- read.table("../../singleCellSeq/data/molecules.txt", header = TRUE,
                    stringsAsFactors = FALSE)
#input bulk reads (pre-filter )
reads_bulk <- read.table("../../singleCellSeq/data/reads-bulk.txt", header = TRUE, stringsAsFactors = FALSE)
```


Create vectors with the list of genes in each set.   
```{r}
genes_observed_reads_bulk <- rownames(reads_bulk)[rowSums(reads_bulk) > 0]
genes_observed_reads <- rownames(reads)[rowSums(reads) > 0]
genes_observed_molecules <- rownames(molecules)[rowSums(molecules) > 0]
single_cell_observed <- union(genes_observed_molecules, genes_observed_reads)
```

Create ven diagram for single cell ERCC vs bulk ERCC 
```{r}

ERCC_observed_bulk =genes_observed_reads_bulk[grep("ERCC", genes_observed_reads_bulk)]
ERCC_observed_singlecell= genes_observed_molecules[grep("ERCC", genes_observed_molecules)]


ERCC_ven <- venn.diagram(x = list("Single Cells" = ERCC_observed_singlecell,
                                    "Bulk " = ERCC_observed_bulk),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(ERCC_ven)

```




###ERCCs only in single cell molecules  

```{r}

gene_onlysinglecell <- setdiff(single_cell_observed,genes_observed_reads_bulk) 

genes_onlysinglecell_ERCC <- gene_onlysinglecell[grep("ERCC", gene_onlysinglecell)]
molecules_df= data.frame(molecules)
molecules_sconly_ERCC <- molecules_df[rownames(molecules_df) %in% genes_onlysinglecell_ERCC,]
dim(molecules_sconly_ERCC)
```


Explore the SC only ERCCs  

```{r}
ERCC_sc_only= rownames(molecules_sconly_ERCC)
ERCC_sc_only
ERCC_sc_mean=  apply(molecules_sconly_ERCC, 1, mean)
ERCC_sc_mean


```

Number of Cells each ERCC is in  
```{r}
non_zero_cells_ERCC <- apply(molecules_sconly_ERCC, 1, function(c)sum(c!=0))
non_zero_cells_ERCC 
```




###Individual Level Analysis 

####19098
```{r}
molecules_sconly_ERCC_19098= molecules_sconly_ERCC[,grep("19098", colnames(molecules_sconly_ERCC))]

non_zero_cells_ERCC_19098 <- apply(molecules_sconly_ERCC_19098, 1, function(c)sum(c!=0))
molecules_sconly_ERCC_19098= cbind(molecules_sconly_ERCC_19098,non_zero_cells_ERCC_19098)

non_zero_sc_ERCC_19098= molecules_sconly_ERCC_19098[molecules_sconly_ERCC_19098$non_zero_cells_ERCC_19098>0,]
ERCC_19098=rownames(non_zero_sc_ERCC_19098)
```

####19101

```{r}
molecules_sconly_ERCC_19101= molecules_sconly_ERCC[,grep("19101", colnames(molecules_sconly_ERCC))]

non_zero_cells_ERCC_19101 <- apply(molecules_sconly_ERCC_19101, 1, function(c)sum(c!=0))
molecules_sconly_ERCC_19101= cbind(molecules_sconly_ERCC_19101,non_zero_cells_ERCC_19101)

non_zero_sc_ERCC_19101= molecules_sconly_ERCC_19101[molecules_sconly_ERCC_19101$non_zero_cells_ERCC_19101>0,]
ERCC_19101= rownames(non_zero_sc_ERCC_19101)
```

####19239  
```{r}
molecules_sconly_ERCC_19239= molecules_sconly_ERCC[,grep("19239", colnames(molecules_sconly_ERCC))]

non_zero_cells_ERCC_19239 <- apply(molecules_sconly_ERCC_19239, 1, function(c)sum(c!=0))
molecules_sconly_ERCC_19239= cbind(molecules_sconly_ERCC_19239,non_zero_cells_ERCC_19239)

non_zero_sc_ERCC_19239= molecules_sconly_ERCC_19239[molecules_sconly_ERCC_19239$non_zero_cells_ERCC_19239>0,]
ERCC_19239=rownames(non_zero_sc_ERCC_19239)
```


####Ven diagram  


```{r}
ERCC_sc_ven <- venn.diagram(x = list("19098" = ERCC_19098,
                                    "19101" = ERCC_19101,
                                    "19239" = ERCC_19239),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(ERCC_sc_ven)
```

Which ERCC are seen in all indiv sc  

```{r}
ERCC_all_ind= Reduce(intersect, list(ERCC_19098, ERCC_19101, ERCC_19239))

ERCC_all_ind


```
```{r}
ERCC_expected= read.table("cms_095046.txt", header=TRUE, stringsAsFactors = FALSE, fill=TRUE)
ERCC_nums=c("ERCC-00048", "ERCC-00083", "ERCC-00097", "ERCC-00104", "ERCC-00164", "ERCC-00168")
ERCC_all_ind_exp= ERCC_expected[ERCC_expected$ID %in% ERCC_nums, ]
ERCC_nt_ind= ERCC_expected[! ERCC_expected$ID %in% ERCC_nums, ]
boxplot(ERCC_nt_ind$concentration, ERCC_all_ind_exp$concentration,names=c("SC only 3 ind", "other ERCCs"), ylab="Concentration", main= "All individual single cell only ERCC \ngenes are low concentration ERCCs")
```

####Histogram for expression of 6 genes

```{r}
 ERCC_all_ind
#molecules_sconly_ERCC has expression for single cell only ERCCs
ERCC_48= molecules_sconly_ERCC[grep("ERCC-00048", rownames(molecules_sconly_ERCC )), ]
ERCC_83= molecules_sconly_ERCC[grep("ERCC-00083", rownames(molecules_sconly_ERCC )), ]
ERCC_97= molecules_sconly_ERCC[grep("ERCC-00097", rownames(molecules_sconly_ERCC )), ]
ERCC_104= molecules_sconly_ERCC[grep("ERCC-00104", rownames(molecules_sconly_ERCC )), ]
ERCC_164= molecules_sconly_ERCC[grep("ERCC-00164", rownames(molecules_sconly_ERCC )), ]
ERCC_168= molecules_sconly_ERCC[grep("ERCC-00168", rownames(molecules_sconly_ERCC )), ]
```

Create hitograms for expression of each ERCC  

```{r}
ERCC_48_vec= as.vector(as.matrix(ERCC_48[1,]))
ERCC_48_vec_no0=ERCC_48_vec[ERCC_48_vec !=0]
length(ERCC_48_vec_no0)
hist_48=hist(ERCC_48_vec_no0, main="Single cell only expression of ERCC 48", xlab="Count", xlim=c(1,5))

```

```{r}
ERCC_83_vec= as.vector(as.matrix(ERCC_83[1,]))
ERCC_83_vec_no0=ERCC_83_vec[ERCC_83_vec !=0]
length(ERCC_83_vec_no0)
hist_83= hist(ERCC_83_vec_no0, main="Single cell only expression of ERCC 83", xlab="Count", xlim=c(1,5))
```

```{r}
ERCC_97_vec= as.vector(as.matrix(ERCC_97[1,]))
ERCC_97_vec_no0=ERCC_97_vec[ERCC_97_vec !=0]
length(ERCC_97_vec_no0)
ERCC_97_vec_no0
hist_97= hist(ERCC_97_vec_no0, main="Single cell only expression of ERCC 97", xlab="Count", xlim=c(1,5))
```
```{r}
ERCC_104_vec= as.vector(as.matrix(ERCC_104[1,]))
ERCC_104_vec_no0=ERCC_104_vec[ERCC_104_vec !=0]
length(ERCC_104_vec_no0)
hist_104= hist(ERCC_104_vec_no0, main="Single cell only expression of ERCC 104", xlab="Count", xlim=c(1,5))
```

```{r}
ERCC_164_vec= as.vector(as.matrix(ERCC_164[1,]))
ERCC_164_vec_no0=ERCC_164_vec[ERCC_164_vec !=0]
length(ERCC_164_vec_no0)
ERCC_164_vec_no0
hist_164= hist(ERCC_164_vec_no0, main="Single cell only expression of ERCC 164", xlab="Count", xlim=c(1,5))
```


```{r}

#not getting rid of zeros!!! 
ERCC_168_vec= as.vector(as.matrix(ERCC_168[1,]))
ERCC_168_vec_no0=ERCC_97_vec[ERCC_168_vec!=0]
length(ERCC_168_vec_no0)
ERCC_168_vec_no0
hist_168= hist(ERCC_168_vec_no0, main="Single cell only expression of ERCC 168", xlab="Count", xlim=c(1,5))
```

Highest expression is 3 reads! thesea are very low! 

### Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
