---
title: "Explore genes only found in bulk or single cells"
author: "Briana Mittleman"
date: 2016-02-10
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

This analysis looks into genes that were only detected in either the single cell or bulk sequencing. Contrary to many's intuitiion there were more genes detected using single cell sequencing than when we pooled cells together. I wanted to explore the expression levels of these genes, variability in expression levels, and the proportion of single cells where these genes were detected. For the bulk specific genes I wanted to see how their expression compares to the expression levels of the genes found in both analysis. Information found here could provide information about bias or provide evidence for the importance of single cell sequencing to detect gene expression.  


###Pull in genes lists  

```{r}
library("biomaRt")
library("data.table")
library("testit")
library("dplyr")
library("edgeR")
library("VennDiagram")
library("ggplot2")
library("vioplot")
theme_set(theme_bw(base_size = 12))
source("../../singleCellSeq/analysis/functions.R")
```


```{r}
anno <- read.table("../../singleCellSeq/data/annotation.txt", header = TRUE,
                   stringsAsFactors = FALSE)
#input reads (pre-filter)
reads <- read.table("../../singleCellSeq/data/reads.txt", header = TRUE,
                    stringsAsFactors = FALSE)
#input molecule (pre-filter)
molecules <- read.table("../../singleCellSeq/data/molecules.txt", header = TRUE,
                    stringsAsFactors = FALSE)
#input bulk reads (pre-filter )
reads_bulk <- read.table("../../singleCellSeq/data/reads-bulk.txt", header = TRUE, stringsAsFactors = FALSE)
```
Create vectors with the list of genes in each set.   
```{r}
genes_observed_reads_bulk <- rownames(reads_bulk)[rowSums(reads_bulk) > 0]
genes_observed_reads <- rownames(reads)[rowSums(reads) > 0]
genes_observed_molecules <- rownames(molecules)[rowSums(molecules) > 0]
```

recreate ven diagram  

```{r}
genes_venn <- venn.diagram(x = list("reads" = genes_observed_reads,
                                    "molecules" = genes_observed_molecules,
                                    "reads bulk" = genes_observed_reads_bulk),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(genes_venn)
```
```{r}

single_cell_observed <- union(genes_observed_molecules, genes_observed_molecules)
gene_onlybulk <- setdiff(genes_observed_reads_bulk, single_cell_observed)
gene_onlysinglecell <- setdiff(single_cell_observed,genes_observed_reads_bulk)
```

There are 301 genes only seen in the bulk samples and 971 only seen in the single cells.

###Only in Single cell

Subset molecule count matrix for genes in genes only single cell

```{r}
molecules_df= data.frame(molecules)
molecules_only_SC <- molecules_df[rownames(molecules_df) %in% gene_onlysinglecell,]

```

Get mean expression for molecule counts.  

```{r}
molecules_mean <- apply(molecules, 1, mean)
molecules_and_mean <- cbind(molecules, molecules_mean)
dim(molecules_and_mean )
#sort molecules and mean by expression
molecules_and_mean_order <- arrange(molecules_and_mean, desc(molecules_mean))

                               
                               
which_onlySC <- as.numeric(rownames(molecules_and_mean) %in% gene_onlysinglecell) + 1
col_vec_onlySC <- c("lightpink", "blue")[as.numeric(which_onlySC)]

library("scales")



mean_expression <- plot((molecules_and_mean_order$molecules_mean) , pch=46, cex=10, col=col_vec_onlySC, main = "Mean expression of molecules per gene", ylab=" Molecule count", xlab = "genes (blue are only single cell)")



```

```{r}
#deal with IQR 

molecule_IQR <- apply(molecules, 1, IQR)
molecule_w.IQR <- cbind(molecules_and_mean, molecule_IQR)
#sort by IQR
molecules_and_IQR_order <- arrange(molecule_w.IQR, desc(molecule_IQR))


IQR_plot <- plot((molecules_and_IQR_order$molecule_IQR) , pch=46, cex=10, col=col_vec_onlySC, main = "IQR of expression of molecules per gene", ylab=" IQR in Molecule count", xlab = "genes (blue are only single cell)")
```


####Proportion of cells showing these genes
```{r}
#molecules_only_SC has the 971 genes we care about
#create a vector containing number of !0 counts in the table and divide by 864 cell

non_zero_cells <- apply(molecules_only_SC, 1, function(c)sum(c!=0))

molecules_with_nonzero <- cbind(molecules_only_SC, non_zero_cells)
#order by number non zero
molecules_non_zero_order <- arrange(molecules_with_nonzero, desc(non_zero_cells))

plot((molecules_non_zero_order$non_zero_cells)/864, main="Most genes are seen in low proportion of cells", ylab="Proportion cells- non-zero molecule count", xlab="gene index")

```

This plot is made with the genes we only detected in the single cell analysis. Of these genes, most of them are seen in a low proportion of cells (<2%). This could be a reason the bulk samples could not capture these cells. These cells are unique and could be interesting but only genes that were seen in the union of single cell and bulk were used in the paper. If genes are differncially expressed in a bulk sample we may not have the power to see this when only bulk RNA seq is done.  


Flip this analysis around and look at the proportion of cells expressing genes for all genes in the set and highlighting the single cell only genes.  

```{r}
non_zero_cells_all <- apply(molecules, 1, function(c)sum(c!=0))
molecules_with_nonzero_all <- cbind(molecules, non_zero_cells_all)
molecules_non_zero_order_all <- arrange(molecules_with_nonzero_all, desc(non_zero_cells_all))

xy <- data.frame(x= c(1:NROW(molecules_with_nonzero_all)),
     y = molecules_with_nonzero_all$non_zero_cells_all/864)
plot(xy,  pch=16, cex=1, 
     col=alpha(col_vec_onlySC, .8), xlab = "Gene index", ylab="Proportion cells- non-zero molecule count" , main= "Genes seen only in sc span distribution")

# plot((molecules_non_zero_order_all$non_zero_cells_all)/864,  pch=16, cex=1, col=alpha(col_vec_onlySC, .8), xlab = "Gene index", ylab="Proportion cells- non-zero molecule count" , main= "Genes seen only in sc span distribution")
# points(((molecules_non_zero_order_all$non_zero_cells_all)/864)[which(rownames(molecules_non_zero_all) %in% rownames(molecules_with_nonzero))], col = "black", cex = 2)



```


```{r}
#maybe 2 boxplots with the percentage of the genes in both vs the genes only in single cell? 
#seperate molecules_with_nonzero_all to take out the cells that are in the only sc sett

#single cell only
summary(molecules_with_nonzero$non_zero_cells)
summary(molecules_with_nonzero_all$non_zero_cells_all)
boxplot((molecules_with_nonzero$non_zero_cells)/864, (molecules_with_nonzero_all$non_zero_cells_all)/864, names=c("only single cell", "all genes"), ylab="Proportion cells- non-zero molecule count")

vioplot((molecules_with_nonzero$non_zero_cells)/864, (molecules_with_nonzero_all$non_zero_cells_all)/864, names=c("only single cell", "all genes"), col="darkviolet")
title(ylab="Proportion cell containing gene")
title("Single cell genes in lower proportion of cells")

table(molecules_with_nonzero$non_zero_cells)
table(molecules_with_nonzero_all$non_zero_cells_all)
#genes found only once in all data (this is y)
dim(molecules_with_nonzero_all[molecules_with_nonzero_all$non_zero_cells_all==1,])[1]
dim(molecules_with_nonzero_all[molecules_with_nonzero_all$non_zero_cells_all==2,])[1]


###In function need to count the number of these genes that are only found found in single cell
prop_cell_function= function(c){
  x= molecules_with_nonzero_all[molecules_with_nonzero_all$non_zero_cells_all==c, ]
  y=x[rownames(x) %in% rownames(molecules_only_SC),]
  return(dim(y)[1])
}

prop_cell_function(1)
prop_cell_function(2)
#do this in sapply for 0-864
prop_cells= c()

for (i in seq(1:864)){
  num= prop_cell_function(i)
  prop_cells=c(prop_cells, num)
}
plot(prop_cells/971, ylab="proportion of single cell only genes", xlab="number of cells")


# For all genes that are detected in single cell or bulk (19,027 genes), we condition on the 
# number of cells (ranging 1 - 864 cells) and given n number of cells, we look at 
# the proportion of genes that are detected in single cell only but not in bulk


```







###Only in Bulk
```{r}
bulk_mean <- apply(reads_bulk, 1, mean)
bulk_and_mean <- cbind(reads_bulk, bulk_mean)
#sort genes and mean by expression
bulk_and_order <- arrange(bulk_and_mean, desc(bulk_mean))

                               
                               
which_onlyBULK <- as.numeric(rownames(bulk_and_mean) %in% gene_onlybulk) + 1
col_vec_onlyBULK <- c("lightpink", "blue")[as.numeric(which_onlyBULK)]


plot((bulk_and_order$bulk_mean) , pch=46, cex=10, col=col_vec_onlyBULK, main = "Mean expression of reads in bulk samples", ylab=" read count", xlab = "genes (blue are only bulk genes)")

```



## Session information

<!-- Insert the session information into the document -->
```{r session-info}
sessionInfo()
```
