---
title: "Create gene-x-sample count matrices and annotation file"
output: html_document
---
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

```{r knitr-opts-chunk, include=FALSE}
```


**Last updated:** `r Sys.Date()`

**Code version:** `r workflowr::extract_commit(".", 1)$sha1`

This is recreating code from http://jdblischak.github.io/singleCellSeq/analysis/prepare-counts.html.

```{r}
library("data.table")
library("dplyr")
library("VennDiagram")
```

This analysis will take the matrix from the fastq to counts anayisis pipeline (sample-by-gene) to a standard gene-x-sample matrix. I will also create annotation files and romove the genes with no observed sequencing data in the experiment. I will use the data from git hub rather than the data I created because I want to look at the whole experiment rather than one plate at this point.  

###Read Counts for single cell samples: 
```{r}
reads_raw <- fread("../../singleCellSeq/data/reads-raw-single-per-sample.txt")
setDF(reads_raw)
```

Explore this data:
```{r}
dim(reads_raw)

```

Ths has rows for each individule and the columns give information about the sample then give counts for each gene.  
  
  
Create annotation file for single cell samples: 
```{r}
anno <- reads_raw %>%
  dplyr::select(individual:well) %>%
  mutate(batch = paste(individual, replicate, sep = "."),
         sample_id = paste(batch, well, sep = "."))
head(anno)
```

This annoation file has information about the sample, replicatte, batch.

Transpose the matrix so that it is gene-x-sample. 

```{r}
reads <- reads_raw %>%
  dplyr::select(starts_with("ENSG"), starts_with("ERCC")) %>%
  t
colnames(reads) <- anno$sample_id
reads[1:5, 1:5]
```

###Molecule Counts for single cell samples  
Import eaw molecules for single cell samples.

```{r}
molecules_raw <- fread("../../singleCellSeq/data/molecules-raw-single-per-sample.txt")
setDF(molecules_raw)
dim(molecules_raw)
```

This data has rows by individuale and has gene columns like the reads data. 

Conform single cell samples in reads and molecules files are in same order. 

```{r}
stopifnot(anno$individual == molecules_raw$individual,
          anno$replicate == molecules_raw$replicate,
          anno$well == molecules_raw$well)
```


Transpose the matrix so it is also gene-x-sample.  

```{r}
molecules <- molecules_raw %>%
  dplyr::select(starts_with("ENSG"), starts_with("ERCC")) %>%
  t
colnames(molecules) <- anno$sample_id
molecules[1:5, 1:5]
```
Now this file looks like the first one.

###Read counts for bulk samples  
Import raw read counts for bulk samples.  

```{r}
reads_bulk_raw <- fread("../../singleCellSeq/data/reads-raw-bulk-per-sample.txt")
setDF(reads_bulk_raw) #converts the data.table to a data.frame

```

Create annotation file for the bulk samples as we did with the single cells. 

```{r}
anno_bulk <- reads_bulk_raw %>%
  dplyr::select(individual:well) %>%
  mutate(batch = paste(individual, replicate, sep = "."),
         sample_id = paste(batch, well, sep = "."))
head(anno_bulk)
```

Now transpose the bulk read informaiton to match the gene-x-sample format.  

```{r}
reads_bulk <- reads_bulk_raw %>%
  dplyr::select(starts_with("ENSG"), starts_with("ERCC")) %>%
  t
colnames(reads_bulk) <- anno_bulk$sample_id
reads_bulk[1:5, 1:5]
```

###Observed genes 
I will look for genes out of the 20419 genes that were never observed in the experiment. 
```{r}
stopifnot(rownames(reads_bulk) == rownames(reads),
          rownames(reads) == rownames(molecules))

genes_observed_reads_bulk <- rownames(reads_bulk)[rowSums(reads_bulk) > 0]
genes_observed_reads <- rownames(reads)[rowSums(reads) > 0]
genes_observed_molecules <- rownames(molecules)[rowSums(molecules) > 0]
stopifnot(genes_observed_molecules %in% genes_observed_reads)
```
18726 genes has at least one observation in the single read data. 18726 genes had at least one observation in the single cell molecule data. 18056 genes has at least one observation in te bulk read data. As expected, all genes with at least one observed molecule in at least one single cell also had at least one observed read in at least one single cell.  

```{r}
genes_venn <- venn.diagram(x = list("reads" = genes_observed_reads,
                                    "molecules" = genes_observed_molecules,
                                    "reads bulk" = genes_observed_reads_bulk),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(genes_venn)
```

This ven diagram shows the intesection of the genes found in each type of sequencing data. I will now remove all genes with no observed read in either bulk or single cell sample.

```{r}
genes_observed <- union(genes_observed_reads, genes_observed_reads_bulk)
reads <- reads[rownames(reads) %in% genes_observed, ]
molecules <- molecules[rownames(molecules) %in% genes_observed, ]
reads_bulk <- reads_bulk[rownames(reads_bulk) %in% genes_observed, ]
```

This leaves 19027 genes. (genes_observed)  

###Write files 
This code outputs the annotation files.  
```{r}
write.table(anno, "../../singleCellSeq/data/annotation.txt", quote=FALSE, sep="\t", row.names = FALSE)
write.table(anno_bulk, "../../singleCellSeq/data/annotation-bulk.txt", quote=FALSE, sep = "\t", row.names=FALSE)
```

Output read counts. 
```{r}
write.table(reads, "../../singleCellSeq/data/reads.txt", quote = FALSE, sep = "\t", col.names = NA)
write.table(reads_bulk, "../../singleCellSeq/data/reads-bulk.txt", quote=FALSE, sep="\t", col.names = NA)

```
Output molecule counts.  
```{r}
write.table(molecules, "../../singleCellSeq/data/molecules.txt", quote= FALSE, sep= "\t", col.names = NA)
```


## Session Information

```{r session-info}
```
