---
title: "Genome wide data comparison"
author: "Briana Mittleman"
date: "2/6/2017"
output: html_document
---

```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

```{r knitr-opts-chunk, include=FALSE}
```

**Last updated:** `r Sys.Date()`

**Code version:** `r workflowr::extract_commit(".", 1)$sha1`

```{r}
library("biomaRt")
library("dplyr")
library("edgeR")
library("ggplot2")
theme_set(theme_bw(base_size = 12))
source("../../singleCellSeq/analysis/functions.R")
```

###Input read data:  

```{r}
reads <-read.table("../../singleCellSeq/data/reads.txt", header=TRUE, stringsAsFactors = FALSE)
#subset to look at same cells I did
reads_subset <-reads[ ,grep("NA19098.r1.A|NA19098.r1.B|NA19098.r1.C|NA19098.r1.D|NA19098.r1.E|NA19098.r1.F", colnames(reads))]
reads_bri<-read.table("../data/reads_bri.txt", header=TRUE, stringsAsFactors= FALSE)
```


Create a single matrix with the common genes in both sets.  

```{r}
#make my data different 
colnames(reads_bri)<- paste0(colnames(reads_bri), "-bri")
#list of common genes
common_genes <- intersect(row.names(reads_subset), row.names(reads_bri))
reads_subset_common <- reads_subset %>% filter(row.names(reads_subset) %in% common_genes)
row.names(reads_subset_common)<- common_genes
reads_bri_common <- reads_bri %>% filter(row.names(reads_bri) %in% common_genes)
row.names(reads_bri_common)<- common_genes
```

```{r}
#combine read matrics

reads_all_common <-cbind(reads_subset_common, reads_bri_common)
#remove genes that are never seen
genes_in_reads_all_common<- rownames(reads_all_common)[rowSums(reads_all_common) > 0]
reads_all_common_no0 <- reads_all_common %>% filter(row.names(reads_all_common) %in% genes_in_reads_all_common)
row.names(reads_all_common_no0) <- genes_in_reads_all_common
```




###Create plots for read data:  


```{r}
pca_commongenes <- run_pca(reads_all_common_no0)

#list_of_cells<- colnames(reads_all_common_no0)
#contains_bri<- c()
#for (i in colnames(reads_all_common_no0)) {
 #  if ("bri" %in i) 
#     contains_bri[contains_bri, "BRI"]
   
 # else
#    contains_bri[contains_bri, "GITHUB"]
#}  

#grep("bri", colnames(reads_all_common_no0))
  
  

pca_commongenes_plot <- plot_pca(pca_commongenes$PCs, explained = pca_commongenes$explained ) + labs(title="Reads in github and Bri analysis") 


#color=reads_all_common_no0[grep("bri", colnames(reads_all_common_no0)),])


pca_commongenes_plot

```



