---
title: "Home"
output:
  html_document:
    toc: false
---

This was made using the workflowr package in R.   

*  Easier to build and extend the website using the new tools in the [rmarkdown][] package and [latest RStudio release][rstudio]  

*  Easier to deploy the website with Git and GitHub by only using one branch  


[rmarkdown]: http://rmarkdown.rstudio.com/rmarkdown_websites.htm
[rstudio]: https://www.rstudio.com/products/rstudio/download/preview/

These are the projects I am working on during the my rotation in the Stephens lab. Each analysis file is treated like a lab notebook file for future anaylsis of single cell data.


* [Cell cycle single cell data](recreate_cell_cycle_analysis.html)

* [Gilad single cell fastq processing recreation](recreate_giladsinglecell.html)

* [Compare Gilad single cell counts to my analysis](compare_count_data.html)

* [Genome wide read count comparison between analysis](genome_wide_data_comparison.html)

Continue the Gilad single cell anaylsis:

* [Create gene-x-sample count matrices and annotation file](gene_x_sample_count_matrix.html)

* [QC single cell libraries](QC_single_cell_library.html)

* [Filter genes and find low quality single cells](filter_genes_low_quality_genes.html)

* [Quality control of UMIs and ERCC spike-ins](quality_control_plotes.html)

* [Compare reads to molecules counts](batch_effect_conversion_efficency.html)
