---
title: "Quality control of UMIs and ERCC spike-ins"
author: "Briana Mittleman"
date: "2/6/2017"
output: html_document
---

```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

```{r knitr-opts-chunk, include=FALSE}
```

**Last updated:** `r Sys.Date()`

**Code version:** `r workflowr::extract_commit(".", 1)$sha1`

###Input
```{r}
library("biomaRt")
library("data.table")
library("dplyr")
library("limma")
library("ggplot2")
library("grid")
library("cowplot")
theme_set(theme_bw(base_size = 16))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank(),
             legend.key = element_blank(),
             plot.title = element_text(size = rel(1)))
source("../../singleCellSeq/analysis/functions.R")
```

Input ERCC molecule counts calculated in capture efficiency.  

```{r}
ercc <- read.table("../../singleCellSeq/data/expected-ercc-molecules.txt", header = TRUE,
                   stringsAsFactors = FALSE)
head(ercc)
```

Input annotation.  

```{r}
anno <- read.table("../../singleCellSeq/data/annotation.txt", header=TRUE, stringsAsFactors = FALSE)
head(anno)
```

Input read counts and filter for quality cells.  

```{r}
reads<- read.table("../../singleCellSeq/data/reads.txt", header=TRUE, stringsAsFactors = FALSE)
quality_single_cells <- scan("../data/quality-single-cells.txt", what = "character")
reads<- reads[, colnames(reads) %in% quality_single_cells]

```
Input read counts in high quality cells for filtered genes.    

```{r}
reads_filter <- read.table("../data/reads-filter.txt", header=TRUE, stringsAsFactors = FALSE)

```

Input molecule counts and filter for quality cells.  

```{r}
molecules <- read.table("../../singleCellSeq/data/molecules.txt", header=TRUE, stringsAsFactors = FALSE)
molecules <- molecules[,colnames(molecules) %in% quality_single_cells]
```

Input molecule counts in high quality cells for filtered genes.  

```{r}
molecules_filter <- read.table("../data/molecules-filter.txt", header = TRUE, stringsAsFactors = FALSE)
```

###Compare reads and molecules  

```{r}
## calculate mean accross all cells for each gene 
#in apply- 1=rows, 2=columns
reads_mean     <- apply(reads, 1, mean)
molecules_mean <- apply(molecules, 1, mean)
distribution <- data.frame(reads_mean, molecules_mean)

reads_filter_mean <- apply(reads_filter, 1, mean)
molecules_filter_mean <- apply(molecules_filter, 1, mean)
distribution_filter <- data.frame(reads_filter_mean, molecules_filter_mean)

## correlation between reads and molecules
cor(distribution)
cor(distribution_filter)
```
There is a higher correlation between molecules and read counts in the filtered data. I would expect a lower correlation if the reads come from sequencing each molecule multiple times. This could be driven by the large numbers of zeros in the data set. 

```{r}
##select ERCC
#distribution has the mean read and molecule count for each gene
distribution$type <- ifelse(grepl("ERCC", rownames(distribution)), "ERCC", "gene")
distribution_filter$type <- ifelse(grepl("ERCC", rownames(distribution_filter)), "ERCC", "gene")
```

