---
title: "recreate_cell_cycle_analysis"
author: "Briana Mittleman"
date: 2017-01-06
output: html_document
---

```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

```{r knitr-opts-chunk, include=FALSE}
```

**Last updated:** `r Sys.Date()`

**Code version:** `r workflowr::extract_commit(".", 1)$sha1`

General information for committing changes to the git gub site 

``` 
git add analysis/*.Rmd
git commit -m "message"
```
Run `commit_site()` in R terminal.

```
 git push origin master
```



#### Configure program
First I need to install the 2013-03-31 version of gmap in my directory on the midway cluster. I used the instructions in the INSTALL text file within the program folder downloaded from online.

```
./configure --prefix=/home/brimittleman/software 
make
make install
```


#### Build reference 
Download the mouse genome for mapping my reads from: http://may2012.archive.ensembl.org/Mus_musculus/Info/Index  


Building the mouse reference database on gsnap. These are some of the options in the function I will need to run the reference build.  

```
    -D, --dir=STRING    destination ("/home/brimittleman/software"")
    
    -k, --kmer=INT      kmer value for gneome (default=15)
        
    -d, --db=STRING     Genome name
    
    -g                  if files are gziped 
```
Secifically I used this code to run the reference build. 

```
./gmap_build -g -d mouseref ../../mouse\ genome/*dna.chrom*fa.gz
```
  
Submit this job to a computing node and requested 8g using command:  

```

sinteractive --mem=8g

```

The files from the assembly are in /home/brimittleman/software/share/mouseref. 

Download ERCC sequences to add to map.  
see : https://github.com/jdblischak/singleCellSeq/blob/2f6371b256829bfa799270909b6eea2814acf2a1/code/create-genome.sh

```
wget http://tools.thermofisher.com/downloads/ERCC92.fa

```

Combine the mouse reference and ERCC sequences and rebuild the genome. Need to unzip the mouse fastq files so I can run the build fucntion with the ERCC and the mouse files. Perform these commands in the mouse ref directory where the files are held.  

```
gunzip *dna.chrom*fa.gz
```
In the software file I can now rerun the build with this command in the software/bin directory.
```
 ./gmap_build -d mouseref_ERCC  ../../mouse\ genome/*.fa 
```


#### Running GSNAP:  

The general documentation looks like this. ```gsnap --gunzip -d <genome> <fastq_1.gz> <fastq_2.gz>```

To test this I will start with running the program on cell 47.  

Files: **ERR489047_2.fastq.gz** 
       **ERR489047_1.fastq.gz**   

Code I am working on for this allignment. Not correct yet. 

```
./gsnap --gunzip -d -D  ../share/mouseref_ERCC ../../scratch-midway/ERR489047_1.fastq.gz ../../scratch-midway/ERR489047_2.fastq.gz
```
