---
title: "Look at single cell only genes by individual"
author: "Briana Mittleman"
date: 2017/03/03
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

###Pull in genes lists  

```{r}
library("biomaRt")
library("data.table")
library("testit")
library("dplyr")
library("edgeR")
library("VennDiagram")
library("ggplot2")
theme_set(theme_bw(base_size = 12))
source("../../singleCellSeq/analysis/functions.R")
require(Humanzee)
library("GOstats")

```



```{r}
anno <- read.table("../../singleCellSeq/data/annotation.txt", header = TRUE,
                   stringsAsFactors = FALSE)
#input reads (pre-filter)
reads <- read.table("../../singleCellSeq/data/reads.txt", header = TRUE,
                    stringsAsFactors = FALSE)
#input molecule (pre-filter)
molecules <- read.table("../../singleCellSeq/data/molecules.txt", header = TRUE,
                    stringsAsFactors = FALSE)
#input bulk reads (pre-filter )
reads_bulk <- read.table("../../singleCellSeq/data/reads-bulk.txt", header = TRUE, stringsAsFactors = FALSE)
```
Create vectors with the list of genes in each set.   
```{r}
genes_observed_reads_bulk <- rownames(reads_bulk)[rowSums(reads_bulk) > 0]
genes_observed_reads_bulk_ENSG= genes_observed_reads_bulk[grep("ERCC", genes_observed_reads_bulk, invert = TRUE)]
genes_observed_reads <- rownames(reads)[rowSums(reads) > 0]
genes_observed_molecules <- rownames(molecules)[rowSums(molecules) > 0]
genes_observed_molecules_ENSG <- genes_observed_molecules[grep("ERCC", genes_observed_molecules, invert = TRUE)]

```

####Genes only in single cell molecules  

```{r}

gene_onlysinglecell_ENSG <- setdiff(genes_observed_molecules_ENSG,genes_observed_reads_bulk_ENSG)
#subset molecules by these genes  
molecules_df= data.frame(molecules)
molecules_only_SC_ENSG <- molecules_df[rownames(molecules_df) %in% gene_onlysinglecell_ENSG,]
dim(molecules_only_SC_ENSG)
```


```{r, eval=FALSE}
save(molecules_only_SC_ENSG, file="../data/molecules_only_SC_ENSG.Rda")
```


Count number of cells we see these genes in  

```{r}
non_zero_cells_ENSG <- apply(molecules_only_SC_ENSG, 1, function(c)sum(c!=0))

molecules_with_nonzero_ENSG <- cbind(molecules_only_SC_ENSG, non_zero_cells_ENSG)

summary(molecules_with_nonzero_ENSG$non_zero_cells_ENSG)

```
##Individual level analysis  

###Individual 19098   

```{r}
molecules_SC_ENSG_19098= molecules_with_nonzero_ENSG[,grep("19098", colnames(molecules_with_nonzero_ENSG))]
# this is a dataframe for the sc only genes in 19098
dim(molecules_SC_ENSG_19098)

non_zero_cells_ENSG_19098 <- apply(molecules_SC_ENSG_19098, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19098 <- cbind(molecules_SC_ENSG_19098, non_zero_cells_ENSG_19098)

summary(molecules_with_nonzero_ENSG_19098$non_zero_cells_ENSG_19098)
plot(molecules_with_nonzero_ENSG_19098$non_zero_cells_ENSG_19098)
```

###Individual 19101  
```{r}
molecules_SC_ENSG_19101= molecules_with_nonzero_ENSG[,grep("19101", colnames(molecules_with_nonzero_ENSG))]
# this is a dataframe for the sc only genes in 19101

dim(molecules_SC_ENSG_19101)

non_zero_cells_ENSG_19101 <- apply(molecules_SC_ENSG_19101, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19101 <- cbind(molecules_SC_ENSG_19101, non_zero_cells_ENSG_19101)

summary(molecules_with_nonzero_ENSG_19101$non_zero_cells_ENSG_19101)
plot(molecules_with_nonzero_ENSG_19101$non_zero_cells_ENSG_19101)
```
  
###Individual 19239   

```{r}
molecules_SC_ENSG_19239= molecules_with_nonzero_ENSG[,grep("19239", colnames(molecules_with_nonzero_ENSG))]
# this is a dataframe for the sc only genes in 19239

dim(molecules_SC_ENSG_19239)

non_zero_cells_ENSG_19239 <- apply(molecules_SC_ENSG_19239, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19239 <- cbind(molecules_SC_ENSG_19239, non_zero_cells_ENSG_19239)

summary(molecules_with_nonzero_ENSG_19239$non_zero_cells_ENSG_19239)
plot(molecules_with_nonzero_ENSG_19239$non_zero_cells_ENSG_19239)
```


###Compare individuals  

I want to look at the genes that are expressed in these individuals and see if there are genes that are unique to one individual  

* find genes with non_zero > 1 (exclude singleton)

* make a list of these genes  

* create a ven diagram with these genes  



```{r extract non singleton gene lists}
#19098
genes_not_singleton_19098= row.names(molecules_with_nonzero_ENSG_19098[molecules_with_nonzero_ENSG_19098$non_zero_cells_ENSG_19098 > 1, ])
length(genes_not_singleton_19098)

#19101
genes_not_singleton_19101= row.names(molecules_with_nonzero_ENSG_19101[molecules_with_nonzero_ENSG_19101$non_zero_cells_ENSG_19101 > 1, ])
length(genes_not_singleton_19101)

#19239
genes_not_singleton_19239= row.names(molecules_with_nonzero_ENSG_19239[molecules_with_nonzero_ENSG_19239$non_zero_cells_ENSG_19239 > 1, ])
length(genes_not_singleton_19239)

```

```{r ven diagram}
non_sing_venn <- venn.diagram(x = list("19098" = genes_not_singleton_19098,
                                    "19101" = genes_not_singleton_19101,
                                    "19239" = genes_not_singleton_19239),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(non_sing_venn)
```

##Replicate level analysis 
###19098
####R1  

```{r}
molecules_SC_ENSG_19098.R1= molecules_with_nonzero_ENSG_19098[,grep("r1", colnames(molecules_with_nonzero_ENSG_19098))]

non_zero_cells_ENSG_19098.R1 <- apply(molecules_SC_ENSG_19098.R1, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19098.R1 <- cbind(molecules_SC_ENSG_19098.R1, non_zero_cells_ENSG_19098.R1)

genes_not_singleton_19098.R1= row.names(molecules_with_nonzero_ENSG_19098.R1[molecules_with_nonzero_ENSG_19098.R1$non_zero_cells_ENSG_19098.R1  > 1, ])
length(genes_not_singleton_19098.R1)
```
####R2  

```{r}
molecules_SC_ENSG_19098.R2= molecules_with_nonzero_ENSG_19098[,grep("r2", colnames(molecules_with_nonzero_ENSG_19098))]

non_zero_cells_ENSG_19098.R2 <- apply(molecules_SC_ENSG_19098.R2, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19098.R2 <- cbind(molecules_SC_ENSG_19098.R2, non_zero_cells_ENSG_19098.R2)

genes_not_singleton_19098.R2= row.names(molecules_with_nonzero_ENSG_19098.R2[molecules_with_nonzero_ENSG_19098.R2$non_zero_cells_ENSG_19098.R2  > 1, ])
length(genes_not_singleton_19098.R2)
```
####R3  

```{r}
molecules_SC_ENSG_19098.R3= molecules_with_nonzero_ENSG_19098[,grep("r3", colnames(molecules_with_nonzero_ENSG_19098))]

non_zero_cells_ENSG_19098.R3 <- apply(molecules_SC_ENSG_19098.R3, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19098.R3 <- cbind(molecules_SC_ENSG_19098.R3, non_zero_cells_ENSG_19098.R3)

genes_not_singleton_19098.R3= row.names(molecules_with_nonzero_ENSG_19098.R3[molecules_with_nonzero_ENSG_19098.R3$non_zero_cells_ENSG_19098.R3  > 1, ])
length(genes_not_singleton_19098.R3)
```

####Create ven diagram  

```{r}
non_sing_venn_19098 <- venn.diagram(x = list("R1" = genes_not_singleton_19098.R1,
                                    "R2" = genes_not_singleton_19098.R2,
                                    "R3" = genes_not_singleton_19098.R3),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(non_sing_venn_19098)
```






###19101
####R1  

```{r}
molecules_SC_ENSG_19101.R1= molecules_with_nonzero_ENSG_19101[,grep("r1", colnames(molecules_with_nonzero_ENSG_19101))]

non_zero_cells_ENSG_19101.R1 <- apply(molecules_SC_ENSG_19101.R1, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19101.R1 <- cbind(molecules_SC_ENSG_19101.R1, non_zero_cells_ENSG_19101.R1)

genes_not_singleton_19101.R1= row.names(molecules_with_nonzero_ENSG_19101.R1[molecules_with_nonzero_ENSG_19101.R1$non_zero_cells_ENSG_19101.R1  > 1, ])
length(genes_not_singleton_19101.R1)
```
####R2  

```{r}
molecules_SC_ENSG_19101.R2= molecules_with_nonzero_ENSG_19101[,grep("r2", colnames(molecules_with_nonzero_ENSG_19101))]

non_zero_cells_ENSG_19101.R2 <- apply(molecules_SC_ENSG_19101.R2, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19101.R2 <- cbind(molecules_SC_ENSG_19101.R2, non_zero_cells_ENSG_19101.R2)

genes_not_singleton_19101.R2= row.names(molecules_with_nonzero_ENSG_19101.R2[molecules_with_nonzero_ENSG_19101.R2$non_zero_cells_ENSG_19101.R2  > 1, ])
length(genes_not_singleton_19101.R2)
```
####R3  

```{r}

molecules_SC_ENSG_19101.R3= molecules_with_nonzero_ENSG_19101[,grep("r3", colnames(molecules_with_nonzero_ENSG_19101))]

non_zero_cells_ENSG_19101.R3 <- apply(molecules_SC_ENSG_19101.R3, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19101.R3 <- cbind(molecules_SC_ENSG_19101.R3, non_zero_cells_ENSG_19101.R3)

genes_not_singleton_19101.R3= row.names(molecules_with_nonzero_ENSG_19101.R3[molecules_with_nonzero_ENSG_19101.R3$non_zero_cells_ENSG_19101.R3  > 1, ])
length(genes_not_singleton_19101.R3)
```

####Create ven diagram  

```{r}
non_sing_venn_19101 <- venn.diagram(x = list("R1" = genes_not_singleton_19101.R1,
                                    "R2" = genes_not_singleton_19101.R2,
                                    "R3" = genes_not_singleton_19101.R3),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(non_sing_venn_19101)
```

###19239
####R1  

```{r}
molecules_SC_ENSG_19239.R1= molecules_with_nonzero_ENSG_19239[,grep("r1", colnames(molecules_with_nonzero_ENSG_19239))]

non_zero_cells_ENSG_19239.R1 <- apply(molecules_SC_ENSG_19239.R1, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19239.R1 <- cbind(molecules_SC_ENSG_19239.R1, non_zero_cells_ENSG_19239.R1)

genes_not_singleton_19239.R1= row.names(molecules_with_nonzero_ENSG_19239.R1[molecules_with_nonzero_ENSG_19239.R1$non_zero_cells_ENSG_19239.R1  > 1, ])
length(genes_not_singleton_19239.R1)
```
####R2  

```{r}
molecules_SC_ENSG_19239.R2= molecules_with_nonzero_ENSG_19239[,grep("r2", colnames(molecules_with_nonzero_ENSG_19239))]

non_zero_cells_ENSG_19239.R2 <- apply(molecules_SC_ENSG_19239.R2, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19239.R2 <- cbind(molecules_SC_ENSG_19239.R2, non_zero_cells_ENSG_19239.R2)

genes_not_singleton_19239.R2= row.names(molecules_with_nonzero_ENSG_19239.R2[molecules_with_nonzero_ENSG_19239.R2$non_zero_cells_ENSG_19239.R2  > 1, ])
length(genes_not_singleton_19239.R2)
```
####R3  

```{r}

molecules_SC_ENSG_19239.R3= molecules_with_nonzero_ENSG_19239[,grep("r3", colnames(molecules_with_nonzero_ENSG_19239))]

non_zero_cells_ENSG_19239.R3 <- apply(molecules_SC_ENSG_19239.R3, 1, function(c)sum(c!=0))
molecules_with_nonzero_ENSG_19239.R3 <- cbind(molecules_SC_ENSG_19239.R3, non_zero_cells_ENSG_19239.R3)

genes_not_singleton_19239.R3= row.names(molecules_with_nonzero_ENSG_19239.R3[molecules_with_nonzero_ENSG_19239.R3$non_zero_cells_ENSG_19239.R3  > 1, ])
length(genes_not_singleton_19239.R3)
```

####Create ven diagram  

```{r}
non_sing_venn_19239<- venn.diagram(x = list("R1" = genes_not_singleton_19239.R1,
                                    "R2" = genes_not_singleton_19239.R2,
                                    "R3" = genes_not_singleton_19239.R3),
                           filename = NULL, euler.d = FALSE, scaled = FALSE)
grid.newpage()
grid.draw(non_sing_venn_19239)
```

<<<<<<< HEAD

##PCA for SC only genes  

```{r}
pca_SConly <- run_pca(molecules_only_SC_ENSG )
pca_SConly_plot <- plot_pca(pca_SConly$PCs, explained = pca_SConly$explained,
         metadata = anno, color = "individual",
         shape = "replicate", alpha = 0.5, size = 3) +
        labs(title="PCA for single cell only genes") +
        geom_text(aes(label= ifelse(PC1 < -60  |  PC2 >40,colnames(molecules_only_SC_ENSG), "") , vjust="inward", hjust="inward"))
                      
                  

pca_SConly_plot
```
The green one is NA19101.r1.C11 and the red one is NA19098.r1.F01  


Check if these cells passes the quality control steps in the original analysis.  

```{r}
quality_single_cells <- scan("../../singleCellSeq//data/quality-single-cells.txt",
                             what = "character")



green_cell= c("NA19101.r1.C11")
red_cell=c("NA19098.r1.F01")

green_cell %in% quality_single_cells

red_cell %in% quality_single_cells
```

Cell NA19101.r1.C11 is in the list of quality cells. Cell NA19098.r1.F01 was not included in the quality cells. 

###Exclude singletons  

```{r}
#ENSG genes that arnt singleton in single cell only
load("../data/gene_names_ensg.txt")
#subset molecules for this  

molecules_sc_notsing_ENSG= molecules[rownames(molecules) %in% gene_names_ensg, ]

```

run and plot PCA  

```{r}
pca_SConly_nonsing <- run_pca(molecules_sc_notsing_ENSG)
pca_SConly_nonsing_plot <- plot_pca(pca_SConly_nonsing$PCs, explained = pca_SConly_nonsing$explained,
         metadata = anno, color = "individual",
         shape = "replicate", alpha = 0.5, size = 3) +
        labs(title="PCA for single cell only genes seen in more than 1 cell") +
  geom_text(aes(label= ifelse(PC1 < -15  |  PC2 >10,colnames(molecules_only_SC_ENSG), "") , vjust="inward", hjust="inward"))
pca_SConly_nonsing_plot 
```


=======
## Session information

<!-- Insert the session information into the document -->
```{r session-info}
sessionInfo()
```
>>>>>>> 86594a472ae2f2872b38459580b5cf8e13b8b533
