---
title: "Final Subsample plots"
author: "Briana Mittleman"
date: 2016-02-08
output: html_document
---

```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

```{r knitr-opts-chunk, include=FALSE}
```

**Last updated:** `r Sys.Date()`

**Code version:** `r workflowr::extract_commit(".", 1)$sha1`

A subsample of the full data, post mapping, was processed to create the count data used here.  


Extra information about the subsample was created using the subsample-pipeline.py pipeline.  
###Input  
```{r}
library("dplyr")
library("tidyr")
library("ggplot2")
library("cowplot")
theme_set(theme_bw(base_size = 12))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank(),
             legend.key = element_blank(),
             plot.title = element_text(size = rel(1)))
```

```{r}
d <- read.table("../../singleCellSeq/data/subsampling-results.txt",
                header = TRUE, sep = "\t", stringsAsFactors = FALSE)
str(d)
```
```{r}
d_grouped <- d %>%
  group_by(type, depth, gene_subset, subsampled_cells,
           individual, potential_cells, available_cells,
           lower_q, upper_q, available_ensg, used_ensg,
           available_ercc, used_ercc) %>%
  summarize(mean_detected = mean(detected_ensg),
            sem_detected = sd(detected_ensg) / sqrt(length(detected_ensg)),
            mean_bulk = mean(pearson_ensg),
            sem_bulk = sd(pearson_ensg) / sqrt(length(pearson_ensg)),
            mean_var = mean(var_pearson),
            sem_var = sd(var_pearson) / sqrt(length(var_pearson)))
```


```{r}
d_filter <- d_grouped %>% filter(individual == "NA19239",
                                type == "molecules",
                                gene_subset %in% c("lower", "upper"))
d_filter$gene_subset <- factor(d_filter$gene_subset,
                               levels = c("lower", "upper"),
                               labels = c("Lower 50% of expressed genes",
                                          "Upper 50% of expressed genes"))

```

###Figures  
```{r}
plot_bulk_title <- "Correlation of gene expression levels \n between single cells and the corresponding bulk samples"
plot_bulk_title <- "Correlation of gene expression levels between single cells and the corresponding bulk samples"

plot_bulk <- ggplot(d_filter,
                 aes(x = subsampled_cells, y = mean_bulk,
                     color = as.factor(depth))) +
  geom_line() +
  geom_errorbar(aes(ymin = mean_bulk - sem_bulk,
                    ymax = mean_bulk + sem_bulk),
                width = 1) +
  facet_wrap(~gene_subset) +
  scale_color_grey(start = 0.8, end = 0.2, name = "Sequencing depth") +
  theme(legend.position = "none") +
  labs(x = "Number of subsampled cells",
       y = "Pearson's r (+/- SEM)",
       title = paste(strwrap(plot_bulk_title, width = 80), collapse = "\n"))
plot_bulk
```
This is comparing a subset of the single cells to the bulk samples. We are looking at the the correlation as a function of how many of the sampled cells the gene was found in.  This uses molecule counts from individual NA19239.  

```{r}
plot_detected <- ggplot(d_filter,
                 aes(x = subsampled_cells, y = mean_detected,
                     color = as.factor(depth))) +
  geom_line() +
  geom_errorbar(aes(ymin = mean_detected - sem_detected,
                    ymax = mean_detected + sem_detected),
                width = 1) +
  facet_wrap(~gene_subset) +
  scale_color_grey(start = 0.8, end = 0.2, name = "Sequencing depth",
                   labels = format(unique(d$depth), big.mark = ",",
                                   scientifc = FALSE, trim = TRUE)) +
  theme(legend.position = c(0.75, 0.35),
        legend.key.size = grid::unit(0.2, "in")) +
  labs(x = "Number of subsampled cells",
       y = "Number of genes detected",
       title = "Number of genes detected as expressed in single cell samples \n at different sequence coverage")
plot_detected
```
This plot looks at the number of genes detected at diffent sequencing depths as a function of the number of subsampled cells used in this analysis.  

```{r}
plot_var <- ggplot(d_filter,
                 aes(x = subsampled_cells, y = mean_var,
                     color = as.factor(depth))) +
  geom_line() +
  geom_errorbar(aes(ymin = mean_var - sem_var,
                    ymax = mean_var + sem_var),
                width = 1) +
  facet_wrap(~gene_subset) +
  scale_color_grey(start = 0.8, end = 0.2, name = "Sequencing depth") +
  theme(legend.position = "none") +
  labs(x = "Number of subsampled cells",
       y = "Pearson's r (+/- SEM)",
       title = "Correlation of the variation in gene expression levels \n between subsets of single cells and the entire single cell dataset")
plot_var
```
Correlation between subsample and all of the cell in terms of the variance gene expression levels. THe different colors are the sequencing depth.  


Plot these next to eachother in a grid.  

```{r}
plot_final <- plot_grid(plot_bulk, plot_detected, plot_var,
                        ncol = 1, labels = letters[1:3], label_size = 12)
plot_final
```
```{r}
# png("../paper/figure/fig-subsample.png", width = 8, height = 12, units = "in", res = 300)
tiff("../paper/figure/fig-subsample.tiff",
     width = 8, height = 12, units = "in", res = 300, compression = "zip")
plot_final
dev.off()
```

###Number of molecules per sequencing depth  

```{r}
molecules_per_depth <- d %>%
  filter(gene_subset == "all", type == "molecules")
#subset for only molecule rows  
```


```{r}
#molecule depth for endogenous genes 
plot_mol_depth_ensg <- ggplot(molecules_per_depth,
                              aes(x = as.factor(depth),
                                  y = mean_counts_ensg)) +
  geom_boxplot() +
  facet_wrap(~individual) +
  scale_y_continuous(breaks = seq(0, 1e5, by = 1e4)) +
  theme(axis.text.x = element_text(angle = 90))
plot_mol_depth_ensg
```

```{r}
#molecule depth for ERCC
plot_mol_depth_ercc <- plot_mol_depth_ensg %+% aes(y = mean_counts_ercc)
plot_mol_depth_ercc
```
###Session info
```{r}
sessionInfo()
```

